

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <title>Gromos Tutorial &mdash; PyGromosTools  documentation</title>








  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>


      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script src="https://unpkg.com/@jupyter-widgets/html-manager@^0.20.0/dist/embed-amd.js"></script>
        <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>

    <script type="text/javascript" src="../_static/js/theme.js"></script>




  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Documentation" href="../_source/modules.html" />
    <link rel="prev" title="Tutorials" href="index.html" />
</head>

<body class="wy-body-for-nav">


  <div class="wy-grid-for-nav">

    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >



            <a href="../index.html" class="icon icon-home"> PyGromosTools



          </a>







<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>


        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">






              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">Welcome to PyGromosTools</a></li>
<li class="toctree-l1"><a class="reference internal" href="../introduction.html#define-file-paths">define file paths</a></li>
<li class="toctree-l1"><a class="reference internal" href="../introduction.html#build-gromos-system">Build gromos System</a></li>
<li class="toctree-l1"><a class="reference internal" href="../introduction.html#run-emin">Run Emin</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Examples/index.html">Examples</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">Tutorials</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">Gromos Tutorial</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Build-the-input-files">Build the input files</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#The-Topology-File">The Topology File</a></li>
<li class="toctree-l4"><a class="reference internal" href="#The-Coordinate-File">The Coordinate File</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Energy-Minimization---Vacuum">Energy Minimization - Vacuum</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#Solvatistation-and-Solvent-Energy-Minimization">Solvatistation and Solvent Energy Minimization</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#build-box-system">build box system</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Add-Ions">Add Ions</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Energy-Minimization-BOX">Energy Minimization BOX</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#Thermalisation-and-equilibration.">Thermalisation and equilibration.</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#Simulation-Paramters">Simulation Paramters</a></li>
<li class="toctree-l4"><a class="reference internal" href="#System-Block">System Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#COMTRANSROT-Block">COMTRANSROT Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#COVALENTFORM-Block">COVALENTFORM Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#WRITETRAJ-Block">WRITETRAJ Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#PRINTOUT-Block">PRINTOUT Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#PAIRLIST-Block">PAIRLIST Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#POSITIONRES-Block">POSITIONRES Block</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Perform-Thermalisation">Perform Thermalisation</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Analysis-of-Thermalisation:">Analysis of Thermalisation:</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#Molecular-dynamics-sampling-simulation.">Molecular dynamics sampling simulation.</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id4">Simulation Paramters</a></li>
<li class="toctree-l4"><a class="reference internal" href="#Perform-MD---Production">Perform MD - Production</a></li>
<li class="toctree-l4"><a class="reference internal" href="#id5">Analysis</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../_source/modules.html">Documentation</a></li>
</ul>



        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">


      <nav class="wy-nav-top" aria-label="top navigation">

          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">PyGromosTools</a>

      </nav>


      <div class="wy-nav-content">

        <div class="rst-content">

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">

      <li><a href="../index.html">Docs</a> &raquo;</li>

          <li><a href="index.html">Tutorials</a> &raquo;</li>

      <li>Gromos Tutorial</li>


      <li class="wy-breadcrumbs-aside">


            <a href="../_sources/Tutorials/tutorial_gromos_pipeline.ipynb.txt" rel="nofollow"> View page source</a>


      </li>

  </ul>


  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">


<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container div.prompt *,
div.nboutput.container div.prompt *,
div.nbinput.container div.input_area pre,
div.nboutput.container div.output_area pre,
div.nbinput.container div.input_area .highlight,
div.nboutput.container div.output_area .highlight {
    border: none;
    padding: 0;
    margin: 0;
    box-shadow: none;
}

div.nbinput.container > div[class*=highlight],
div.nboutput.container > div[class*=highlight] {
    margin: 0;
}

div.nbinput.container div.prompt *,
div.nboutput.container div.prompt * {
    background: none;
}

div.nboutput.container div.output_area .highlight,
div.nboutput.container div.output_area pre {
    background: unset;
}

div.nboutput.container div.output_area div.highlight {
    color: unset;  /* override Pygments text color */
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    /*background: #f5f5f5;*/
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
    margin: 0;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt .copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<section id="Gromos-Tutorial">
<h1>Gromos Tutorial<a class="headerlink" href="#Gromos-Tutorial" title="Permalink to this headline">¶</a></h1>
<p>This tutorial will provide a quick introduction to how to setup, perform and analyze a MD simulation of a peptide in Gromos.</p>
<p>First we will generated the required input files, afterwards the simulation will be prepared with energy minimizations and equilibration. Finally a short MD simulation will be executed and quickly analyzed.</p>
<p>But first some minor definitions will be prepared.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="c1"># Check if the path to this pacakge is set, else try to add it.</span>
<span class="k">try</span><span class="p">:</span>
    <span class="kn">import</span> <span class="nn">pygromos</span>
<span class="k">except</span><span class="p">:</span>
    <span class="kn">import</span> <span class="nn">os</span><span class="o">,</span> <span class="nn">sys</span><span class="o">,</span> <span class="nn">copy</span>
    <span class="n">root_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">()</span>
    <span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">root_dir</span><span class="o">+</span><span class="s2">&quot;/..&quot;</span><span class="p">)</span>

    <span class="c1">#if package is not installed and path not set correct - this helps you out :)</span>
    <span class="kn">import</span> <span class="nn">pygromos</span>

<span class="kn">from</span> <span class="nn">pygromos.utils</span> <span class="kn">import</span> <span class="n">bash</span>
<br/></pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#General path definitions:</span>
<span class="c1">## Project dir - The project dir will contain  all files and results generated from this notebook.</span>
<span class="n">project_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">abspath</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">dirname</span><span class="p">(</span><span class="n">pygromos</span><span class="o">.</span><span class="vm">__file__</span><span class="p">)</span><span class="o">+</span><span class="s2">&quot;/../examples/example_files/Tutorial_System&quot;</span><span class="p">)</span>
<span class="n">input_dir</span> <span class="o">=</span> <span class="n">project_dir</span><span class="o">+</span><span class="s2">&quot;/input&quot;</span> <span class="c1"># contains prepared files (pdb of the peptide)</span>

<span class="c1">## Gromos Bin Path</span>
<span class="n">gromosXX_bin</span> <span class="o">=</span> <span class="s2">&quot;/home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin&quot;</span>
<span class="n">gromosPP_bin</span> <span class="o">=</span> <span class="kc">None</span> <span class="c1">#None uses the Path variable - can be used . if you require a specific compiled gromos version, add a path here.</span>
<br/><br/></pre></div>
</div>
</div>
<section id="Build-the-input-files">
<h2>Build the input files<a class="headerlink" href="#Build-the-input-files" title="Permalink to this headline">¶</a></h2>
<p>When a simulation study of a particular system or process is to be carried out, a number of choices have to be made with respect to the set-up of the simulation.</p>
<p>In PyGromosTools, the central hub to manage all these files is a GromosSystem object. Each file can be stored or generated with this object and it can be used to construct the system, that should be simulated.</p>
<p>The essential files of a Simulation are: * a topology file, containing all the topological and force-field information of the molecular system to be studied. * a coordinate file, representing the system of interest. * a simulation parameter file, telling the simulation engine, which simulation technique should be used and which physical parameters should be set.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.files.gromos_system</span> <span class="kn">import</span> <span class="n">Gromos_System</span>

<span class="c1"># Build Gromos System object</span>
<span class="n">build_system</span> <span class="o">=</span> <span class="n">Gromos_System</span><span class="p">(</span><span class="n">work_folder</span><span class="o">=</span><span class="n">project_dir</span><span class="p">,</span> <span class="n">system_name</span><span class="o">=</span><span class="s1">&#39;peptide&#39;</span><span class="p">)</span>

<span class="c1">#set file building folder</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">bash</span><span class="o">.</span><span class="n">make_folder</span><span class="p">(</span><span class="n">project_dir</span><span class="o">+</span><span class="s2">&quot;/a_build_initial_files&quot;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"version_major": 2, "version_minor": 0, "model_id": "0510be9c94334d1db60232e2545d4e9a"}</script></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/mlehner/PyGromosTools/pygromos/files/gromos_system/gromos_system.py:869: UserWarning: Did not change file path as its only promised None
  warnings.warn(&#34;Did not change file path as its only promised &#34; + str(file_obj.path))
</pre></div></div>
</div>
<p><strong>Expert Tip</strong>: Whenever you wonder about what a class or a function in our package does, you can use the standard help function of Python to get the information of the Docstring of the source code! This often helps understanding the code.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">build_system</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on Gromos_System in module pygromos.files.gromos_system.gromos_system object:

class Gromos_System(builtins.object)
 |  Gromos_System(work_folder: str, system_name: str, rdkitMol: rdkit.Chem.rdchem.Mol = None, in_mol2_file: str = None, readIn: bool = True, forcefield: pygromos.files.forcefield._generic_force_field._generic_force_field = &lt;pygromos.files.forcefield._generic_force_field._generic_force_field object at 0x7f3ac20207f0&gt;, auto_convert: bool = False, adapt_imd_automatically: bool = True, verbose: bool = False, in_smiles: str = None, in_residue_list: List = None, in_top_path: str = None, in_cnf_path: str = None, in_imd_path: str = None, in_disres_path: str = None, in_ptp_path: str = None, in_posres_path: str = None, in_refpos_path: str = None, in_qmmm_path: str = None, in_gromosXX_bin_dir: str = None, in_gromosPP_bin_dir: str = None)
 |
 |  Methods defined here:
 |
 |  __deepcopy__(self, memo)
 |
 |  __getstate__(self)
 |      preperation for pickling:
 |      remove the non trivial pickling parts
 |
 |  __init__(self, work_folder: str, system_name: str, rdkitMol: rdkit.Chem.rdchem.Mol = None, in_mol2_file: str = None, readIn: bool = True, forcefield: pygromos.files.forcefield._generic_force_field._generic_force_field = &lt;pygromos.files.forcefield._generic_force_field._generic_force_field object at 0x7f3ac20207f0&gt;, auto_convert: bool = False, adapt_imd_automatically: bool = True, verbose: bool = False, in_smiles: str = None, in_residue_list: List = None, in_top_path: str = None, in_cnf_path: str = None, in_imd_path: str = None, in_disres_path: str = None, in_ptp_path: str = None, in_posres_path: str = None, in_refpos_path: str = None, in_qmmm_path: str = None, in_gromosXX_bin_dir: str = None, in_gromosPP_bin_dir: str = None)
 |          The Gromos_System class is the central unit of PyGromosTools for files and states.
 |          With this class all files can be read-in or the files can be automatically generated from smiles.
 |          Additionally to that can all gromos++ functions be used from the Gromos System, so system generation can be easily accomplished.
 |
 |          if you want to remove all binary checks, do the following:
 |          &gt;&gt;&gt; from pygromos.files.gromos_system.gromos_system import Gromos_System
 |          &gt;&gt;&gt; Gromos_System._gromos_noBinary_checks = True
 |          &gt;&gt;&gt; sys = Gromos_System()
 |
 |      Parameters
 |      ----------
 |      work_folder : str
 |          This gives the initial working folder for the system.
 |      system_name : str
 |          the name of the system, also used as file prefix
 |      in_smiles : str, optional
 |          Molecule input SMILES for file generation, by default None
 |      in_top_path : str, optional
 |          input Gromos topology path (.top), by default None
 |      in_cnf_path : str, optional
 |          input Gromos coordinate path (.cnf), by default None
 |      in_imd_path : str, optional
 |          input Gromos simulation parameters path (.imd), by default None
 |      in_disres_path : str, optional
 |          input Gromos distance restraint path (.disres), by default None
 |      in_ptp_path : str, optional
 |          input pertubation file for free energy calculations (.ptp), by default None
 |      in_posres_path : str, optional
 |          input position restraints file (.por), by default None
 |      in_qmmm_path: str, optional
 |          qmmm parameter file (.qmmm), by default None
 |      in_refpos_path : str, optional
 |          input reference position file (.rpf), by default None
 |      in_gromosXX_bin_dir : str, optional
 |          path to the binary dir of GromosXX, by default None -&gt; uses the set binaries in the PATH variable
 |      in_gromosPP_bin_dir : str, optional
 |          path to the binary dir of GromosPP, by default None -&gt; uses the set binaries in the PATH variable
 |      rdkitMol : Chem.rdchem.Mol, optional
 |          input rdkit Molecule, by default None
 |      in_mol2_file : str, optional
 |          path to input mol2 file, by default None
 |      readIn : bool, optional
 |          readIn all provided files?, by default True
 |      Forcefield : forcefield_system, optional
 |          input PyGromos - forcefield Class , by default forcefield_system()
 |      auto_convert : bool, optional
 |          automatically convert rdkit MOL and smiles to gromos files, by default False
 |      adapt_imd_automatically : bool, optional
 |          adjust the input imd file to the GromosSystem, by default True
 |      verbose : bool, optional
 |          Stay a while and listen!, by default False
 |
 |      Raises
 |      ------
 |      Warning
 |          Rises warning if files are not present.
 |
 |  __repr__(self) -&gt; str
 |      Return repr(self).
 |
 |  __setstate__(self, state)
 |
 |  __str__(self) -&gt; str
 |      Return str(self).
 |
 |  adapt_imd(self, not_ligand_residues: List[str] = [])
 |
 |  auto_convert(self)
 |
 |  copy(self)
 |
 |  generate_posres(self, residues: List = [&lt;class &#39;int&#39;&gt;], keep_residues: bool = True, verbose: bool = False)
 |
 |  get_file_paths(self) -&gt; Dict[str, str]
 |          get the paths of the files in a dict.
 |      Returns
 |      -------
 |      Dict[str, str]
 |          returns alle file paths, with attribute file name as key.
 |
 |  get_script_generation_command(self, var_name: str = None, var_prefixes: str = &#39;&#39;) -&gt; str
 |
 |  parse_attribute_files(self, file_mapping: Dict[str, str], readIn: bool = True, verbose: bool = False)
 |          This function sets dynamically builds the output folder, the file objs of this class and checks their dependencies.
 |
 |      Parameters
 |      ----------
 |      file_mapping: Dict[str, Union[str, None]]
 |          attribute name: input path
 |
 |      Returns
 |      -------
 |
 |  prepare_for_simulation(self, not_ligand_residues: List[str] = [])
 |
 |  rdkit2Gromos(self)
 |
 |  rdkit2GromosName(self) -&gt; str
 |
 |  rdkitImport(self, inputMol: rdkit.Chem.rdchem.Mol)
 |
 |  read_files(self, verbose: bool = False)
 |
 |  rebase_files(self)
 |
 |  save(self, path: Union[str, _io.FileIO] = None, safe: bool = True) -&gt; str
 |      This method stores the Class as binary obj to a given path or fileBuffer.
 |
 |  work_folder_no_update(self, work_folder: str)
 |      # Updates the work folder without updating all file paths
 |
 |  write_files(self, cnf: bool = False, imd: bool = False, top: bool = False, ptp: bool = False, disres: bool = False, posres: bool = False, refpos: bool = False, qmmm: bool = False, mol: bool = False, all: bool = True, verbose: bool = False)
 |
 |  ----------------------------------------------------------------------
 |  Class methods defined here:
 |
 |  load(path: Union[str, _io.FileIO] = None) -&gt; object from builtins.type
 |      This method stores the Class as binary obj to a given path or fileBuffer.
 |
 |  ----------------------------------------------------------------------
 |  Readonly properties defined here:
 |
 |  all_file_paths
 |      Returns
 |      -------
 |
 |  all_files
 |      Returns
 |      -------
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors defined here:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  cnf
 |
 |  disres
 |
 |  gromosPP
 |
 |  gromosPP_bin_dir
 |
 |  gromosXX
 |
 |  gromosXX_bin_dir
 |
 |  imd
 |
 |  name
 |
 |  posres
 |
 |  ptp
 |
 |  qmmm
 |
 |  refpos
 |
 |  top
 |
 |  trc
 |
 |  tre
 |
 |  work_folder
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;_checkpoint_path&#39;: &lt;class &#39;str&#39;&gt;, &#39;_future_promise...
 |
 |  optional_files = {&#39;disres&#39;: &lt;class &#39;pygromos.files.topology.disres.Dis...
 |
 |  required_files = {&#39;cnf&#39;: &lt;class &#39;pygromos.files.coord.cnf.Cnf&#39;&gt;, &#39;imd&#39;...
 |
 |  traj_files = {&#39;trc&#39;: &lt;class &#39;pygromos.files.trajectory.trc.Trc&#39;&gt;, &#39;tre...

</pre></div></div>
</div>
<section id="The-Topology-File">
<h3>The Topology File<a class="headerlink" href="#The-Topology-File" title="Permalink to this headline">¶</a></h3>
<p>The first task is to generate a molecular topology file containing the topological and force-field data concerning the molecular system under consideration. Specifying a complete molecular topology for a large molecule, however, is a tedious task. Therefore, in GROMOS a molecular topology is generated from molecular topology building blocks which carry the topological information of molecules like amino acid residues, nucleotides, etc., see Vol. 3. The molecular topology building blocks can be
linked together into a complete molecular topology file using the GROMOS++ program <em>make_top</em>.</p>
<p>Many molecular topology building blocks are available in the molecular topology building block files (<em>*.mtb</em>), which are standard data files that come together with the GROMOS package.</p>
<p>In case the needed molecular topology building blocks are not part of the standard distribution, they must be constructed. Constructing a new topology building block may require estimation of additional force-field parameters, which have to be added to the interaction function parameter file (<em>*.ifp</em>).</p>
<p>When generating a molecular topology for the system of interest one should also address the protonation state of the molecular groups according to the pH and of the solvent and counter ions that need to be included in the simulation box. In case of a molecular complex, e.g. a DNA-protein complex, the two separately generated molecular topologies for the protein and the DNA can be merged using the GROMOS++ program com top.</p>
<section id="Building-the-Topology-File">
<h4>Building the Topology File<a class="headerlink" href="#Building-the-Topology-File" title="Permalink to this headline">¶</a></h4>
<p>In this section you should build a molecular topology of a linear charged penta-peptide with water as a solvent, including two Cl− counter ions.</p>
<p><strong>programs: make_top, check_top</strong></p>
<section id="build-topology-for-single-molecule">
<h5>build topology for single molecule<a class="headerlink" href="#build-topology-for-single-molecule" title="Permalink to this headline">¶</a></h5>
<p>You will build the molecular topology file of the linear charged penta-peptide Val-Tyr-Arg-Lys-Gln by using the GROMOS++ program <em>make_top</em>. As input following parameters will be provided: * <strong>in_building_block_lib_path</strong>: the molecular topology building block file is specified. In the code below, you will see, that we gather the forcefield information from the <em>Gromos54A7</em> force field, which is directly provided by PyGromosTools * <strong>in_parameter_lib_path</strong>: specifies the interaction function
parameter file. Also here we take the parameters directly from the <em>Gromos54A7</em> forcefield from PyGromosTools. * <strong>in_solvent</strong>: The string name of the desired solvent. * <strong>in_sequence</strong>: the sequence of the building blocks for the amino acid residues, including the amino and carboxy terminus is specified (<em>NH3+ VAL TYR ARG LYSH GLN COO-</em>). (Notice that both termini are charged)</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.data.ff</span> <span class="kn">import</span> <span class="n">Gromos54A7</span> <span class="c1">#Get standard information of the GROMOS54A7 force field.</span>

<span class="c1">#Generate the topology</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">make_top</span><span class="p">(</span><span class="n">in_sequence</span><span class="o">=</span><span class="s2">&quot;NH3+ VAL TYR ARG LYSH GLN COO-&quot;</span><span class="p">,</span>
                     <span class="n">in_solvent</span><span class="o">=</span><span class="s2">&quot;H2O&quot;</span><span class="p">,</span>
                     <span class="n">in_building_block_lib_path</span> <span class="o">=</span> <span class="n">Gromos54A7</span><span class="o">.</span><span class="n">mtb</span><span class="p">,</span>
                     <span class="n">in_parameter_lib_path</span><span class="o">=</span> <span class="n">Gromos54A7</span><span class="o">.</span><span class="n">ifp</span><span class="p">)</span>

<span class="c1">#Here the residues in the topology file will be printed out :)  - this corresponds to the RESNAME Block in the topology file:</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">top</span><span class="o">.</span><span class="n">RESNAME</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
RESNAME
5
VAL
TYR
ARG
LYSH
GLN
END
</pre></div></div>
</div>
</section>
</section>
</section>
<section id="The-Coordinate-File">
<h3>The Coordinate File<a class="headerlink" href="#The-Coordinate-File" title="Permalink to this headline">¶</a></h3>
<p>Coordinates for biomolecules are often available from X-ray or NMR experiments and can be obtained in <em>Protein Data Bank</em> (<em>PDB</em>) format, which can be converted to <em>GROMOS</em> format using the GROMOS++ program <em>pdb2g96</em>. However, the conversion is not always straightforward since the naming and numbering of the atoms in the <em>PDB</em> format usually do not match the <em>GROMOS</em> format.</p>
<p>Moreover, the coordinates for hydrogen atoms are not present in the <em>PDB</em> files (when the structure was determined using X-ray diffraction data) and have to be generated using the GROMOS++ program <em>gch</em>.</p>
<p>When the structure is determined using NMR data, the PDB structure often contains more hydrogen atoms than are needed for GROMOS, as in the GROMOS force field only polar and aromatic hydrogens are explicitly represented. Aliphatic hydrogens are non-existing due to the use of so-called united atoms. The aliphatic hydrogen and carbon atoms are merged to form united atoms which have their own parameters. If no atomic coordinates for the solute are available from experimental data, the coordinates
have to be generated using molecular modeling software. Often parts of the structure (e.g. flexible loops) are not resolved in the experiment and therefore not available in the PDB and have to be modeled as well.</p>
<section id="Periodic-Boxes">
<h4>Periodic Boxes<a class="headerlink" href="#Periodic-Boxes" title="Permalink to this headline">¶</a></h4>
<p>When a simulation of a solute in solution is to be carried out, a (periodic) box (be it rectangular, triclinic or truncated octahedral) is put around the solute and filled with solvent molecules up to the required density. The solvent coordinates can e.g. be generated using the GROMOS++ program sim box. The generated box should be sufficiently large to allow the use of a reasonable non-bonded interaction cut-off radius. Putting the solute in a box of solvent using the <em>sim_box</em> program will
result in several high-energy atom-atom contacts at the solute-solvent interface and at the box edges. In order to relax the generated configuration the solvent configuration should be energy minimized while positionally restraining the solute. Counter-ion atomic coordinates can then be generated using the GROMOS++ program <em>ion</em>, which can replace a number of solvent molecules by ions.</p>
<p><strong>programs: pdb2g96, gch</strong></p>
</section>
<section id="Building-the-Coordinate-File">
<h4>Building the Coordinate File<a class="headerlink" href="#Building-the-Coordinate-File" title="Permalink to this headline">¶</a></h4>
<p>The initial coordinate <em>.pdb</em> file is already provided in the <em>input</em> directory as <em>peptide.pdb</em>. Open the file <em>peptide.pdb</em> and check if the atom names match the names in the molecular topology object <strong>SOLUTEATOM</strong> block. In the pdb file <em>peptide.pdb</em> the coordinates for hydrogen atoms are not given and have to be generated. If the atom names are correct, you can continue in the second next cell by converting the <em>PDB</em> file <em>peptide.pdb</em> into the GROMOS format using the GROMOS++ program
<em>pdb2g96</em>. The hydrogen atoms will be added to the coordinate file according to the topological requirements.</p>
<hr class="docutils" />
<p><strong>Warning</strong>: When converting coordinate files from the Protein Data Bank to GROMOS format many difficulties may emerge. If you encounter problems using the pdb2g96 program, have a look at Sec. 4-7.3. There you can find further documentation on the advanced usage of this program. Especially the use of a library that matches residue and atom names might be useful in many cases. pdb2g96.lib which you can find in the directory is an example of the PDB library file. ***</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Compare the atom names of the pdb file input/peptide.pdb with the printed ones here:</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">top</span><span class="o">.</span><span class="n">SOLUTEATOM</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
SOLUTEATOM
#   NRP: number of solute atoms
         71
#  ATNM: atom number
#  MRES: residue number
#  PANM: atom name of solute atom
#   IAC: integer (van der Waals) atom type code
#  MASS: mass of solute atom
#    CG: charge of solute atom
#   CGC: charge group code (0 or 1)
#   INE: number of excluded atoms
# INE14: number of 1-4 interactions
# ATNM MRES PANM IAC     MASS       CG  CGC INE
#                                           INE14
        1       1       H1      21      1.008   0.248   0       4       2       3       4       5
                                                                                2       6       9
        2       1       H2      21      1.008   0.248   0       3       3       4       5
                                                                                2       6       9
        3       1       N       8       14.0067 0.129   0       4       4       5       6       9
                                                                                4       7       8       10      11
        4       1       H3      21      1.008   0.248   0       1       5
                                                                                2       6       9
        5       1       CA      14      13.019  0.127   1       6       6       7       8       9       10      11
                                                                                2       12      13
        6       1       CB      14      13.019  0.0     0       3       7       8       9
                                                                                2       10      11
        7       1       CG1     16      15.035  0.0     0       1       8
                                                                                1       9
        8       1       CG2     16      15.035  0.0     1       0
                                                                                1       9
        9       1       C       12      12.011  0.45    0       4       10      11      12      13
                                                                                2       14      27
        10      1       O       1       15.9994 -0.45   1       1       11
                                                                                2       12      13
        11      2       N       6       14.0067 -0.31   0       4       12      13      14      27
                                                                                3       15      28      29
        12      2       H       21      1.008   0.31    1       1       13
                                                                                2       14      27
        13      2       CA      14      13.019  0.0     0       5       14      15      27      28      29
                                                                                4       16      18      30      31
        14      2       CB      15      14.027  0.0     0       8       15      16      17      18      19      20
                                                                                        22      27
                                                                                2       28      29
        15      2       CG      12      12.011  0.0     1       9       16      17      18      19      20      21
                                                                                        22      23      24
                                                                                1       27
        16      2       CD1     12      12.011  -0.14   0       8       17      18      19      20      21      22
                                                                                        24      25
                                                                                0
        17      2       HD1     20      1.008   0.14    1       4       18      20      21      24
                                                                                0
        18      2       CD2     12      12.011  -0.14   0       6       19      20      22      23      24      25
                                                                                0
        19      2       HD2     20      1.008   0.14    1       3       22      23      24
                                                                                0
        20      2       CE1     12      12.011  -0.14   0       5       21      22      23      24      25
                                                                                1       26
        21      2       HE1     20      1.008   0.14    1       3       22      24      25
                                                                                0
        22      2       CE2     12      12.011  -0.14   0       3       23      24      25
                                                                                1       26
        23      2       HE2     20      1.008   0.14    1       2       24      25
                                                                                0
        24      2       CZ      12      12.011  0.203   0       2       25      26
                                                                                0
        25      2       OH      3       15.9994 -0.611  0       1       26
                                                                                0
        26      2       HH      21      1.008   0.408   1       0
                                                                                0
        27      2       C       12      12.011  0.45    0       4       28      29      30      31
                                                                                2       32      44
        28      2       O       1       15.9994 -0.45   1       1       29
                                                                                2       30      31
        29      3       N       6       14.0067 -0.31   0       4       30      31      32      44
                                                                                3       33      45      46
        30      3       H       21      1.008   0.31    1       1       31
                                                                                2       32      44
        31      3       CA      14      13.019  0.0     0       5       32      33      44      45      46
                                                                                3       34      47      48
        32      3       CB      15      14.027  0.0     0       3       33      34      44
                                                                                3       35      45      46
        33      3       CG      15      14.027  0.0     1       2       34      35
                                                                                3       36      37      44
        34      3       CD      15      14.027  0.09    0       3       35      36      37
                                                                                2       38      41
        35      3       NE      11      14.0067 -0.11   0       4       36      37      38      41
                                                                                4       39      40      42      43
        36      3       HE      21      1.008   0.24    0       1       37
                                                                                2       38      41
        37      3       CZ      12      12.011  0.34    0       6       38      39      40      41      42      43
                                                                                0
        38      3       NH1     10      14.0067 -0.26   0       3       39      40      41
                                                                                2       42      43
        39      3       HH11    21      1.008   0.24    0       1       40
                                                                                1       41
        40      3       HH12    21      1.008   0.24    0       0
                                                                                1       41
        41      3       NH2     10      14.0067 -0.26   0       2       42      43
                                                                                0
        42      3       HH21    21      1.008   0.24    0       1       43
                                                                                0
        43      3       HH22    21      1.008   0.24    1       0
                                                                                0
        44      3       C       12      12.011  0.45    0       4       45      46      47      48
                                                                                2       49      57
        45      3       O       1       15.9994 -0.45   1       1       46
                                                                                2       47      48
        46      4       N       6       14.0067 -0.31   0       4       47      48      49      57
                                                                                3       50      58      59
        47      4       H       21      1.008   0.31    1       1       48
                                                                                2       49      57
        48      4       CA      14      13.019  0.0     0       5       49      50      57      58      59
                                                                                3       51      60      61
        49      4       CB      15      14.027  0.0     1       3       50      51      57
                                                                                3       52      58      59
        50      4       CG      15      14.027  0.0     0       2       51      52
                                                                                2       53      57
        51      4       CD      15      14.027  0.0     1       2       52      53
                                                                                3       54      55      56
        52      4       CE      15      14.027  0.127   0       4       53      54      55      56
                                                                                0
        53      4       NZ      8       14.0067 0.129   0       3       54      55      56
                                                                                0
        54      4       HZ1     21      1.008   0.248   0       2       55      56
                                                                                0
        55      4       HZ2     21      1.008   0.248   0       1       56
                                                                                0
        56      4       HZ3     21      1.008   0.248   1       0
                                                                                0
        57      4       C       12      12.011  0.45    0       4       58      59      60      61
                                                                                2       62      69
        58      4       O       1       15.9994 -0.45   1       1       59
                                                                                2       60      61
        59      5       N       6       14.0067 -0.31   0       4       60      61      62      69
                                                                                3       63      70      71
        60      5       H       21      1.008   0.31    1       1       61
                                                                                2       62      69
        61      5       CA      14      13.019  0.0     0       5       62      63      69      70      71
                                                                                1       64
        62      5       CB      15      14.027  0.0     0       3       63      64      69
                                                                                4       65      66      70      71
        63      5       CG      15      14.027  0.0     1       3       64      65      66
                                                                                3       67      68      69
        64      5       CD      12      12.011  0.29    0       4       65      66      67      68
                                                                                0
        65      5       OE1     1       15.9994 -0.45   0       1       66
                                                                                2       67      68
        66      5       NE2     7       14.0067 -0.72   0       2       67      68
                                                                                0
        67      5       HE21    21      1.008   0.44    0       1       68
                                                                                0
        68      5       HE22    21      1.008   0.44    1       0
                                                                                0
        69      5       C       12      12.011  0.27    0       2       70      71
                                                                                0
        70      5       O1      2       15.9994 -0.635  0       1       71
                                                                                0
        71      5       O2      2       15.9994 -0.635  1       0
                                                                                0
END
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Generate coordinate file:</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">pdb2gromos</span><span class="p">(</span><span class="n">in_pdb_path</span><span class="o">=</span><span class="n">input_dir</span><span class="o">+</span><span class="s2">&quot;/peptide.pdb&quot;</span><span class="p">)</span>

<span class="c1">#show the coordinates that were generated.</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"version_major": 2, "version_minor": 0, "model_id": "ddfa8573606e4e0e9b14d29ea3130984"}</script></div>
</div>
<section id="Initial-Write-out">
<h5>Initial Write out<a class="headerlink" href="#Initial-Write-out" title="Permalink to this headline">¶</a></h5>
<p>Next we will write out all generated files, such we have the initial products of our efforts.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Now write all files, such that you can check them directly.</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Path before rebase: &quot;</span><span class="o">+</span><span class="nb">str</span><span class="p">(</span><span class="n">build_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">path</span><span class="p">))</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">rebase_files</span><span class="p">()</span>

<span class="c1">#check this</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Path after rebase: &quot;</span><span class="o">+</span><span class="n">build_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">path</span><span class="p">)</span>

<span class="c1">#Check also how the system path and attributes were automatically updated.</span>
<span class="n">build_system</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Path before rebase: None
File imd is empty , can not be written!
Path after rebase: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/a_build_initial_files/peptide.cnf
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

GROMOS SYSTEM: peptide
################################################################################
WORKDIR: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/a_build_initial_files
LAST CHECKPOINT: None

GromosXX_bin: None
GromosPP_bin: None
FILES:
        imd: None
        top: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/a_build_initial_files/peptide.top
        cnf: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/a_build_initial_files/peptide.cnf
FUTURE PROMISE: False
SYSTEM:
        PROTEIN:        protein  nresidues: 5 natoms: 71
        LIGANDS:        []  resID: []  natoms: 0
        Non-LIGANDS:    []  nmolecules: 0  natoms: 0
        SOLVENT:        []  nmolecules: 0  natoms: 0


</pre></div></div>
</div>
</section>
</section>
<section id="add-hydrogens">
<h4>add hydrogens<a class="headerlink" href="#add-hydrogens" title="Permalink to this headline">¶</a></h4>
<p>Have a look at the cnf coordinates in the GromosSystem (next cell). You will notice that the hydrogen atoms have been added to the coordinate file with the Cartesian coordinates being set to zero.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#print x lines of the Position Block of valine</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="nb">str</span><span class="p">(</span><span class="n">atomP</span><span class="p">)</span> <span class="k">for</span> <span class="n">atomP</span> <span class="ow">in</span> <span class="n">build_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">POSITION</span> <span class="k">if</span><span class="p">(</span><span class="s2">&quot;VAL&quot;</span> <span class="o">==</span> <span class="n">atomP</span><span class="o">.</span><span class="n">resName</span><span class="p">)]))</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
    1 VAL   H1         1    0.000000000    0.000000000    0.000000000
    1 VAL   H2         2    0.000000000    0.000000000    0.000000000
    1 VAL   N          3    1.196200000    1.413300000    1.529800000
    1 VAL   H3         4    0.000000000    0.000000000    0.000000000
    1 VAL   CA         5    1.237300000    1.322400000    1.421900000
    1 VAL   CB         6    1.116100000    1.268700000    1.345400000
    1 VAL   CG1        7    1.041700000    1.369500000    1.257600000
    1 VAL   CG2        8    1.157800000    1.148100000    1.261100000
    1 VAL   C          9    1.335700000    1.393300000    1.328600000
    1 VAL   O         10    1.447800000    1.344200000    1.316600000

</pre></div></div>
</div>
<p>In order to generate meaningful coordinates for the hydrogen atoms run the GROMOS++ program <em>protonate</em> (or gch). It will generate the coordinates for hydrogen atoms by geometric means using the information from the molecular topology file. The argument <em>tolerance</em> sets the tolerance that is used for keeping the coordinates of hydrogens that are already present in the coordinate file.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Add the hydrogen positions</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">add_hydrogens</span><span class="p">()</span>

<span class="c1">#store the current files with a different name:</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;peptideH&quot;</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">rebase_files</span><span class="p">()</span>

<span class="c1">#visualize again the nice structure</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
File imd is empty , can not be written!
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"version_major": 2, "version_minor": 0, "model_id": "2eb3167caf424bfa8a829e101be1a6e3"}</script></div>
</div>
</section>
<section id="Optional:-Convert-cnf-to-pdb">
<h4>Optional: Convert cnf to pdb<a class="headerlink" href="#Optional:-Convert-cnf-to-pdb" title="Permalink to this headline">¶</a></h4>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pdb_path</span> <span class="o">=</span> <span class="n">build_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">write_pdb</span><span class="p">(</span><span class="n">build_system</span><span class="o">.</span><span class="n">work_folder</span><span class="o">+</span><span class="s2">&quot;/vacuum_hpeptide.pdb&quot;</span><span class="p">)</span>
<span class="n">pdb_path</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&#39;/home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/a_build_initial_files/vacuum_hpeptide.pdb&#39;
</pre></div></div>
</div>
</section>
</section>
<section id="Energy-Minimization---Vacuum">
<h3>Energy Minimization - Vacuum<a class="headerlink" href="#Energy-Minimization---Vacuum" title="Permalink to this headline">¶</a></h3>
<p>Before putting the penta-peptide into a box of solvent, its configuration can be relaxed by energy minimisation.</p>
<section id="Simulation-Parameter-File">
<h4>Simulation Parameter File<a class="headerlink" href="#Simulation-Parameter-File" title="Permalink to this headline">¶</a></h4>
<p>The GROMOS simulation parameter file (also called imd) template_emin_vac can be parsed like follows and contains the following blocks:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.data.simulation_parameters_templates</span> <span class="kn">import</span>  <span class="n">template_emin_vac</span>

<span class="c1">#load simulation parameter file (Imd) File</span>
<span class="n">build_system</span><span class="o">.</span><span class="n">imd</span> <span class="o">=</span> <span class="n">template_emin_vac</span>

<span class="c1">#for nicer code we will store the simulation parameter file in a variable.</span>
<span class="n">emin_vac_imd_file</span> <span class="o">=</span> <span class="n">build_system</span><span class="o">.</span><span class="n">imd</span>
<br/></pre></div>
</div>
</div>
<section id="TITLE-Block">
<h5>TITLE Block<a class="headerlink" href="#TITLE-Block" title="Permalink to this headline">¶</a></h5>
<p>In the TITLE block you specify what is done with following input file, so you know what you did with it later and can easily reuse it.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">TITLE</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
TITLE
        steepest descent energy minimization of the peptide in vacuum

        &gt;&gt;&gt; Generated with PyGromosTools (riniker group) &lt;&lt;&lt;
END
</pre></div></div>
</div>
</section>
<section id="ENERGYMIN-Block">
<h5>ENERGYMIN Block<a class="headerlink" href="#ENERGYMIN-Block" title="Permalink to this headline">¶</a></h5>
<p>The existence of the ENERGYMIN block means that Gromos will perform an energy minimisation (EM) run. The NTEM switch indicates which minimisation algorithm to be used. With NTEM = 1 we indicate that the steepest-descent algorithm (Sec. 2-11.2) is used. NCYC gives the number of steps before resetting of conjugate-gradient search direction in case we would use the conjugate gradient method (NTEM = 2). Using DELE the energy threshold (the difference in energy between two energy minimisation steps)
for stopping the minimisation process (convergence) is specified. The initial step size and maximum step size is given in DX0 and DXM, respectively. Using FLIM the absolute value of the forces can be limited to a maximum value before the algorithm is applied (see also 4-93).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">ENERGYMIN</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on ENERGYMIN in module pygromos.files.blocks.imd_blocks object:

class ENERGYMIN(_generic_imd_block)
 |  ENERGYMIN(NTEM: int = 0, NCYC: int = 0, DELE: float = 0, DX0: float = 0, DXM: float = 0, NMIN: int = 0, FLIM: float = 0, content: List[str] = None)
 |
 |  ENERGYMIN block
 |  This block takes care of managing the Energyminimization controls.
 |
 |  Attributes
 |  ----------
 |  NTEM:   int
 |      controls energy minimisation mode.
 |      0: do not do energy minimisation (default)
 |      1: steepest-descent minimisation
 |      2: Fletcher-Reeves conjugate-gradient minimisation
 |      3: Polak-Ribiere conjugate-gradient minimisation
 |  NCYC:   int
 |      &gt;0 number of steps before resetting the conjugate-gradient search direction
 |      =0 reset only if the energy grows in the search direction
 |  DELE:   float
 |      &gt;0.0 energy threshold for convergence
 |      &gt;0.0 (conjugate-gradient) RMS force threshold for convergence
 |  DX0:    float
 |      &gt;0.0 initial step size
 |  DXM:    float
 |      &gt;0.0 maximum step size
 |  NMIN:   float
 |      &gt;0 minimum number of minimisation steps
 |  FLIM:   float
 |      &gt;=0.0 limit force to maximum value (FLIM &gt; 0.0 is not recommended)
 |
 |  Method resolution order:
 |      ENERGYMIN
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTEM: int = 0, NCYC: int = 0, DELE: float = 0, DX0: float = 0, DXM: float = 0, NMIN: int = 0, FLIM: float = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;DELE&#39;: &lt;class &#39;float&#39;&gt;, &#39;DX0&#39;: &lt;class &#39;float&#39;&gt;, &#39;D...
 |
 |  name = &#39;ENERGYMIN&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># The current ENERGYMIN Block</span>
<span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">ENERGYMIN</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
ENERGYMIN
#        NTEM    NCYC    DELE    DX0     DXM     NMIN    FLIM
         1       0       0.100000        0.010000        0.050000        2000    0.000000
END
</pre></div></div>
</div>
</section>
<section id="SYSTEM-Block">
<h5>SYSTEM Block<a class="headerlink" href="#SYSTEM-Block" title="Permalink to this headline">¶</a></h5>
<p>In the SYSTEM block you specify the number of solutes (NPM) and solvent (NSM) molecules. You only have one solute NPM = 1 and no solvent molecules NSM = 0 because you still did not add any solvent molecules to the configuration file and the peptide is still in vacuum. Otherwise you would have to tell MD++ how many solvent molecules you are using.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">SYSTEM</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on SYSTEM in module pygromos.files.blocks.imd_blocks object:

class SYSTEM(_generic_imd_block)
 |  SYSTEM(NPM: int = 0, NSM: int = 0, content: List[str] = None)
 |
 |  System Block
 |
 |      The system block defines the number of solute molecules and solvent molecules
 |
 |  Attributes
 |  ----------
 |  NPM:    int
 |      Number of Solute Molecules
 |  NSM:    int
 |      Number of Solvent Molecules
 |
 |  Method resolution order:
 |      SYSTEM
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NPM: int = 0, NSM: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NPM&#39;: &lt;class &#39;int&#39;&gt;, &#39;NSM&#39;: &lt;class &#39;int&#39;&gt;, &#39;name&#39;:...
 |
 |  name = &#39;SYSTEM&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">SYSTEM</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
SYSTEM
#        NPM     NSM
         1       0
END
</pre></div></div>
</div>
</section>
<section id="STEP-Block">
<h5>STEP Block<a class="headerlink" href="#STEP-Block" title="Permalink to this headline">¶</a></h5>
<p>The step block takes core of how long and with</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">STEP</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on STEP in module pygromos.files.blocks.imd_blocks object:

class STEP(_generic_imd_block)
 |  STEP(NSTLIM: int = 0, T: float = 0, DT: float = 0, content: List[str] = None)
 |
 |  Step Block
 |
 |      This Block gives the number of simulation steps,
 |
 |  Attributes
 |  -----------
 |  NSTLIM: int
 |      number of simulations Step till terminating.
 |  T:  float
 |      Starting Time
 |  DT: float
 |      time step [fs]
 |
 |  Method resolution order:
 |      STEP
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NSTLIM: int = 0, T: float = 0, DT: float = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;DT&#39;: &lt;class &#39;float&#39;&gt;, &#39;NSTLIM&#39;: &lt;class &#39;int&#39;&gt;, &#39;T&#39;...
 |
 |  name = &#39;STEP&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">STEP</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
STEP
#        NSTLIM          T       DT
         3000    0.000000        0.002000
END
</pre></div></div>
</div>
<p>In the BOUNDCOND block you specify which periodic boundary conditions (PBC) you are going to use in the EM procedure. NTB = 0 defines a vacuum simulation: PBC are not applied. To indicate the truncated octahedron (t) PBC, NTB is set to -1, for rectangular (r) PBC NTB is 1, and for the triclinic (c) PBC NTB is 2. NDFMIN defines the number of degrees of freedom subtracted from the total number of degrees of freedom for the calculation of the temperature.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">BOUNDCOND</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on BOUNDCOND in module pygromos.files.blocks.imd_blocks object:

class BOUNDCOND(_generic_imd_block)
 |  BOUNDCOND(NTB: int = 0, NDFMIN: int = 0, content: List[str] = None)
 |
 |  Boundary Condition Block
 |
 |      This block describes the boundary condition of the coordinate system.
 |
 |  Attributes
 |  ----------
 |      NTB : int, optional
 |          Boundary conditions, by default 0
 |          -1 : truncated octahedral
 |           0 : vacuum
 |           1 : rectangular
 |           2 : triclinic
 |      NDFMIN : int, optional
 |          number of degrees of freedom subtracted for temperature, by default 0
 |
 |  Method resolution order:
 |      BOUNDCOND
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTB: int = 0, NDFMIN: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NDFMIN&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTB&#39;: &lt;class &#39;int&#39;&gt;, &#39;nam...
 |
 |  name = &#39;BOUNDCOND&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">BOUNDCOND</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
BOUNDCOND
#        NTB     NDFMIN
         0       0
END
</pre></div></div>
</div>
<p>With the PRINTOUT block you can specify how often (every NTPRth step) you are printing out the energies to the output file.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">PRINTOUT</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on PRINTOUT in module pygromos.files.blocks.imd_blocks object:

class PRINTOUT(_generic_imd_block)
 |  PRINTOUT(NTPR: int = 0, NTPP: int = 0, content: List[str] = None)
 |
 |  PRINTOUT block
 |
 |      This Block manages the output frequency into the .omd/std-out file.
 |
 |  Attributes
 |  ----------
 |  NTPR: int
 |      print out energies, etc. every NTPR steps
 |  NTPP: int
 |      =1 perform dihedral angle transition monitoring
 |
 |  Method resolution order:
 |      PRINTOUT
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTPR: int = 0, NTPP: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NTPP&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTPR&#39;: &lt;class &#39;int&#39;&gt;, &#39;name...
 |
 |  name = &#39;PRINTOUT&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">PRINTOUT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PRINTOUT
#        NTPR    NTPP
         10      0
END
</pre></div></div>
</div>
<p>Bonds vibrate at high frequencies (hν ≫ kBT ). Therefore, these vibrations are of quantum-mechanical nature. So constraining the bond lengths is a better approximation than treating them as classical harmonic oscillators. Constraining all bond lengths of the solute and solvent (NTC=3) allows the use of a rather large time step of 2 fs. In this example the constraints are imposed by the SHAKE algorithm for both solute (NTCP=1) and solvent (NTCS=1) with a tolerance of 0.0001. See 4-90 for more
information.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">CONSTRAINT</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on CONSTRAINT in module pygromos.files.blocks.imd_blocks object:

class CONSTRAINT(_generic_imd_block)
 |  CONSTRAINT(NTC: int = 0, NTCP: int = 0, NTCP0: float = 0, NTCS: int = 0, NTCS0: float = 0, content: List[str] = None)
 |
 |  CONSTRAINT block
 |      This block is controlling constraining the atoms during a simulation.
 |
 |  Attributes
 |  ----------
 |  NTC:    int
 |  NTCP:   int
 |  NTCP0:  int
 |  NTCS:   int
 |  NTCS0:  int
 |
 |  Method resolution order:
 |      CONSTRAINT
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTC: int = 0, NTCP: int = 0, NTCP0: float = 0, NTCS: int = 0, NTCS0: float = 0, content: List[str] = None)
 |      Args:
 |          NTC:
 |          NTCP:
 |          NTCP0:
 |          NTCS:
 |          NTCS0:
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NTC&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTCP&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTCP0...
 |
 |  name = &#39;CONSTRAINT&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">CONSTRAINT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
CONSTRAINT
#        NTC
         3
#        NTCP    NTCP0(1)
         1       0.000100
#        NTCS    NTCS0(1)
         1       0.000100
END
</pre></div></div>
</div>
<p>In the FORCE block you tell MD++ which terms it should use for the energy and force calculation. For bond angles, improper dihedrals, torsional dihedrals and the non-bonded interactions the standard terms of the GROMOS force field are switched on (1). Because we are using bond-length constraints and the SHAKE algorithm, we have to switch off (0) the bond-stretching terms for the bonds involving hydrogen atoms and not involving hydrogen atoms.. In the last line of this block, the energy groups
are defined. In general, we define one or more energy groups for every molecule, and one comprising all the solvent molecules. The first integer is the number of energy groups we want to use (in the present case we only have one energy group). The following numbers are the atom sequence numbers of the last atom of each energy group. By defining these energy groups we 7-9 tell MD++ to sum up the energies between the atoms within these groups and calculate the inter-group energies, which can be
very useful.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">FORCE</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
FORCE
#        BONDS   ANGLES          IMPROPER        DIHEDRAL        ELECTROSTATIC   VDW
         0       1       1       1       1       1
#        NEGR    NRE
         1       71
END
</pre></div></div>
</div>
<p>In the PAIRLIST block you specify which algorithm you will use for the pairlist generation. The cut-off used in the short-range pairlist construction is given by RCUTP and for GROMOS it is usually 0.8 nm. The cut-off used in the long-range interactions is given by RCUTL and for GROMOS it is usually 1.4 nm. The pairlist is generated every 5th (NSNB) step. TYPE specifies the type of the cut-off, whether it is based on the distance between charge-groups (0) or on the distance between atoms (1).</p>
<hr class="docutils" />
<p><strong>Warning</strong>: Think very carefully about the definition of energy groups before running the simulation. Energies of energy groups can not be calculated from the trajectories in an efficient way. So, changing an energy-group definition will result in rerunning the simulation. ***</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">PAIRLIST</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on PAIRLIST in module pygromos.files.blocks.imd_blocks object:

class PAIRLIST(_generic_imd_block)
 |  PAIRLIST(ALGORITHM: int = 0, NSNB: int = 0, RCUTP: float = 0, RCUTL: float = 0, SIZE: Union[str, float] = 0, TYPE: Union[str, bool] = False, content: List[str] = None)
 |
 |  PAIRLIST Block
 |
 |          This block is controlling the pairlist control.
 |
 |  Attributes
 |  ----------
 |  ALGORITHM: int
 |      standard(0) (gromos96 like pairlist)
 |      grid(1) (md++ grid pairlist)
 |      grid_cell(2) (creates a mask)
 |  NSNB:   int
 |      frequency (number of steps) a pairlist is constructed
 |  RCUTPL: float
 |      short-range cut-off in twin-range
 |  RCUTL:  float
 |      intermediate-range cut-off in twin-range
 |  SIZE:   str, float
 |      grid cell size (or auto = 0.5 * RCUTP)
 |  TYPE:   str, bool
 |      chargegoup(0) (chargegroup based cutoff)
 |      atomic(1)     (atom based cutoff)
 |
 |  Method resolution order:
 |      PAIRLIST
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, ALGORITHM: int = 0, NSNB: int = 0, RCUTP: float = 0, RCUTL: float = 0, SIZE: Union[str, float] = 0, TYPE: Union[str, bool] = False, content: List[str] = None)
 |      Args:
 |          ALGORITHM:
 |          NSNB:
 |          RCUTP:
 |          RCUTL:
 |          SIZE:
 |          TYPE:
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;ALGORITHM&#39;: &lt;class &#39;int&#39;&gt;, &#39;NSNB&#39;: &lt;class &#39;int&#39;&gt;, ...
 |
 |  name = &#39;PAIRLIST&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">PAIRLIST</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PAIRLIST
#        ALGORITHM       NSNB    RCUTP   RCUTL   SIZE    TYPE
         0       5       0.800000        1.400000        0.4     0
END
</pre></div></div>
</div>
<p>In the NONBONDED block you specify using NLRELE which method for the evaluation of long-range electrostatic interactions is used. Since you will use the reaction-field method, the value of NLRELE should be equal to 1. The long-range electrostatic interactions are truncated beyond a certain cutoff (RCUTL in the PAIRLIST block). Beyond the reaction-field cut-off radius (RCRF) the electrostatic interactions are replaced by a static reaction field with a dielectric permittivity of EPSRF. RCRF and
RCUTL should be identical. Because we are doing the energy minimisation in vacuo EPSRF is set to 1. With NSLFEXCL equal to 1, you include the contributions of excluded atoms to the electrostatic energy. The ionic strength of the continuum is set to 0 (APPAK). All other switches are not used for the reaction-field method. See 4-98 for more information.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">NONBONDED</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on NONBONDED in module pygromos.files.blocks.imd_blocks object:

class NONBONDED(_generic_imd_block)
 |  NONBONDED(NLRELE: int = 0, APPAK: float = 0, RCRF: float = 0, EPSRF: float = 0, NSLFEXCL: bool = False, NSHAPE: int = 0, ASHAPE: float = 0, NA2CLC: int = 0, TOLA2: str = 0, EPSLS: float = 0, NKX: int = 0, NKY: int = 0, NKZ: int = 0, KCUT: float = 0, NGX: int = 0, NGY: int = 0, NGZ: int = 0, NASORD: int = 0, NFDORD: int = 0, NALIAS: int = 0, NSPORD: int = 0, NQEVAL: int = 0, FACCUR: float = 0, NRDGRD: bool = False, NWRGRD: bool = False, NLRLJ: bool = False, SLVDNS: float = 0, content: List[str] = None)
 |
 |  NONBONDED block
 |
 |      This block is controlling the Nonbonded term evaluation
 |
 |  Attributes
 |  ----------
 |  NLRELE: int
 |      1-3 method to handle electrostatic interactions
 |      -1 : reaction-field (LSERF compatibility mode)
 |      0 : no electrostatic interactions
 |      1 : reaction-field
 |      2 : Ewald method
 |      3 : P3M method
 |  APPAK: float
 |      &gt;= 0.0 reaction-field inverse Debye screening length
 |  RCRF: float
 |      &gt;= 0.0 reaction-field radius
 |  EPSRF: float
 |       = 0.0 || &gt; 1.0 reaction-field permittivity
 |  NSLFEXCL: bool
 |      contribution of excluded atoms to reaction field false=off/true=on
 |  NSHAPE: float
 |      -1..10 lattice sum charge-shaping function
 |      -1 : gaussian
 |      0..10 : polynomial
 |  ASHAPE: float
 |      &gt; 0.0 width of the lattice sum charge-shaping function
 |  NA2CLC: int
 |      0..4 controls evaluation of lattice sum A2 term
 |      0 : A2 = A2~ = 0
 |      1 : A2~ exact, A2 = A2~
 |      2 : A2 numerical, A2~ = A2
 |      3 : A2~ exact from Ewald or from mesh and atom coords, A2 numerical
 |      4 : A2~ averaged from mesh only, A2 numerical
 |  TOLA2: float
 |       &gt; 0.0 tolerance for numerical A2 evaluation
 |  EPSLS: float
 |      = 0.0 || &gt; 1.0 lattice sum permittivity (0.0 = tinfoil)
 |  NKX, NKY, NKZ: float
 |      &gt; 0 maximum absolute Ewald k-vector components
 |  KCUT: float
 |      &gt; 0.0 Ewald k-space cutoff
 |  NGX, NGY, NGZ: float
 |      &gt; 0 P3M number of grid points
 |  NASORD: int
 |      1..5 order of mesh charge assignment function
 |  NFDORD: int
 |      0..5 order of the mesh finite difference operator
 |      0 : ik - differentiation
 |      1..5 : finite differentiation
 |  NALIAS: float
 |      &gt; 0 number of mesh alias vectors considered
 |  NSPORD: float
 |      order of SPME B-spline functions (not available)
 |  NQEVAL: float
 |      &gt;= 0 controls accuracy reevaluation
 |      0 : do not reevaluate
 |      &gt; 0 : evaluate every NQEVAL steps
 |  FACCUR: float
 |      &gt; 0.0 rms force error threshold to recompute influence function
 |  NRDGRD: bool
 |      0,1 read influence function
 |      0 : calculate influence function at simulation start up
 |      1 : read influence function from file (not yet implemented)
 |  NWRGRD: bool
 |      0,1 write influence function
 |      0 : do not write
 |      1 : write at the end of the simulation (not yet implemented)
 |  NLRLJ: bool
 |      0,1 controls long-range Lennard-Jones corrections
 |      0 : no corrections
 |      1 : do corrections (not yet implemented)
 |  SLVDNS: float
 |      &gt; 0.0 average solvent density for long-range LJ correction (ignored)
 |
 |  Method resolution order:
 |      NONBONDED
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NLRELE: int = 0, APPAK: float = 0, RCRF: float = 0, EPSRF: float = 0, NSLFEXCL: bool = False, NSHAPE: int = 0, ASHAPE: float = 0, NA2CLC: int = 0, TOLA2: str = 0, EPSLS: float = 0, NKX: int = 0, NKY: int = 0, NKZ: int = 0, KCUT: float = 0, NGX: int = 0, NGY: int = 0, NGZ: int = 0, NASORD: int = 0, NFDORD: int = 0, NALIAS: int = 0, NSPORD: int = 0, NQEVAL: int = 0, FACCUR: float = 0, NRDGRD: bool = False, NWRGRD: bool = False, NLRLJ: bool = False, SLVDNS: float = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;APPAK&#39;: &lt;class &#39;float&#39;&gt;, &#39;ASHAPE&#39;: &lt;class &#39;float&#39;&gt;...
 |
 |  name = &#39;NONBONDED&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">emin_vac_imd_file</span><span class="o">.</span><span class="n">NONBONDED</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
NONBONDED
#        NLRELE
         1
#        APPAK   RCRF    EPSRF   NSLFEXCL
         0.000000        1.400000        61.000000       1
#        NSHAPE          ASHAPE          NA2CLC          TOLA2   EPSLS
         -1      1.400000        2       1e-10   0.000000
#        NKX     NKY     NKZ     KCUT
         10      10      10      100.000000
#        NGX     NGY     NGZ     NASORD          NFDORD          NALIAS          NSPORD
         32      32      32      3       2       3       4
#        NQEVAL          FACCUR          NRDGRD          NWRGRD
         100000          1.600000        0       0
#        NLRLJ   SLVDNS
         0       33.300000
END
</pre></div></div>
</div>
</section>
</section>
<section id="Perform-Emin">
<h4>Perform Emin<a class="headerlink" href="#Perform-Emin" title="Permalink to this headline">¶</a></h4>
<p>In order to run the MD++ program, a shell script needs to be prepared. Open the shell script em peptide. run and adapt the paths and the names of the files according to your system.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">emin</span>

<span class="n">out_prefix</span> <span class="o">=</span> <span class="s2">&quot;emin_vacuum&quot;</span>

<span class="c1">## Preparing emin gromos system</span>
<span class="n">in_emin_system</span> <span class="o">=</span> <span class="n">build_system</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">in_emin_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">project_dir</span>
<span class="n">in_emin_system</span><span class="o">.</span><span class="n">_gromosXX_bin_dir</span> <span class="o">=</span> <span class="n">gromosXX_bin</span>
<span class="n">in_emin_system</span><span class="o">.</span><span class="n">imd</span> <span class="o">=</span> <span class="n">emin_vac_imd_file</span> <span class="c1"># This step is not necessary, as we did this before</span>

<span class="kn">from</span> <span class="nn">pygromos.files.blocks.imd_blocks</span> <span class="kn">import</span> <span class="n">WRITETRAJ</span>
<span class="n">in_emin_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">add_block</span><span class="p">(</span><span class="n">block</span><span class="o">=</span><span class="n">WRITETRAJ</span><span class="p">(</span><span class="n">NTWE</span><span class="o">=</span><span class="mi">25</span><span class="p">,</span> <span class="n">NTWX</span><span class="o">=</span><span class="mi">25</span><span class="p">))</span>

<span class="n">in_emin_system</span><span class="o">.</span><span class="n">prepare_for_simulation</span><span class="p">()</span>

<span class="n">in_emin_system</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

GROMOS SYSTEM: peptideH
################################################################################
WORKDIR: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System
LAST CHECKPOINT: None

GromosXX_bin: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin
GromosPP_bin: None
FILES:
        imd: /home/mlehner/PyGromosTools/pygromos/data/simulation_parameters_templates/vac_emin.imd
        top: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/peptideH.top
        cnf: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/peptideH.cnf
FUTURE PROMISE: False
SYSTEM:
        PROTEIN:        protein  nresidues: 5 natoms: 71
        LIGANDS:        []  resID: []  natoms: 0
        Non-LIGANDS:    []  nmolecules: 0  natoms: 0
        SOLVENT:        []  nmolecules: 0  natoms: 0


</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[32]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_emin_system</span> <span class="o">=</span> <span class="n">emin</span><span class="p">(</span><span class="n">in_gromos_system</span><span class="o">=</span><span class="n">in_emin_system</span><span class="p">,</span>
                       <span class="n">step_name</span><span class="o">=</span><span class="s2">&quot;b_&quot;</span><span class="o">+</span><span class="n">out_prefix</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
################################################################################

b_emin_vacuum
################################################################################

Script:  /home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/simulation_scheduler.py

################################################################################
Simulation Setup:
################################################################################

steps_per_run:  3000
equis:  0
simulation runs:  1
#####################################################################################
                ERROR in Submission
#####################################################################################
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Traceback (most recent call last):
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/submission_systems/local.py&#34;, line 71, in submit_to_queue
    process = bash.execute(command=command, catch_STD=True, env=self.environment)
  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 935, in execute
    return execute_subprocess(command=command, verbose=verbose, catch_STD=catch_STD, env=env)
  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 829, in execute_subprocess
    raise ChildProcessError(msg)
ChildProcessError: SubProcess Failed due to returncode: 1
 COMMAND:
        /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/job_b_emin_vacuum_1.sh
STDOUT:
        b&#39;Could not find TMPDIR!\n&#39;
        b&#39; Switched to outdir for work\n&#39;
        b&#39;workDIR: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1\n&#39;
        b&#34;input:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#34;gromosXX._bin:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#34;gromosXX.bin:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#39;________________________________________\n&#39;
        b&#39;\n&#39;
        b&#39; start MD b_emin_vacuum_1.imd\n&#39;
        b&#39;\n&#39;
        b&#39;COMMAND:  export OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;Failed! process returned: \n&#39;
        b&#39; Err: \n&#39;
        b&#39;SubProcess Failed due to returncode: 127\n&#39;
        b&#39; COMMAND: \n&#39;
        b&#39;\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;STDOUT:\n&#39;
        b&#39;\t\n&#39;
        b&#39;STDERR:\n&#39;
        b&#34;\tb&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\n&#39;\n&#34;
        b&#39;GROMOS OUTPUT:\n&#39;
STDERR:
        b&#39;Traceback (most recent call last):\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/workers/simulation_workers/simulation_run_worker.py&#34;, line 202, in work\n&#39;
        b&#39;    omd_file_path = gromosXX.md_run(\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/gromos/_gromosClass.py&#34;, line 78, in convert_pyGromos_types\n&#39;
        b&#39;    return func(self, *nargs, **kwargs)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/gromos/gromosXX.py&#34;, line 215, in md_run\n&#39;
        b&#39;    process = bash.execute(command_text)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 935, in execute\n&#39;
        b&#39;    return execute_subprocess(command=command, verbose=verbose, catch_STD=catch_STD, env=env)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 829, in execute_subprocess\n&#39;
        b&#39;    raise ChildProcessError(msg)\n&#39;
        b&#39;ChildProcessError: SubProcess Failed due to returncode: 127\n&#39;
        b&#39; COMMAND: \n&#39;
        b&#39;\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;STDOUT:\n&#39;
        b&#39;\t\n&#39;
        b&#39;STDERR:\n&#39;
        b&#34;\tb&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\n&#39;\n&#34;
        b&#39;\n&#39;
        b&#39;Failed during simulations: \n&#39;
        b&#34;&lt;class &#39;ChildProcessError&#39;&gt;\n&#34;
        b&#39;(&#34;SubProcess Failed due to returncode: 127\\n COMMAND: \\n\\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\\n\\nSTDOUT:\\n\\t\\nSTDERR:\\n\\tb\&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\\\n\&#39;&#34;,)\n&#39;

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/simulation_scheduler.py&#34;, line 188, in do
    previous_job_ID, tmp_jobname, in_simSystem = chain_submission(
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/scheduler_functions.py&#34;, line 308, in chain_submission
    previous_job_ID = job_submission_system.submit_to_queue(sub_job)
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/submission_systems/local.py&#34;, line 84, in submit_to_queue
    print(process)
UnboundLocalError: local variable &#39;process&#39; referenced before assignment
Traceback (most recent call last):
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/submission_systems/local.py&#34;, line 71, in submit_to_queue
    process = bash.execute(command=command, catch_STD=True, env=self.environment)
  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 935, in execute
    return execute_subprocess(command=command, verbose=verbose, catch_STD=catch_STD, env=env)
  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 829, in execute_subprocess
    raise ChildProcessError(msg)
ChildProcessError: SubProcess Failed due to returncode: 1
 COMMAND:
        /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/job_b_emin_vacuum_1.sh
STDOUT:
        b&#39;Could not find TMPDIR!\n&#39;
        b&#39; Switched to outdir for work\n&#39;
        b&#39;workDIR: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1\n&#39;
        b&#34;input:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#34;gromosXX._bin:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#34;gromosXX.bin:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#39;________________________________________\n&#39;
        b&#39;\n&#39;
        b&#39; start MD b_emin_vacuum_1.imd\n&#39;
        b&#39;\n&#39;
        b&#39;COMMAND:  export OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;Failed! process returned: \n&#39;
        b&#39; Err: \n&#39;
        b&#39;SubProcess Failed due to returncode: 127\n&#39;
        b&#39; COMMAND: \n&#39;
        b&#39;\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;STDOUT:\n&#39;
        b&#39;\t\n&#39;
        b&#39;STDERR:\n&#39;
        b&#34;\tb&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\n&#39;\n&#34;
        b&#39;GROMOS OUTPUT:\n&#39;
STDERR:
        b&#39;Traceback (most recent call last):\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/workers/simulation_workers/simulation_run_worker.py&#34;, line 202, in work\n&#39;
        b&#39;    omd_file_path = gromosXX.md_run(\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/gromos/_gromosClass.py&#34;, line 78, in convert_pyGromos_types\n&#39;
        b&#39;    return func(self, *nargs, **kwargs)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/gromos/gromosXX.py&#34;, line 215, in md_run\n&#39;
        b&#39;    process = bash.execute(command_text)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 935, in execute\n&#39;
        b&#39;    return execute_subprocess(command=command, verbose=verbose, catch_STD=catch_STD, env=env)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 829, in execute_subprocess\n&#39;
        b&#39;    raise ChildProcessError(msg)\n&#39;
        b&#39;ChildProcessError: SubProcess Failed due to returncode: 127\n&#39;
        b&#39; COMMAND: \n&#39;
        b&#39;\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;STDOUT:\n&#39;
        b&#39;\t\n&#39;
        b&#39;STDERR:\n&#39;
        b&#34;\tb&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\n&#39;\n&#34;
        b&#39;\n&#39;
        b&#39;Failed during simulations: \n&#39;
        b&#34;&lt;class &#39;ChildProcessError&#39;&gt;\n&#34;
        b&#39;(&#34;SubProcess Failed due to returncode: 127\\n COMMAND: \\n\\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\\n\\nSTDOUT:\\n\\t\\nSTDERR:\\n\\tb\&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\\\n\&#39;&#34;,)\n&#39;

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/simulation_scheduler.py&#34;, line 188, in do
    previous_job_ID, tmp_jobname, in_simSystem = chain_submission(
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/scheduler_functions.py&#34;, line 308, in chain_submission
    previous_job_ID = job_submission_system.submit_to_queue(sub_job)
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/submission_systems/local.py&#34;, line 84, in submit_to_queue
    print(process)
UnboundLocalError: local variable &#39;process&#39; referenced before assignment

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/modules/general_simulation_modules.py&#34;, line 208, in simulation
    last_jobID = simulation_scheduler.do(**MD_job_vars)
  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/simulation_scheduler.py&#34;, line 234, in do
    raise IOError(&#34;ERROR in SUBMISSION!&#34;) from err
OSError: ERROR in SUBMISSION!
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">ChildProcessError</span>                         Traceback (most recent call last)
<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/hpc_queuing/submission_systems/local.py</span> in <span class="ansi-cyan-fg">submit_to_queue</span><span class="ansi-blue-fg">(self, sub_job)</span>
<span class="ansi-green-intense-fg ansi-bold">     70</span>             <span class="ansi-green-fg">try</span><span class="ansi-blue-fg">:</span>
<span class="ansi-green-fg">---&gt; 71</span><span class="ansi-red-fg">                 </span>process <span class="ansi-blue-fg">=</span> bash<span class="ansi-blue-fg">.</span>execute<span class="ansi-blue-fg">(</span>command<span class="ansi-blue-fg">=</span>command<span class="ansi-blue-fg">,</span> catch_STD<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">,</span> env<span class="ansi-blue-fg">=</span>self<span class="ansi-blue-fg">.</span>environment<span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">     72</span>                 std_out_buff <span class="ansi-blue-fg">=</span> map<span class="ansi-blue-fg">(</span>str<span class="ansi-blue-fg">,</span> process<span class="ansi-blue-fg">.</span>stdout<span class="ansi-blue-fg">.</span>readlines<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span>

<span class="ansi-green-fg">~/PyGromosTools/pygromos/utils/bash.py</span> in <span class="ansi-cyan-fg">execute</span><span class="ansi-blue-fg">(command, verbose, catch_STD, env)</span>
<span class="ansi-green-intense-fg ansi-bold">    934</span> ):
<span class="ansi-green-fg">--&gt; 935</span><span class="ansi-red-fg">     </span><span class="ansi-green-fg">return</span> execute_subprocess<span class="ansi-blue-fg">(</span>command<span class="ansi-blue-fg">=</span>command<span class="ansi-blue-fg">,</span> verbose<span class="ansi-blue-fg">=</span>verbose<span class="ansi-blue-fg">,</span> catch_STD<span class="ansi-blue-fg">=</span>catch_STD<span class="ansi-blue-fg">,</span> env<span class="ansi-blue-fg">=</span>env<span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">    936</span>

<span class="ansi-green-fg">~/PyGromosTools/pygromos/utils/bash.py</span> in <span class="ansi-cyan-fg">execute_subprocess</span><span class="ansi-blue-fg">(command, catch_STD, env, verbose)</span>
<span class="ansi-green-intense-fg ansi-bold">    828</span>         msg <span class="ansi-blue-fg">+=</span> <span class="ansi-blue-fg">&#34;NONE&#34;</span> <span class="ansi-green-fg">if</span> <span class="ansi-blue-fg">(</span>p<span class="ansi-blue-fg">.</span>stdout <span class="ansi-green-fg">is</span> <span class="ansi-green-fg">None</span><span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">else</span> <span class="ansi-blue-fg">&#34;\n\t&#34;</span><span class="ansi-blue-fg">.</span>join<span class="ansi-blue-fg">(</span>map<span class="ansi-blue-fg">(</span>str<span class="ansi-blue-fg">,</span> p<span class="ansi-blue-fg">.</span>stderr<span class="ansi-blue-fg">.</span>readlines<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-fg">--&gt; 829</span><span class="ansi-red-fg">         </span><span class="ansi-green-fg">raise</span> ChildProcessError<span class="ansi-blue-fg">(</span>msg<span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">    830</span>     <span class="ansi-green-fg">if</span> verbose<span class="ansi-blue-fg">:</span>

<span class="ansi-red-fg">ChildProcessError</span>: SubProcess Failed due to returncode: 1
 COMMAND:
        /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/job_b_emin_vacuum_1.sh
STDOUT:
        b&#39;Could not find TMPDIR!\n&#39;
        b&#39; Switched to outdir for work\n&#39;
        b&#39;workDIR: /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1\n&#39;
        b&#34;input:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#34;gromosXX._bin:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#34;gromosXX.bin:  &lt;class &#39;str&#39;&gt; /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/\n&#34;
        b&#39;________________________________________\n&#39;
        b&#39;\n&#39;
        b&#39; start MD b_emin_vacuum_1.imd\n&#39;
        b&#39;\n&#39;
        b&#39;COMMAND:  export OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;Failed! process returned: \n&#39;
        b&#39; Err: \n&#39;
        b&#39;SubProcess Failed due to returncode: 127\n&#39;
        b&#39; COMMAND: \n&#39;
        b&#39;\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;STDOUT:\n&#39;
        b&#39;\t\n&#39;
        b&#39;STDERR:\n&#39;
        b&#34;\tb&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\n&#39;\n&#34;
        b&#39;GROMOS OUTPUT:\n&#39;
STDERR:
        b&#39;Traceback (most recent call last):\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/workers/simulation_workers/simulation_run_worker.py&#34;, line 202, in work\n&#39;
        b&#39;    omd_file_path = gromosXX.md_run(\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/gromos/_gromosClass.py&#34;, line 78, in convert_pyGromos_types\n&#39;
        b&#39;    return func(self, *nargs, **kwargs)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/gromos/gromosXX.py&#34;, line 215, in md_run\n&#39;
        b&#39;    process = bash.execute(command_text)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 935, in execute\n&#39;
        b&#39;    return execute_subprocess(command=command, verbose=verbose, catch_STD=catch_STD, env=env)\n&#39;
        b&#39;  File &#34;/home/mlehner/PyGromosTools/pygromos/utils/bash.py&#34;, line 829, in execute_subprocess\n&#39;
        b&#39;    raise ChildProcessError(msg)\n&#39;
        b&#39;ChildProcessError: SubProcess Failed due to returncode: 127\n&#39;
        b&#39; COMMAND: \n&#39;
        b&#39;\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\n&#39;
        b&#39;\n&#39;
        b&#39;STDOUT:\n&#39;
        b&#39;\t\n&#39;
        b&#39;STDERR:\n&#39;
        b&#34;\tb&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\n&#39;\n&#34;
        b&#39;\n&#39;
        b&#39;Failed during simulations: \n&#39;
        b&#34;&lt;class &#39;ChildProcessError&#39;&gt;\n&#34;
        b&#39;(&#34;SubProcess Failed due to returncode: 127\\n COMMAND: \\n\\texport OMP_NUM_THREADS=1  &amp;&amp;  /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md @topo /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.top @conf /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/input/b_emin_vacuum.cnf @input /home/mlehner/PyGromosTools/examples/example_files/Tutorial_System/b_emin_vacuum/simulation/b_emin_vacuum_1/b_emin_vacuum_1.imd @fin b_emin_vacuum_1.cnf @trc b_emin_vacuum_1.trc @tre b_emin_vacuum_1.tre &gt; b_emin_vacuum_1.omd\\n\\nSTDOUT:\\n\\t\\nSTDERR:\\n\\tb\&#39;/bin/sh: 1: /home/bschroed/Documents/projects/gromosXX/gromosXX/build/bin/md: not found\\\\n\&#39;&#34;,)\n&#39;

During handling of the above exception, another exception occurred:

<span class="ansi-red-fg">UnboundLocalError</span>                         Traceback (most recent call last)
<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/simulation_scheduler.py</span> in <span class="ansi-cyan-fg">do</span><span class="ansi-blue-fg">(in_simSystem, out_dir_path, simulation_run_num, equilibration_run_num, initialize_first_run, reinitialize_every_run, analysis_script_path, submission_system, previous_job_ID, _no_double_submit_check, _work_dir, verbose, verbose_lvl)</span>
<span class="ansi-green-intense-fg ansi-bold">    187</span>         previous_job_ID <span class="ansi-blue-fg">=</span> previous_job_ID <span class="ansi-green-fg">if</span> <span class="ansi-blue-fg">(</span>eq_job_id <span class="ansi-green-fg">is</span> <span class="ansi-green-fg">None</span><span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">else</span> eq_job_id
<span class="ansi-green-fg">--&gt; 188</span><span class="ansi-red-fg">         previous_job_ID, tmp_jobname, in_simSystem = chain_submission(
</span><span class="ansi-green-intense-fg ansi-bold">    189</span>             simSystem<span class="ansi-blue-fg">=</span>in_simSystem<span class="ansi-blue-fg">,</span>

<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/scheduler_functions.py</span> in <span class="ansi-cyan-fg">chain_submission</span><span class="ansi-blue-fg">(simSystem, out_dir_path, out_prefix, chain_job_repetitions, worker_script, job_submission_system, jobname, run_analysis_script_every_x_runs, in_analysis_script_path, start_run_index, prefix_command, previous_job_ID, work_dir, initialize_first_run, reinitialize_every_run, verbose, verbose_lvl)</span>
<span class="ansi-green-intense-fg ansi-bold">    307</span>                 )
<span class="ansi-green-fg">--&gt; 308</span><span class="ansi-red-fg">                 </span>previous_job_ID <span class="ansi-blue-fg">=</span> job_submission_system<span class="ansi-blue-fg">.</span>submit_to_queue<span class="ansi-blue-fg">(</span>sub_job<span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">    309</span>                 <span class="ansi-green-fg">if</span> verbose<span class="ansi-blue-fg">:</span>

<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/hpc_queuing/submission_systems/local.py</span> in <span class="ansi-cyan-fg">submit_to_queue</span><span class="ansi-blue-fg">(self, sub_job)</span>
<span class="ansi-green-intense-fg ansi-bold">     83</span>                 <span class="ansi-green-fg">try</span><span class="ansi-blue-fg">:</span>
<span class="ansi-green-fg">---&gt; 84</span><span class="ansi-red-fg">                     </span>print<span class="ansi-blue-fg">(</span>process<span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">     85</span>                 <span class="ansi-green-fg">except</span> ChildProcessError<span class="ansi-blue-fg">:</span>

<span class="ansi-red-fg">UnboundLocalError</span>: local variable &#39;process&#39; referenced before assignment

The above exception was the direct cause of the following exception:

<span class="ansi-red-fg">OSError</span>                                   Traceback (most recent call last)
<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/modules/general_simulation_modules.py</span> in <span class="ansi-cyan-fg">simulation</span><span class="ansi-blue-fg">(in_gromos_simulation_system, override_project_dir, step_name, in_imd_path, submission_system, simulation_runs, equilibration_runs, previous_simulation_run, force_simulation, initialize_first_run, reinitialize_every_run, analysis_script, analysis_control_dict, _no_double_submit_check, _work_dir, verbose, verbose_lvl, _template_imd_path)</span>
<span class="ansi-green-intense-fg ansi-bold">    207</span>         <span class="ansi-green-fg">else</span><span class="ansi-blue-fg">:</span>
<span class="ansi-green-fg">--&gt; 208</span><span class="ansi-red-fg">             </span>last_jobID <span class="ansi-blue-fg">=</span> simulation_scheduler<span class="ansi-blue-fg">.</span>do<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">**</span>MD_job_vars<span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">    209</span>     <span class="ansi-green-fg">except</span> Exception <span class="ansi-green-fg">as</span> err<span class="ansi-blue-fg">:</span>

<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/hpc_queuing/job_scheduling/schedulers/simulation_scheduler.py</span> in <span class="ansi-cyan-fg">do</span><span class="ansi-blue-fg">(in_simSystem, out_dir_path, simulation_run_num, equilibration_run_num, initialize_first_run, reinitialize_every_run, analysis_script_path, submission_system, previous_job_ID, _no_double_submit_check, _work_dir, verbose, verbose_lvl)</span>
<span class="ansi-green-intense-fg ansi-bold">    233</span>         traceback<span class="ansi-blue-fg">.</span>print_exception<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">*</span>sys<span class="ansi-blue-fg">.</span>exc_info<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-fg">--&gt; 234</span><span class="ansi-red-fg">         </span><span class="ansi-green-fg">raise</span> IOError<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#34;ERROR in SUBMISSION!&#34;</span><span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">from</span> err
<span class="ansi-green-intense-fg ansi-bold">    235</span>

<span class="ansi-red-fg">OSError</span>: ERROR in SUBMISSION!

During handling of the above exception, another exception occurred:

<span class="ansi-red-fg">Exception</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1356800820.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> out_emin_system = emin(in_gromos_system=in_emin_system,
</span><span class="ansi-green-intense-fg ansi-bold">      2</span>                        step_name=&#34;b_&#34;+out_prefix)

<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/modules/preset_simulation_modules.py</span> in <span class="ansi-cyan-fg">emin</span><span class="ansi-blue-fg">(in_gromos_system, step_name, override_project_dir, in_imd_path, submission_system, simulation_runs, equilibration_runs, previous_simulation_run, _template_imd_path, _no_double_submit_check, initialize_first_run, analysis_script, verbose)</span>
<span class="ansi-green-intense-fg ansi-bold">     42</span>             template_emin_control_dict<span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">&#34;concat&#34;</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">&#34;cat_trg&#34;</span><span class="ansi-blue-fg">]</span> <span class="ansi-blue-fg">=</span> <span class="ansi-green-fg">False</span>
<span class="ansi-green-intense-fg ansi-bold">     43</span>
<span class="ansi-green-fg">---&gt; 44</span><span class="ansi-red-fg">     return simulation(
</span><span class="ansi-green-intense-fg ansi-bold">     45</span>         in_gromos_simulation_system<span class="ansi-blue-fg">=</span>in_gromos_system<span class="ansi-blue-fg">,</span>
<span class="ansi-green-intense-fg ansi-bold">     46</span>         override_project_dir<span class="ansi-blue-fg">=</span>override_project_dir<span class="ansi-blue-fg">,</span>

<span class="ansi-green-fg">~/PyGromosTools/pygromos/simulations/modules/general_simulation_modules.py</span> in <span class="ansi-cyan-fg">simulation</span><span class="ansi-blue-fg">(in_gromos_simulation_system, override_project_dir, step_name, in_imd_path, submission_system, simulation_runs, equilibration_runs, previous_simulation_run, force_simulation, initialize_first_run, reinitialize_every_run, analysis_script, analysis_control_dict, _no_double_submit_check, _work_dir, verbose, verbose_lvl, _template_imd_path)</span>
<span class="ansi-green-intense-fg ansi-bold">    209</span>     <span class="ansi-green-fg">except</span> Exception <span class="ansi-green-fg">as</span> err<span class="ansi-blue-fg">:</span>
<span class="ansi-green-intense-fg ansi-bold">    210</span>         traceback<span class="ansi-blue-fg">.</span>print_exception<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">*</span>sys<span class="ansi-blue-fg">.</span>exc_info<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-fg">--&gt; 211</span><span class="ansi-red-fg">         </span><span class="ansi-green-fg">raise</span> Exception<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#34;Could not submit the commands\n\t&#34;</span> <span class="ansi-blue-fg">+</span> <span class="ansi-blue-fg">&#34;\n\t&#34;</span><span class="ansi-blue-fg">.</span>join<span class="ansi-blue-fg">(</span>map<span class="ansi-blue-fg">(</span>str<span class="ansi-blue-fg">,</span> err<span class="ansi-blue-fg">.</span>args<span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">    212</span>
<span class="ansi-green-intense-fg ansi-bold">    213</span>     time<span class="ansi-blue-fg">.</span>sleep<span class="ansi-blue-fg">(</span>time_wait_s_for_filesystem<span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">Exception</span>: Could not submit the commands
        ERROR in SUBMISSION!
</pre></div></div>
</div>
</section>
<section id="Analysis">
<h4>Analysis<a class="headerlink" href="#Analysis" title="Permalink to this headline">¶</a></h4>
<p>Once the energy minimisation is finished, the file with the minimized coordinates, peptide min.cnf, and the general output file, em peptide.omd, that reports the progress of the minimisation, will be written out. Have a look at both files and check if the minimisation has finished successfully.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[33]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_emin_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3999014210.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>out_emin_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>view

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
<p>This is nice, but how do we actually get to this structure! In the next cell the development of the total Potential energy over the energyminimization steps is shown. As you can see, the Potential energy of the whole system is decreasing, which indicates a relaxatio of the peptide structure and an optimization of the different forcefield terms. Ergo finding a minimal total potential system energy as possible is desirable in energyminimizations.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[34]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_emin_system</span><span class="o">.</span><span class="n">tre</span><span class="o">.</span><span class="n">get_totals</span><span class="p">()</span><span class="o">.</span><span class="n">totene</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;steps&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;$V_</span><span class="si">{tot}</span><span class="s2">~[kJ/mol]$&quot;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3301154550.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>out_emin_system<span class="ansi-blue-fg">.</span>tre<span class="ansi-blue-fg">.</span>get_totals<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>totene<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;steps&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;$V_{tot}~[kJ/mol]$&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
<p>The effect seend in the total potential energy can also be observed of course in the peptide molecue coordinats.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[35]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">view</span> <span class="o">=</span> <span class="n">out_emin_system</span><span class="o">.</span><span class="n">trc</span><span class="o">.</span><span class="n">view</span>
<span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1080557569.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>view <span class="ansi-blue-fg">=</span> out_emin_system<span class="ansi-blue-fg">.</span>trc<span class="ansi-blue-fg">.</span>view
<span class="ansi-green-intense-fg ansi-bold">      2</span> view

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
<p>Finally we wan to store our energy minimization results as a python pickle obj.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[36]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_emin_system_path</span> <span class="o">=</span> <span class="n">out_emin_system</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="n">out_emin_system</span><span class="o">.</span><span class="n">work_folder</span><span class="o">+</span><span class="s2">&quot;/emin_vac_result.obj&quot;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3081055068.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>out_emin_system_path <span class="ansi-blue-fg">=</span> out_emin_system<span class="ansi-blue-fg">.</span>save<span class="ansi-blue-fg">(</span>out_emin_system<span class="ansi-blue-fg">.</span>work_folder<span class="ansi-blue-fg">+</span><span class="ansi-blue-fg">&#34;/emin_vac_result.obj&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[37]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#if you want to load the file:</span>
<span class="c1">#out_emin_system = Gromos_System.load(out_emin_system.work_folder+&quot;/emin_vac_result.obj&quot;)</span>
</pre></div>
</div>
</div>
</section>
</section>
</section>
<section id="Solvatistation-and-Solvent-Energy-Minimization">
<h2>Solvatistation and Solvent Energy Minimization<a class="headerlink" href="#Solvatistation-and-Solvent-Energy-Minimization" title="Permalink to this headline">¶</a></h2>
<p>Now you can put the energy minimized penta-peptide in a box of solvent to simulate an aequos environment. This can be done with the <em>sim_box</em> function of the gromosSyste, which can solvate the solute in a pre-equilibrated box of solvent molecules. As input for the function <em>sim_box</em> you have to specify the following input arguments: * <strong>periodic_boundary_condition</strong>: the resulting box shape under the argument (r - rectangular, t - truncated octahedron, c - triclinic) *
<strong>in_solvent_cnf_file_path</strong>: the coordinate file of the pre-equilibrated box of solvent molecules under the argument (we are using a template <span class="math notranslate nohighlight">\(H_2O\)</span> of the SPC model provided in pygromos.data.solvent_coordinates) * <strong>minwall</strong>: minimum solute-to-wall distance in <span class="math notranslate nohighlight">\(nm\)</span> * <strong>threshold</strong>: the minimum solute-to-solvent distance ins <span class="math notranslate nohighlight">\(nm\)</span> * <strong>rotate</strong>: If you are using a rectangular box (&#64;pbc r) it is recommended to use an additional argument. With this additional argument the
solute can be rotated (before solvating) such that the largest distance between any two solute atoms is directed along the z-axis, and the largest atom-atom distance in the xy-plane lies in the y-direction. An input file sim box peptide.arg is already prepared.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[38]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">out_emin_system</span><span class="o">.</span><span class="n">sim_box</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1955939196.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>help<span class="ansi-blue-fg">(</span>out_emin_system<span class="ansi-blue-fg">.</span>sim_box<span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[39]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">vars</span><span class="p">(</span><span class="n">out_emin_system</span><span class="o">.</span><span class="n">trc</span><span class="p">)</span><span class="o">.</span><span class="n">keys</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/838425567.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>vars<span class="ansi-blue-fg">(</span>out_emin_system<span class="ansi-blue-fg">.</span>trc<span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>keys<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
<section id="build-box-system">
<h3>build box system<a class="headerlink" href="#build-box-system" title="Permalink to this headline">¶</a></h3>
<p>To put the solvent box around the penta-peptide use following commands:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[40]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.data.solvent_coordinates</span> <span class="kn">import</span> <span class="n">spc</span>

<span class="c1">#setup a fresh gromos System:</span>
<span class="n">box_system</span> <span class="o">=</span> <span class="n">out_emin_system</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="c1">#box_system = gromos_system.Gromos_System.load(out_emin_system_path) #if you do this step after a break, you could also decide to load the serialized obj from before.</span>

<span class="n">box_system</span><span class="o">.</span><span class="n">imd</span> <span class="o">=</span> <span class="kc">None</span>
<span class="n">box_system</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;solvate_box&quot;</span>
<span class="n">box_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">bash</span><span class="o">.</span><span class="n">make_folder</span><span class="p">(</span><span class="n">project_dir</span><span class="o">+</span><span class="s2">&quot;/c_&quot;</span><span class="o">+</span><span class="n">box_system</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>


<span class="c1">## set box and solvate the system</span>
<span class="n">box_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">add_empty_box</span><span class="p">()</span>
<span class="n">box_system</span><span class="o">.</span><span class="n">sim_box</span><span class="p">(</span><span class="n">in_solvent_cnf_file_path</span><span class="o">=</span><span class="n">spc</span><span class="p">,</span>
                 <span class="n">periodic_boundary_condition</span><span class="o">=</span><span class="s2">&quot;r&quot;</span><span class="p">,</span>
                 <span class="n">minwall</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                 <span class="n">threshold</span><span class="o">=</span><span class="mf">0.23</span><span class="p">,</span>
                 <span class="n">rotate</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1">#box_system.cnf</span>
<span class="n">box_system</span><span class="o">.</span><span class="n">rebase_files</span><span class="p">()</span> <span class="c1">#write out the files (optional) - so you can check them in the folder</span>
<span class="n">box_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span> <span class="c1">#show the results</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2018787445.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> <span class="ansi-red-fg">#setup a fresh gromos System:</span>
<span class="ansi-green-fg">----&gt; 4</span><span class="ansi-red-fg"> </span>box_system <span class="ansi-blue-fg">=</span> out_emin_system<span class="ansi-blue-fg">.</span>copy<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> <span class="ansi-red-fg">#box_system = gromos_system.Gromos_System.load(out_emin_system_path) #if you do this step after a break, you could also decide to load the serialized obj from before.</span>
<span class="ansi-green-intense-fg ansi-bold">      6</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_emin_system&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Add-Ions">
<h3>Add Ions<a class="headerlink" href="#Add-Ions" title="Permalink to this headline">¶</a></h3>
<p>After solvating the system, two <span class="math notranslate nohighlight">\(Cl-\)</span> ions shall be added to the box as counter charges for the two positive peptide residues (Arginine and Lysine).</p>
<p>To add the ions, we need to know the name of the building block in Gromos54A7. We can checkout the names of the Atomtype names in the topology of our system:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[41]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">([</span> <span class="n">a</span> <span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">box_system</span><span class="o">.</span><span class="n">top</span><span class="o">.</span><span class="n">ATOMTYPENAME</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2889188333.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>print<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">[</span> a <span class="ansi-green-fg">for</span> a <span class="ansi-green-fg">in</span> box_system<span class="ansi-blue-fg">.</span>top<span class="ansi-blue-fg">.</span>ATOMTYPENAME<span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;box_system&#39; is not defined
</pre></div></div>
</div>
<p>After finding the atomtype name (hint: ‘CL-’), we next can use it to add two of these ions to our system with the ion function.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[42]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">box_system</span><span class="o">.</span><span class="n">ion</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/560721524.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>help<span class="ansi-blue-fg">(</span>box_system<span class="ansi-blue-fg">.</span>ion<span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;box_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[43]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.data.solvent_coordinates</span> <span class="kn">import</span> <span class="n">spc</span>

<span class="c1">## Build directory and setup a fresh Gromos System</span>
<span class="n">ion_system</span> <span class="o">=</span> <span class="n">box_system</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">ion_system</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;ion&quot;</span>
<span class="n">ion_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">bash</span><span class="o">.</span><span class="n">make_folder</span><span class="p">(</span><span class="n">project_dir</span><span class="o">+</span><span class="s2">&quot;/d_&quot;</span><span class="o">+</span><span class="n">ion_system</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>

<span class="c1">#Add the ions to the System</span>
<span class="n">ion_system</span><span class="o">.</span><span class="n">ion</span><span class="p">(</span><span class="n">negative</span><span class="o">=</span><span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="s2">&quot;CL-&quot;</span><span class="p">])</span>


<span class="n">ion_system</span><span class="o">.</span><span class="n">rebase_files</span><span class="p">()</span>   <span class="c1">#write out the files, so you can check them (optional)</span>
<span class="n">ion_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>  <span class="c1">#show the results.</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/54869407.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> <span class="ansi-red-fg">## Build directory and setup a fresh Gromos System</span>
<span class="ansi-green-fg">----&gt; 4</span><span class="ansi-red-fg"> </span>ion_system <span class="ansi-blue-fg">=</span> box_system<span class="ansi-blue-fg">.</span>copy<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> ion_system<span class="ansi-blue-fg">.</span>name <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">&#34;ion&#34;</span>
<span class="ansi-green-intense-fg ansi-bold">      6</span> ion_system<span class="ansi-blue-fg">.</span>work_folder <span class="ansi-blue-fg">=</span> bash<span class="ansi-blue-fg">.</span>make_folder<span class="ansi-blue-fg">(</span>project_dir<span class="ansi-blue-fg">+</span><span class="ansi-blue-fg">&#34;/d_&#34;</span><span class="ansi-blue-fg">+</span>ion_system<span class="ansi-blue-fg">.</span>name<span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;box_system&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Energy-Minimization-BOX">
<h3>Energy Minimization BOX<a class="headerlink" href="#Energy-Minimization-BOX" title="Permalink to this headline">¶</a></h3>
<p>In order to relax the unfavorable atom-atom contacts between the solute and the solvent, energy minimization of the solvent should be performed while keeping the solute positionally restrained (i.e. connecting the atom to a reference position by a spring). In order to do that two additional files, in which the positionally restrained atoms and the reference coordinates are specified, have to be generated from the coordinate file in our gromos system. Afterwards, we are going to run the second
energy minimization</p>
<section id="build-posistion-restraints">
<h4>build posistion restraints<a class="headerlink" href="#build-posistion-restraints" title="Permalink to this headline">¶</a></h4>
<p>To apply position restraints to our simulation for the peptide, we first need to generate a selection of residues, that shall not be modified by the energyminimization. This translates to the residues of the peptide. After this the function generate_posres can be used to generate two files. * <strong>position restraints - posres(.por)</strong> This file defines the selection of atoms, that shall be restrained during the simulation. * <strong>reference position - refpos (.rpf)</strong> This files defines a reference
position for the system. Gromos will restrain the selected atoms in the posres file to the reference position file. So if the atom position during the simulation deviates from the refpos file, the program will restrain the atom, such it moves towards the reference position.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[44]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><br/><span></span><span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">emin</span>
<span class="kn">from</span> <span class="nn">pygromos.data.simulation_parameters_templates</span> <span class="kn">import</span> <span class="n">template_emin</span>

<span class="c1"># Preparing emin gromos system</span>
<span class="n">in_eminBox_system</span> <span class="o">=</span> <span class="n">ion_system</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;emin_solvBox&quot;</span>
<span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">project_dir</span>

<span class="c1"># Build position restraints</span>
<span class="c1">## Build selection for residues</span>
<span class="n">restrain_res</span> <span class="o">=</span> <span class="p">[</span><span class="n">k</span> <span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">residues</span> <span class="k">if</span><span class="p">(</span><span class="ow">not</span> <span class="n">k</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;CL-&quot;</span><span class="p">,</span> <span class="s2">&quot;SOLV&quot;</span><span class="p">))]</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Selection of residues: &quot;</span><span class="p">,</span> <span class="n">restrain_res</span><span class="p">)</span>
<span class="c1">## Build the restrain files</span>
<span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">generate_posres</span><span class="p">(</span><span class="n">residues</span><span class="o">=</span><span class="n">restrain_res</span><span class="p">)</span>

<span class="c1"># Check simulation params</span>
<span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">imd</span> <span class="o">=</span> <span class="n">template_emin</span>   <span class="c1">#Here we use template simulation parameters, The blocks are the same as above in the vacuum case with slight deviations.</span>
<span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">INITIALISE</span><span class="o">.</span><span class="n">NTISHI</span> <span class="o">=</span> <span class="mi">1</span>

<span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">prepare_for_simulation</span><span class="p">()</span>

<span class="n">in_eminBox_system</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/682416482.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> <span class="ansi-red-fg"># Preparing emin gromos system</span>
<span class="ansi-green-fg">----&gt; 5</span><span class="ansi-red-fg"> </span>in_eminBox_system <span class="ansi-blue-fg">=</span> ion_system<span class="ansi-blue-fg">.</span>copy<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      6</span> in_eminBox_system<span class="ansi-blue-fg">.</span>name <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">&#34;emin_solvBox&#34;</span>
<span class="ansi-green-intense-fg ansi-bold">      7</span> in_eminBox_system<span class="ansi-blue-fg">.</span>work_folder <span class="ansi-blue-fg">=</span> project_dir

<span class="ansi-red-fg">NameError</span>: name &#39;ion_system&#39; is not defined
</pre></div></div>
</div>
<p>In the ouptut of the in_eminBox_system two new files appeared the posres and refpos file:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[45]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">posres</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3933610349.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>print<span class="ansi-blue-fg">(</span>in_eminBox_system<span class="ansi-blue-fg">.</span>posres<span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;in_eminBox_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[46]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">refpos</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1470707088.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>in_eminBox_system<span class="ansi-blue-fg">.</span>refpos

<span class="ansi-red-fg">NameError</span>: name &#39;in_eminBox_system&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="id1">
<h4>Perform Emin<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h4>
<p>Now we will perform the solvent box energy minimization with the restrained peptide.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[47]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># run emin</span>
<span class="n">out_eminBox_system</span> <span class="o">=</span> <span class="n">emin</span><span class="p">(</span><span class="n">in_gromos_system</span><span class="o">=</span><span class="n">in_eminBox_system</span><span class="p">,</span>
                          <span class="n">step_name</span><span class="o">=</span><span class="s2">&quot;e_&quot;</span><span class="o">+</span><span class="n">in_eminBox_system</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>
<br/><br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1280377867.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg"># run emin</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> out_eminBox_system = emin(in_gromos_system=in_eminBox_system,
</span><span class="ansi-green-intense-fg ansi-bold">      3</span>                           step_name=&#34;e_&#34;+in_eminBox_system.name)
<span class="ansi-green-intense-fg ansi-bold">      4</span>

<span class="ansi-red-fg">NameError</span>: name &#39;in_eminBox_system&#39; is not defined
</pre></div></div>
</div>
<p>Whoops what happened here? What you see is the effect of the periodic box and the peptide placed directly at the corners of the box. This is absolutely ok and there is nothing to worry about. We can recenter the peptide and then we see the expected same configuration like from the vacuum emin.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[48]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#show resulting coordinates</span>
<span class="n">view</span> <span class="o">=</span> <span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>
<span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3841672548.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#show resulting coordinates</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> </span>view <span class="ansi-blue-fg">=</span> out_eminBox_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>view
<span class="ansi-green-intense-fg ansi-bold">      3</span> view

<span class="ansi-red-fg">NameError</span>: name &#39;out_eminBox_system&#39; is not defined
</pre></div></div>
</div>
<p>Now lets fix that!</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[49]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#out_eminBox_system.cnf.recenter_pbc()</span>

<span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">recreate_view</span><span class="p">()</span>
<span class="n">recentered_view</span> <span class="o">=</span> <span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>
<span class="n">recentered_view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1370968844.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#out_eminBox_system.cnf.recenter_pbc()</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-fg">----&gt; 3</span><span class="ansi-red-fg"> </span>out_eminBox_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>recreate_view<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> recentered_view <span class="ansi-blue-fg">=</span> out_eminBox_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>view
<span class="ansi-green-intense-fg ansi-bold">      5</span> recentered_view

<span class="ansi-red-fg">NameError</span>: name &#39;out_eminBox_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[50]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_eminBox_system_path</span> <span class="o">=</span> <span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">work_folder</span><span class="o">+</span><span class="s2">&quot;/emin_box_result.obj&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1182860247.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>out_eminBox_system_path <span class="ansi-blue-fg">=</span> out_eminBox_system<span class="ansi-blue-fg">.</span>save<span class="ansi-blue-fg">(</span>out_eminBox_system<span class="ansi-blue-fg">.</span>work_folder<span class="ansi-blue-fg">+</span><span class="ansi-blue-fg">&#34;/emin_box_result.obj&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_eminBox_system&#39; is not defined
</pre></div></div>
</div>
</section>
</section>
</section>
<section id="Thermalisation-and-equilibration.">
<h2>Thermalisation and equilibration.<a class="headerlink" href="#Thermalisation-and-equilibration." title="Permalink to this headline">¶</a></h2>
<p>In the previous steps you have generated a topology and initial coordinates of your system. At this point, you have to generate initial velocities. In the process of thermalisation and equilibration, initial velocities are sampled from a Maxwell-Boltzmann distribution at a low temperature and the system is slowly heated up to the final production simulation temperature. The atoms of the solute are again positionally restrained and these restraints are loosened while heating up. With the help of
these restraints you make sure that the random initial velocities do not disrupt the initial conformation too much.</p>
<section id="Simulation-Paramters">
<h3>Simulation Paramters<a class="headerlink" href="#Simulation-Paramters" title="Permalink to this headline">¶</a></h3>
<p>To performe the described equilibration we require additional Simulation Parameter blocks that will be described in the following. First we will load a simulation parameter template for MD simulations.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[51]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.data.simulation_parameters_templates</span> <span class="kn">import</span> <span class="n">template_md_tut</span> <span class="k">as</span> <span class="n">template_md</span>
<span class="kn">from</span> <span class="nn">pygromos.files.simulation_parameters</span> <span class="kn">import</span> <span class="n">imd</span>


<span class="n">md_imd_file</span> <span class="o">=</span> <span class="n">imd</span><span class="o">.</span><span class="n">Imd</span><span class="p">(</span><span class="n">template_md</span><span class="p">)</span> <span class="c1">#in_eq_system.imd</span>
<br/></pre></div>
</div>
</div>
<section id="Initialise-Block">
<h4>Initialise Block<a class="headerlink" href="#Initialise-Block" title="Permalink to this headline">¶</a></h4>
<p>In the INITIALISE block the NTIVEL tells GROMOS whether it should generate the initial velocities or read them from the configuration file. NTISHK is used to restore bond length constraints (SHAKE). NTINHT and NTINHB are only used for Nose-Hoover thermo- and barostats and can be ignored in our case. Every time an atom is leaving the periodic box and entering it from the opposite site this incident is recorded in the so-called lattice shift vectors. Using NTISHI we want to make sure that these
vectors are initialised to zero. As you don’t want to use roto-translational constraints NTIRTC can be ignored. NTICOM is used for initial removal of centre of mass motion. NTISTI is used to reset the stochastic integrals used in stochastic dynamics (SD) simulations. IG is the random number generator seed and TEMPI the initial temperature used to generate the Maxwell-Boltzmann distribution for generation of initial velocities. See also 4-94 for more information.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[52]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">INITIALISE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on INITIALISE in module pygromos.files.blocks.imd_blocks object:

class INITIALISE(_generic_imd_block)
 |  INITIALISE(NTIVEL: bool = False, NTISHK: int = 0, NTINHT: bool = False, NTINHB: bool = False, NTISHI: bool = False, NTIRTC: bool = False, NTICOM: int = 0, NTISTI: bool = False, IG: int = 0, TEMPI: float = 0, content: List[str] = None)
 |
 |  INITIALISE block
 |
 |      This block controls the Initialisation of a simulation.
 |
 |  Attributes
 |  ----------
 |  NTIVEL: bool
 |      0,1 controls generation of initial velocities
 |      0: read from configuration (default)
 |      1: generate from Maxell distribution at temperature TEMPI
 |  NTISHK: int
 |      0..3 controls shaking of initial configuration
 |      0: no intial SHAKE (default)
 |      1: initial SHAKE on coordinates only
 |      2: initial SHAKE on velocities only
 |      3: initial SHAKE on coordinates and velocities
 |  NTINHT: bool
 |      0,1 controls generation of initial Nose-Hoover chain variables
 |      0: read from configuration (default)
 |      1: reset variables to zero.
 |  NTINHB: bool
 |      0,1 controls generation of initial Nose-Hoover (chain) barostat variables
 |      0: read from strartup file (if applicable) (default)
 |      1: reset variables to zero
 |  NTISHI: bool
 |      0,1 controls initial setting for lattice shift vectors
 |      0: read from configuration (default)
 |      1: reset shifts to zero.
 |  NTIRTC: bool
 |      0,1 controls initial setting of positions and orientations for roto-translational constraints
 |      0: read from configuration (default)
 |      1: reset based on initial configuraion of startup file
 |  NTICOM: int
 |      0,1,2 controls initial removal of COM motion
 |      0: no initial system COM motion removal (default)
 |      1: initial COM translation is removed
 |      2: initial COM rotation is removed
 |  NTISTI: bool
 |      0,1 controls generation of stochastic integrals
 |      0: read stochastic integrals and IG from configuration (default)
 |      1: set stochastic integrals to zero and use IG from here.
 |  IG: int
 |      random number generator seed
 |  TEMPI:  float
 |      initial temperature
 |
 |  Method resolution order:
 |      INITIALISE
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTIVEL: bool = False, NTISHK: int = 0, NTINHT: bool = False, NTINHB: bool = False, NTISHI: bool = False, NTIRTC: bool = False, NTICOM: int = 0, NTISTI: bool = False, IG: int = 0, TEMPI: float = 0, content: List[str] = None)
 |      Args:
 |          NTIVEL:
 |          NTISHK:
 |          NTINHT:
 |          NTINHB:
 |          NTISHI:
 |          NTIRTC:
 |          NTICOM:
 |          NTISTI:
 |          IG:
 |          TEMPI:
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;IG&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTICOM&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTIN...
 |
 |  name = &#39;INITIALISE&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<p>And now we will check the set options in our template:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[53]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">INITIALISE</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[53]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
INITIALISE
#        NTIVEL          NTISHK          NTINHT          NTINHB
         1       0       0       0
#        NTISHI          NTIRTC          NTICOM
         1       0       1
#        NTISTI
         0
#        IG      TEMPI
         12416   300.000000
END
</pre></div></div>
</div>
</section>
</section>
<section id="System-Block">
<h3>System Block<a class="headerlink" href="#System-Block" title="Permalink to this headline">¶</a></h3>
<p>In the SYSTEM block you would need to replace NSM with the number of solvent molecules in your system, but the <em>prepare_simulation</em> function takes care of that for you.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[54]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#The last SOLV molecule can be found out like this:</span>
<span class="nb">str</span><span class="p">(</span><span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">POSITION</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3553747723.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#The last SOLV molecule can be found out like this:</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> </span>str<span class="ansi-blue-fg">(</span>out_eminBox_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>POSITION<span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">-</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_eminBox_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[55]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#the template currently contains these values - note NSM is 1172, which is wrong!</span>
<span class="n">md_imd_file</span><span class="o">.</span><span class="n">SYSTEM</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[55]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
SYSTEM
#        NPM     NSM
         1       5
END
</pre></div></div>
</div>
<section id="id3">
<h4>STEP Block<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h4>
<p>In the STEP block you specify how many steps you want to simulate (NSTLIM), at what time your simulation starts (T) and how big the integration time step (DT) is. In this case you want to start at time 0 and you want to carry out a 20 ps simulation, because the time unit happens to be ps.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[56]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">STEP</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on STEP in module pygromos.files.blocks.imd_blocks object:

class STEP(_generic_imd_block)
 |  STEP(NSTLIM: int = 0, T: float = 0, DT: float = 0, content: List[str] = None)
 |
 |  Step Block
 |
 |      This Block gives the number of simulation steps,
 |
 |  Attributes
 |  -----------
 |  NSTLIM: int
 |      number of simulations Step till terminating.
 |  T:  float
 |      Starting Time
 |  DT: float
 |      time step [fs]
 |
 |  Method resolution order:
 |      STEP
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NSTLIM: int = 0, T: float = 0, DT: float = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;DT&#39;: &lt;class &#39;float&#39;&gt;, &#39;NSTLIM&#39;: &lt;class &#39;int&#39;&gt;, &#39;T&#39;...
 |
 |  name = &#39;STEP&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[57]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">STEP</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[57]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
STEP
#        NSTLIM          T       DT
         1000    0.000000        0.002000
END
</pre></div></div>
</div>
</section>
<section id="BOUNDCOND-Block">
<h4>BOUNDCOND Block<a class="headerlink" href="#BOUNDCOND-Block" title="Permalink to this headline">¶</a></h4>
<p>As previously described, with the BOUNDCOND block you specify which PBC you will use. With NTB=1 you specify rectangular PBC.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[58]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">BOUNDCOND</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on BOUNDCOND in module pygromos.files.blocks.imd_blocks object:

class BOUNDCOND(_generic_imd_block)
 |  BOUNDCOND(NTB: int = 0, NDFMIN: int = 0, content: List[str] = None)
 |
 |  Boundary Condition Block
 |
 |      This block describes the boundary condition of the coordinate system.
 |
 |  Attributes
 |  ----------
 |      NTB : int, optional
 |          Boundary conditions, by default 0
 |          -1 : truncated octahedral
 |           0 : vacuum
 |           1 : rectangular
 |           2 : triclinic
 |      NDFMIN : int, optional
 |          number of degrees of freedom subtracted for temperature, by default 0
 |
 |  Method resolution order:
 |      BOUNDCOND
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTB: int = 0, NDFMIN: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NDFMIN&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTB&#39;: &lt;class &#39;int&#39;&gt;, &#39;nam...
 |
 |  name = &#39;BOUNDCOND&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[59]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">BOUNDCOND</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[59]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
BOUNDCOND
#        NTB     NDFMIN
         1       3
END
</pre></div></div>
</div>
</section>
<section id="MULTIBATH-Block">
<h4>MULTIBATH Block<a class="headerlink" href="#MULTIBATH-Block" title="Permalink to this headline">¶</a></h4>
<p>In our case we want to run the simulation at constant temperature. For this purpose, we have to add the MULTIBATH input block (see 4-96). First we specify which algorithm we will use. In this case we will use the weak-coupling scheme (ALGORITHM=0). How many temperature baths we want to couple to the system is specified by NBATHS. You can specify the temperature using the TEMP0 parameter. TAUT is the coupling time used in the weak-coupling method for this bath. DOFSET specifies the number of
distiguishable sets of degrees of freedom. LAST is pointing to the last atom for the set of degrees of freedom; thus, you put the number of last atom of your system instead of LSTATM. COMBATH is the temperature bath to which we want to couple the centre of mass motion of this set of degrees of freedom IRBATH is the temperature bath to which the internal and rotational degrees of freedom of this set of degrees of freedom are coupled.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[60]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">MULTIBATH</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on MULTIBATH in module pygromos.files.blocks.imd_blocks object:

class MULTIBATH(_generic_imd_block)
 |  MULTIBATH(ALGORITHM: int = 0, NBATHS: int = 0, TEMP0: List[float] = [], TAU: List[float] = [], DOFSET: int = 0, LAST: List[int] = [], COMBATH: List[int] = [], IRBATH: List[int] = [], NUM: int = None, content: List[str] = None)
 |
 |  MULTIBATH Block
 |
 |      This block controls the Thermostat of a simulation. Multiple temperature baths can be coupled.
 |
 |  Attributes
 |  ----------
 |  ALGORITHM:  int
 |      temperature coupling algorithm
 |      weak-coupling(0)
 |      nose-hoover(1)
 |      nose-hoover-chains(2)   num
 |      (where num is the number of chains to use)
 |  NUM: int, optional
 |      Mumber of chains in Nosé Hoover chains scheme [only specify when needed]
 |  NBATHS: int
 |      Number of temperature baths
 |  TEMP0:  List[float]
 |      temperature of each bath (list len == NBATHS)
 |  TAU:    List[float]
 |          coupling of the temperaturebath to the system  (list len == NBATHS)
 |  DOFSET: int
 |      Number of set of Degrees of freedom.
 |  LAST:   List[int]   (list len == DOFSET)
 |      Last atoms of each DOFSet
 |  COMBATH:    List[int]   (list len == DOFSET)
 |      Index of the temperature baths
 |  IRBATH: List[int]   (list len == DOFSET)
 |      IRBAHT?
 |
 |  See Also
 |  --------
 |      _generic_imd_block, _generic_gromos_block
 |
 |  Method resolution order:
 |      MULTIBATH
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, ALGORITHM: int = 0, NBATHS: int = 0, TEMP0: List[float] = [], TAU: List[float] = [], DOFSET: int = 0, LAST: List[int] = [], COMBATH: List[int] = [], IRBATH: List[int] = [], NUM: int = None, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  adapt_multibath(self, last_atoms_bath: Dict[int, int], algorithm: int = None, num: int = None, T: Union[float, List[float]] = None, TAU: float = None) -&gt; None
 |      adapt_multibath
 |              This function is adding each atom set into a single multibath.
 |              #TODO implementation not correct with com_bath and irbath! Works for super simple cases though
 |
 |      Parameters
 |      ----------
 |      last_atoms_bath :   Dict[int,int]
 |          lastatom:bath
 |      algorithm : int
 |          int code for algorihtm
 |      T : float,List[float], optional
 |          temperature value
 |      TAU :   float, optional
 |          coupling var
 |
 |      Returns
 |      -------
 |      None
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;ALGORITHM&#39;: &lt;class &#39;int&#39;&gt;, &#39;COMBATH&#39;: typing.List[...
 |
 |  name = &#39;MULTIBATH&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[61]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">MULTIBATH</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[61]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
MULTIBATH
# ALGORITHM
  0
# NBATHS
  2
# TEMP0(1 ... NBATHS)    TAU(1 ... NBATHS)
  300.0          0.1
  300.0          0.1
# DOFSET
  2
# LAST(1 ... DOFSET)     COMBATH(1 ... DOFSET)   IRBATH(1 ... DOFSET)
  73     1       1
  2863   2       2
END
</pre></div></div>
</div>
</section>
</section>
<section id="COMTRANSROT-Block">
<h3>COMTRANSROT Block<a class="headerlink" href="#COMTRANSROT-Block" title="Permalink to this headline">¶</a></h3>
<p>This block is needed to remove the centre of mass motion (translation and rotation). Without this block it can happen that all the kinetic energy is converted to centre of mass translation (flying ice cube problem). With <em>NSCM</em> we specify how often the center-of-mass (COM) motion is removed. If <em>NSCM</em> is <span class="math notranslate nohighlight">\(&lt; 0\)</span> translation and rotation motion are removed every <span class="math notranslate nohighlight">\(|\)</span> <em>NSCM</em> <span class="math notranslate nohighlight">\(|\)</span> th step. If <em>NSCM</em> is <span class="math notranslate nohighlight">\(&gt; 0\)</span> only translation motion is removed every <em>NSCM</em>th step.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[62]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">COMTRANSROT</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on COMTRANSROT in module pygromos.files.blocks.imd_blocks object:

class COMTRANSROT(_generic_imd_block)
 |  COMTRANSROT(NSCM: int = 0, content: List[str] = None)
 |
 |  COMTRANSROT block
 |
 |  This block controls the center of mass translation and rotation removal. (flying ice cube problem)
 |
 |  Attributes
 |  ----------
 |
 |  NSCM : int
 |      controls system centre-of-mass (com) motion removal
 |      0: no com motion removal (default)
 |      &lt; 0: com translation and rotation are removed every abs(NSCM) steps.
 |      &gt; 0: com translation is removed every NSCM steps.
 |
 |  Method resolution order:
 |      COMTRANSROT
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NSCM: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NSCM&#39;: &lt;class &#39;int&#39;&gt;, &#39;name&#39;: &lt;class &#39;str&#39;&gt;}
 |
 |  name = &#39;COMTRANSROT&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[63]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">COMTRANSROT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[63]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
COMTRANSROT
#        NSCM
         1000
END
</pre></div></div>
</div>
</section>
<section id="COVALENTFORM-Block">
<h3>COVALENTFORM Block<a class="headerlink" href="#COVALENTFORM-Block" title="Permalink to this headline">¶</a></h3>
<p>This block is needed to define which functional form we will use for bond-stretching (NTBBH), bond-angle bending (NTBAH) and for torsional dihedral (NTBDN). We just use the default options for all functional forms.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[64]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">COVALENTFORM</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on COVALENTFORM in module pygromos.files.blocks.imd_blocks object:

class COVALENTFORM(_generic_imd_block)
 |  COVALENTFORM(NTBBH: int = 0, NTBAH: int = 0, NTBDN: int = 0, content: List[str] = None)
 |
 |  COVALENTFORM Block
 |
 |      The COVALENTFORM Block manages the functional form of the Bonded contributions to the force field.
 |      It is optional in an imd file to define the block.
 |
 |  Attributes
 |  ----------
 |  NTBBH:  int
 |      0,1 controls bond-stretching potential
 |      0: quartic form (default)
 |      1: harmonic form
 |  NTBAH:  int
 |      0,1 controls bond-angle bending potential
 |      0: cosine-harmonic (default)
 |      1: harmonic
 |  NTBDN:  int
 |      0,1 controls torsional dihedral potential
 |      0: arbitrary phase shifts (default)
 |      1: phase shifts limited to 0 and 180 degrees.
 |
 |  Method resolution order:
 |      COVALENTFORM
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTBBH: int = 0, NTBAH: int = 0, NTBDN: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NTBAH&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTBBH&#39;: &lt;class &#39;int&#39;&gt;, &#39;NT...
 |
 |  name = &#39;COVALENTFORM&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[65]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">COVALENTFORM</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[65]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
COVALENTFORM
#        NTBBH   NTBAH   NTBDN
         0       0       0
END
</pre></div></div>
</div>
</section>
<section id="WRITETRAJ-Block">
<h3>WRITETRAJ Block<a class="headerlink" href="#WRITETRAJ-Block" title="Permalink to this headline">¶</a></h3>
<p>Gromos MD simulations produce a massive amount of data and it is impossible to store all the data it produces. The WRITETRAJ block meets this demand: Here you specify how often the coordinate trajectory (<em>NTWX</em>), the velocity trajectory (<em>NTWV</em>), the force trajectory (<em>NTWF</em>), the energy trajectory (<em>NTWE</em>), the free energy trajectory (<em>NTWG</em>) and the block averaged energy trajectory (<em>NTWB</em>) are written out. In the present case, we are only interested in the coordinates (<em>NTWX</em>) and energies
(<em>NTWE</em>) and we write them every 100th step. The second switch (<em>NTWSE</em>) defines selection criterion for trajectories: if NTWSE = 0 the normal coordinate trajectory will be written, or if <em>NTWSE</em> <span class="math notranslate nohighlight">\(&gt; 0\)</span> a minimum energy trajectory will be written.</p>
<hr class="docutils" />
<p><strong>Warning</strong>: It makes no sense to write out configurations too often. First, it needs a lot of disk space. Second, the data is highly correlated and so no additional information is gained from it ***</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[66]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">WRITETRAJ</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on WRITETRAJ in module pygromos.files.blocks.imd_blocks object:

class WRITETRAJ(_generic_imd_block)
 |  WRITETRAJ(NTWX: int = 0, NTWSE: int = 0, NTWV: int = 0, NTWF: int = 0, NTWE: int = 0, NTWG: int = 0, NTWB: int = 0, content: List[str] = None)
 |
 |  WRITETRAJ
 |
 |      The WRITETRAJ Block manages the output frequency of GROMOS. Here you can determine how often a file should be written.
 |
 |  Attributes
 |  ----------
 |  NTWX:   int
 |      controls writing of coordinate trajectory
 |      0: no coordinate trajectory is written (default)
 |      &gt;0: write solute and solvent coordinates every NTWX steps
 |      &lt;0: write solute coordinates every |NTWX| steps
 |  NTWSE:   int
 |      &gt;= 0 selection criteria for coordinate trajectory writing
 |      0: write normal coordinate trajectory
 |      &gt;0: write minimum-energy coordinate and energy trajectory (based on the
 |      energy entry selected by NTWSE and as blocks of length NTWX)
 |      (see configuration/energy.cc or ene_ana library for indices)
 |  NTWV:   int
 |      controls writing of velocity trajectory
 |      0: no velocity trajectory is written (default)
 |      &gt;0: write solute and solvent velocities every NTWV steps
 |      &lt;0: write solute velocities every |NTWV| steps
 |  NTWF:   int
 |      controls writing of force trajectory
 |      0: no force trajectory is written (default)
 |      &gt;0: write solute and solvent forces every NTWF steps
 |      &lt;0: write solute forces every |NTWF| steps
 |  NTWE:   int
 |      &gt;= 0 controls writing of energy trajectory
 |      0: no energy trajectory is written (default)
 |      &gt;0: write energy trajectory every NTWE steps
 |  NTWG:   int
 |      &gt;= 0 controls writing of free energy trajectory
 |      0: no free energy trajectory is written (default)
 |      &gt;0: write free energy trajectory every NTWG steps
 |  NTWB:   int
 |      &gt;= 0 controls writing of block-averaged energy trajectory
 |      0: no block averaged energy trajectory is written (default)
 |      &gt;0: write block-averaged energy variables every |NTWB| steps
 |          (and free energies if NTWG &gt; 0) trajectory
 |
 |  Method resolution order:
 |      WRITETRAJ
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTWX: int = 0, NTWSE: int = 0, NTWV: int = 0, NTWF: int = 0, NTWE: int = 0, NTWG: int = 0, NTWB: int = 0, content: List[str] = None)
 |      Args:
 |          NTWX:
 |          NTWSE:
 |          NTWV:
 |          NTWF:
 |          NTWE:
 |          NTWG:
 |          NTWB:
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NTWB&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTWE&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTWF...
 |
 |  name = &#39;WRITETRAJ&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[67]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">WRITETRAJ</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[67]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
WRITETRAJ
#        NTWX    NTWSE   NTWV    NTWF    NTWE    NTWG    NTWB
         10      0       0       0       10      0       0
END
</pre></div></div>
</div>
</section>
<section id="PRINTOUT-Block">
<h3>PRINTOUT Block<a class="headerlink" href="#PRINTOUT-Block" title="Permalink to this headline">¶</a></h3>
<p>This block is very similar to the <em>WRITETRAJ</em> block but the information about the energies (<em>NTPR</em>) is printed to the output file (.omd). By giving <em>NTPP</em>, dihedral angle transitions are written to the special trajectory (.trs).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[68]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">PRINTOUT</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on PRINTOUT in module pygromos.files.blocks.imd_blocks object:

class PRINTOUT(_generic_imd_block)
 |  PRINTOUT(NTPR: int = 0, NTPP: int = 0, content: List[str] = None)
 |
 |  PRINTOUT block
 |
 |      This Block manages the output frequency into the .omd/std-out file.
 |
 |  Attributes
 |  ----------
 |  NTPR: int
 |      print out energies, etc. every NTPR steps
 |  NTPP: int
 |      =1 perform dihedral angle transition monitoring
 |
 |  Method resolution order:
 |      PRINTOUT
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTPR: int = 0, NTPP: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;NTPP&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTPR&#39;: &lt;class &#39;int&#39;&gt;, &#39;name...
 |
 |  name = &#39;PRINTOUT&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[69]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">PRINTOUT</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[69]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PRINTOUT
#        NTPR    NTPP
         250     0
END
</pre></div></div>
</div>
</section>
<section id="PAIRLIST-Block">
<h3>PAIRLIST Block<a class="headerlink" href="#PAIRLIST-Block" title="Permalink to this headline">¶</a></h3>
<p>MD++ knows different algorithms for the generation of the pairlist, a list containing the atoms interacting with each other. Here, we use a grid based pairlist generation: the space is discretized into grid cells and only the neighboring cells are searched for interacting partners. The use of this algorithm results in a The pairlist is generated every 5th (<em>NSNB</em>) step. <em>RCUTP</em> and <em>RCUTL</em> are the cutoffs for the pairlist construction for the short-range and the long-range interactions. The
pairlist is generated every 5th (<em>NSNB</em>) step. <em>RCUTP</em> and <em>RCUTL</em> are the cutoffs for the pairlist construction for the short-range and the long-range interactions.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[70]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">PAIRLIST</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on PAIRLIST in module pygromos.files.blocks.imd_blocks object:

class PAIRLIST(_generic_imd_block)
 |  PAIRLIST(ALGORITHM: int = 0, NSNB: int = 0, RCUTP: float = 0, RCUTL: float = 0, SIZE: Union[str, float] = 0, TYPE: Union[str, bool] = False, content: List[str] = None)
 |
 |  PAIRLIST Block
 |
 |          This block is controlling the pairlist control.
 |
 |  Attributes
 |  ----------
 |  ALGORITHM: int
 |      standard(0) (gromos96 like pairlist)
 |      grid(1) (md++ grid pairlist)
 |      grid_cell(2) (creates a mask)
 |  NSNB:   int
 |      frequency (number of steps) a pairlist is constructed
 |  RCUTPL: float
 |      short-range cut-off in twin-range
 |  RCUTL:  float
 |      intermediate-range cut-off in twin-range
 |  SIZE:   str, float
 |      grid cell size (or auto = 0.5 * RCUTP)
 |  TYPE:   str, bool
 |      chargegoup(0) (chargegroup based cutoff)
 |      atomic(1)     (atom based cutoff)
 |
 |  Method resolution order:
 |      PAIRLIST
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, ALGORITHM: int = 0, NSNB: int = 0, RCUTP: float = 0, RCUTL: float = 0, SIZE: Union[str, float] = 0, TYPE: Union[str, bool] = False, content: List[str] = None)
 |      Args:
 |          ALGORITHM:
 |          NSNB:
 |          RCUTP:
 |          RCUTL:
 |          SIZE:
 |          TYPE:
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;ALGORITHM&#39;: &lt;class &#39;int&#39;&gt;, &#39;NSNB&#39;: &lt;class &#39;int&#39;&gt;, ...
 |
 |  name = &#39;PAIRLIST&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[71]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">PAIRLIST</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[71]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PAIRLIST
#        ALGORITHM       NSNB    RCUTP   RCUTL   SIZE    TYPE
         1       5       0.800000        1.400000        0.4     0
END
</pre></div></div>
</div>
</section>
<section id="POSITIONRES-Block">
<h3>POSITIONRES Block<a class="headerlink" href="#POSITIONRES-Block" title="Permalink to this headline">¶</a></h3>
<p>Finally, we want to restrain the position of our solute. The restraining is achieved by a harmonic special force term with a force constant of <em>CPOR</em>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[72]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.files.blocks.imd_blocks</span> <span class="kn">import</span> <span class="n">POSITIONRES</span>

<span class="n">help</span><span class="p">(</span><span class="n">POSITIONRES</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on class POSITIONRES in module pygromos.files.blocks.imd_blocks:

class POSITIONRES(_generic_imd_block)
 |  POSITIONRES(NTPOR: int = 0, NTPORB: int = 0, NTPORS: int = 0, CPOR: int = 0, content: List[str] = None)
 |
 |  POSITIONRES block
 |
 |      This block allows the managment of the Position Restraints during the Simulation.
 |
 |  Attributes
 |  ----------
 |  NTPOR:
 |      0..3 controls atom positions re(con)straining.
 |      0: no position re(con)straints (default)
 |      1: restraining with force constant CPOR
 |      2: restraining with force constant CPOR weighted by atomic B-factors
 |      3: constraining
 |  NTPORB:
 |      0,1 controls reading of reference positions and B-factors
 |      0: read reference positions from startup file.
 |      1: read reference positions and B-factors from special file
 |  NTPORS:
 |      0,1 controls scaling of reference positions upon pressure scaling
 |      0: do not scale reference positions
 |      1: scale reference positions
 |  CPOR:
 |      &gt;= 0 position restraining force constant
 |
 |  Method resolution order:
 |      POSITIONRES
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, NTPOR: int = 0, NTPORB: int = 0, NTPORS: int = 0, CPOR: int = 0, content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;CPOR&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTPOR&#39;: &lt;class &#39;int&#39;&gt;, &#39;NTP...
 |
 |  name = &#39;POSITIONRES&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[73]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">md_imd_file</span><span class="o">.</span><span class="n">add_block</span><span class="p">(</span><span class="n">block</span><span class="o">=</span><span class="n">POSITIONRES</span><span class="p">(</span><span class="n">NTPOR</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">NTPORB</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">CPOR</span><span class="o">=</span><span class="mi">25000</span><span class="p">))</span>

<span class="n">md_imd_file</span><span class="o">.</span><span class="n">POSITIONRES</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[73]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
POSITIONRES
#        NTPOR   NTPORB          NTPORS          CPOR
         1       1       0       25000
END
</pre></div></div>
</div>
<p>Now you should know the main blocks of the Gromos input files.</p>
</section>
<section id="Perform-Thermalisation">
<h3>Perform Thermalisation<a class="headerlink" href="#Perform-Thermalisation" title="Permalink to this headline">¶</a></h3>
<p>Next we will run multiple simulations, which will slowly heat up the simulation system. This can easily be done by modifying parameters on the run. First we will setup a gromos system and prepare it. Next we will perform the simulations.</p>
<hr class="docutils" />
<p><strong>Warning</strong>: Depending on your system’s speed this will take up to five hours. As we are using the simulation directory as the working directory there will be an error message after every job which can be ignored. ***</p>
<hr class="docutils" />
<p><em>Hint</em> : Have a look at all the output files eq peptide <em>.omd. If anything goes wrong, a message will be printed to the output file.</em>**</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[74]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">md</span>

<span class="c1">## Preparing emin gromos system</span>
<span class="n">in_eq_system</span> <span class="o">=</span> <span class="n">out_eminBox_system</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">in_eq_system</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;eq_thermalisation&quot;</span>
<span class="n">in_eq_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">project_dir</span>

<span class="c1">### Build position restraints</span>
<span class="n">restrain_res</span> <span class="o">=</span> <span class="p">[</span><span class="n">k</span> <span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="n">in_eq_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">residues</span> <span class="k">if</span><span class="p">(</span><span class="ow">not</span> <span class="n">k</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;CL-&quot;</span><span class="p">,</span> <span class="s2">&quot;SOLV&quot;</span><span class="p">))]</span>
<span class="n">in_eq_system</span><span class="o">.</span><span class="n">generate_posres</span><span class="p">(</span><span class="n">residues</span><span class="o">=</span><span class="n">restrain_res</span><span class="p">)</span>

<span class="c1">### Check simulation params</span>
<span class="n">in_eq_system</span><span class="o">.</span><span class="n">imd</span> <span class="o">=</span> <span class="n">md_imd_file</span>

<span class="c1">### Set simulation lengths:</span>
<span class="n">in_eq_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">STEP</span><span class="o">.</span><span class="n">NSTLIM</span> <span class="o">=</span> <span class="mi">1000</span> <span class="c1"># each temperature will do 1000 steps, this might take some time!</span>

<span class="n">in_eq_system</span><span class="o">.</span><span class="n">prepare_for_simulation</span><span class="p">(</span><span class="n">not_ligand_residues</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;CL-&quot;</span><span class="p">])</span>
<span class="n">sys_same</span> <span class="o">=</span> <span class="n">in_eq_system</span><span class="o">.</span><span class="n">name</span>

<span class="n">in_eq_system</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/464153645.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> <span class="ansi-red-fg">## Preparing emin gromos system</span>
<span class="ansi-green-fg">----&gt; 4</span><span class="ansi-red-fg"> </span>in_eq_system <span class="ansi-blue-fg">=</span> out_eminBox_system<span class="ansi-blue-fg">.</span>copy<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> in_eq_system<span class="ansi-blue-fg">.</span>name <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">&#34;eq_thermalisation&#34;</span>
<span class="ansi-green-intense-fg ansi-bold">      6</span> in_eq_system<span class="ansi-blue-fg">.</span>work_folder <span class="ansi-blue-fg">=</span> project_dir

<span class="ansi-red-fg">NameError</span>: name &#39;out_eminBox_system&#39; is not defined
</pre></div></div>
</div>
<p>After preparing the system we can now start simulating.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[75]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">## run Thermalisation.</span>
<span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">simulation</span>

<span class="n">temperatures</span> <span class="o">=</span> <span class="p">[</span><span class="mi">60</span><span class="p">,</span> <span class="mi">120</span><span class="p">,</span> <span class="mi">180</span><span class="p">,</span> <span class="mi">240</span><span class="p">,</span> <span class="mi">300</span><span class="p">]</span> <span class="c1">#the temperatures we want to simulate.</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Heating upt to: &quot;</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">runID</span><span class="p">,</span> <span class="n">temperature</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">temperatures</span><span class="p">):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">temperature</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>

    <span class="c1">#adapt temperature</span>
    <span class="n">in_eq_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">MULTIBATH</span><span class="o">.</span><span class="n">TEMP0</span> <span class="o">=</span> <span class="p">[</span><span class="n">temperature</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">in_eq_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">MULTIBATH</span><span class="o">.</span><span class="n">NBATHS</span><span class="p">)]</span>

    <span class="c1">#turn off the posres for the last run.</span>
    <span class="k">if</span><span class="p">(</span><span class="n">runID</span><span class="o">+</span><span class="mi">1</span> <span class="o">==</span> <span class="nb">len</span><span class="p">(</span><span class="n">temperatures</span><span class="p">)):</span> <span class="c1">#Last Run</span>
        <span class="n">in_eq_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">POSITIONRES</span><span class="o">.</span><span class="n">NTPOR</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">in_eq_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">POSITIONRES</span><span class="o">.</span><span class="n">CPOR</span> <span class="o">=</span> <span class="mi">0</span>

        <span class="n">out_eq_system</span> <span class="o">=</span> <span class="n">simulation</span><span class="p">(</span><span class="n">in_gromos_simulation_system</span><span class="o">=</span><span class="n">in_eq_system</span><span class="p">,</span>
                                   <span class="n">step_name</span><span class="o">=</span><span class="s2">&quot;f_&quot;</span><span class="o">+</span><span class="n">sys_same</span><span class="p">,</span>
                                   <span class="n">simulation_runs</span><span class="o">=</span><span class="n">runID</span><span class="o">+</span><span class="mi">1</span><span class="p">)</span>
        <span class="k">break</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">out_eq_system</span> <span class="o">=</span> <span class="n">simulation</span><span class="p">(</span><span class="n">in_gromos_simulation_system</span><span class="o">=</span><span class="n">in_eq_system</span><span class="p">,</span>
                                   <span class="n">step_name</span><span class="o">=</span><span class="s2">&quot;f_&quot;</span><span class="o">+</span><span class="n">sys_same</span><span class="p">,</span>
                                   <span class="n">simulation_runs</span><span class="o">=</span><span class="n">runID</span><span class="o">+</span><span class="mi">1</span><span class="p">,</span>
                                   <span class="n">analysis_script</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
                                   <span class="n">verbose</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

    <span class="n">in_eq_system</span> <span class="o">=</span> <span class="n">out_eq_system</span>


<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;done&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Heating upt to:         60
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/788546794.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      9</span>
<span class="ansi-green-intense-fg ansi-bold">     10</span>     <span class="ansi-red-fg">#adapt temperature</span>
<span class="ansi-green-fg">---&gt; 11</span><span class="ansi-red-fg">     </span>in_eq_system<span class="ansi-blue-fg">.</span>imd<span class="ansi-blue-fg">.</span>MULTIBATH<span class="ansi-blue-fg">.</span>TEMP0 <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">[</span>temperature <span class="ansi-green-fg">for</span> x <span class="ansi-green-fg">in</span> range<span class="ansi-blue-fg">(</span>in_eq_system<span class="ansi-blue-fg">.</span>imd<span class="ansi-blue-fg">.</span>MULTIBATH<span class="ansi-blue-fg">.</span>NBATHS<span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">]</span>
<span class="ansi-green-intense-fg ansi-bold">     12</span>
<span class="ansi-green-intense-fg ansi-bold">     13</span>     <span class="ansi-red-fg">#turn off the posres for the last run.</span>

<span class="ansi-red-fg">NameError</span>: name &#39;in_eq_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[76]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#This took some time, so lets store it! So we can use it later again.</span>
<span class="n">out_eq_therm_system_path</span> <span class="o">=</span> <span class="n">out_eq_system</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="n">out_eq_system</span><span class="o">.</span><span class="n">work_folder</span><span class="o">+</span><span class="s2">&quot;/eq_therm_result2.obj&quot;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2176658082.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#This took some time, so lets store it! So we can use it later again.</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> </span>out_eq_therm_system_path <span class="ansi-blue-fg">=</span> out_eq_system<span class="ansi-blue-fg">.</span>save<span class="ansi-blue-fg">(</span>out_eq_system<span class="ansi-blue-fg">.</span>work_folder<span class="ansi-blue-fg">+</span><span class="ansi-blue-fg">&#34;/eq_therm_result2.obj&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_eq_system&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Analysis-of-Thermalisation:">
<h3>Analysis of Thermalisation:<a class="headerlink" href="#Analysis-of-Thermalisation:" title="Permalink to this headline">¶</a></h3>
<p>Next we want to check the results of our Thermalisation. We will have a look at the coordinate trajectory and the temperatures of the system.</p>
<section id="Check-the-coordinate-traj">
<h4>Check the coordinate traj<a class="headerlink" href="#Check-the-coordinate-traj" title="Permalink to this headline">¶</a></h4>
<p>First we will check the trajectory of the termalisation. Again we first need to shift the coordinates, so we can se the fully connected peptide.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[77]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">coordinate_traj</span> <span class="o">=</span> <span class="n">out_eq_system</span><span class="o">.</span><span class="n">trc</span>
<span class="n">coordinate_traj</span><span class="o">.</span><span class="n">image_molecules</span><span class="p">()</span>
<span class="n">view</span> <span class="o">=</span> <span class="n">coordinate_traj</span><span class="o">.</span><span class="n">view</span>
<span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/673463103.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>coordinate_traj <span class="ansi-blue-fg">=</span> out_eq_system<span class="ansi-blue-fg">.</span>trc
<span class="ansi-green-intense-fg ansi-bold">      2</span> coordinate_traj<span class="ansi-blue-fg">.</span>image_molecules<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> view <span class="ansi-blue-fg">=</span> coordinate_traj<span class="ansi-blue-fg">.</span>view
<span class="ansi-green-intense-fg ansi-bold">      4</span> view

<span class="ansi-red-fg">NameError</span>: name &#39;out_eq_system&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Check-the-temperature">
<h4>Check the temperature<a class="headerlink" href="#Check-the-temperature" title="Permalink to this headline">¶</a></h4>
<p>After this, we would like to see the temperature development. First lets collect all temperature calculations from the simulation:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[78]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">energy_traj</span> <span class="o">=</span> <span class="n">out_eq_system</span><span class="o">.</span><span class="n">tre</span>
<span class="n">temperatures</span><span class="o">=</span> <span class="n">energy_traj</span><span class="o">.</span><span class="n">get_temperature</span><span class="p">()</span>
<span class="n">temperatures</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/965721347.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>energy_traj <span class="ansi-blue-fg">=</span> out_eq_system<span class="ansi-blue-fg">.</span>tre
<span class="ansi-green-intense-fg ansi-bold">      2</span> temperatures<span class="ansi-blue-fg">=</span> energy_traj<span class="ansi-blue-fg">.</span>get_temperature<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> temperatures

<span class="ansi-red-fg">NameError</span>: name &#39;out_eq_system&#39; is not defined
</pre></div></div>
</div>
<p>Next we want to visualize the data to get an impresstionof what happened. Can you see how the Temperature was constantly raised per simulation and started to equilibrate forming plateus?</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[79]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>

<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">temperatures</span><span class="o">.</span><span class="n">bath1</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;TBath1&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">temperatures</span><span class="o">.</span><span class="n">bath2</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;TBath2&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;$T~[K]$&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;$t~[ps]$&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">AttributeError</span>                            Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2602914951.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-green-fg">from</span> matplotlib <span class="ansi-green-fg">import</span> pyplot <span class="ansi-green-fg">as</span> plt
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-fg">----&gt; 3</span><span class="ansi-red-fg"> </span>plt<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>temperatures<span class="ansi-blue-fg">.</span>bath1<span class="ansi-blue-fg">,</span> label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;TBath1&#34;</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> plt<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>temperatures<span class="ansi-blue-fg">.</span>bath2<span class="ansi-blue-fg">,</span> label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;TBath2&#34;</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span>

<span class="ansi-red-fg">AttributeError</span>: &#39;list&#39; object has no attribute &#39;bath1&#39;
</pre></div></div>
</div>
</section>
</section>
</section>
<section id="Molecular-dynamics-sampling-simulation.">
<h2>Molecular dynamics sampling simulation.<a class="headerlink" href="#Molecular-dynamics-sampling-simulation." title="Permalink to this headline">¶</a></h2>
<p>The equilibration period already produced a short simulation at constant temperature and volume. At this point we want to elongate the simulation to a nanosecond under constant temperature and pressure.</p>
<section id="id4">
<h3>Simulation Paramters<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<p>First, we don’t use position restraining anymore and so, the POSITIONRES block in the next step. Next we want to simulate under constant pressure rather than constant volume (NVT-&gt; NPT). For this purpose we have to add an additional block:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[80]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.data.simulation_parameters_templates</span> <span class="kn">import</span> <span class="n">template_md_tut</span> <span class="k">as</span> <span class="n">template_md</span>
<span class="kn">from</span> <span class="nn">pygromos.files.simulation_parameters.imd</span> <span class="kn">import</span> <span class="n">Imd</span>

<span class="n">imd_file</span> <span class="o">=</span> <span class="n">Imd</span><span class="p">(</span><span class="n">template_md</span><span class="p">)</span>

<span class="n">help</span><span class="p">(</span><span class="n">imd_file</span><span class="o">.</span><span class="n">PRESSURESCALE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Help on PRESSURESCALE in module pygromos.files.blocks.imd_blocks object:

class PRESSURESCALE(_generic_imd_block)
 |  PRESSURESCALE(COUPLE: int = 0, SCALE: int = 0, COMP: float = 0, TAUP: float = 0, VIRIAL: int = 0, SEMIANISOTROPIC: List[int] = [], PRES0: List[List[float]] = [[0, 0, 0], [0, 0, 0], [0, 0, 0]], content: List[str] = None)
 |
 |  PRESSURESCALE Block
 |  This block controls the barostat of the simulation
 |
 |  Attributes
 |  -----------
 |  COUPLE: int
 |      off(0), calc(1), scale(2)
 |  SCALE:  int
 |      off(0), iso(1), aniso(2), full(3), semianiso(4)
 |  COMP:   float
 |      compessibility
 |  TAUP:   float
 |      coupling strength
 |  VIRIAL: int
 |      none(0), atomic(1), group(2)
 |  SEMIANISOTROPIC:    List[int]
 |      (semianisotropic couplings: X, Y, Z)
 |      e.g. 1 1 2: x and y jointly coupled and z separately coupled
 |      e.g. 0 0 1: constant area (xy-plane) and z coupled to a bath
 |  PRES0:  List[List[float]]
 |      reference pressure
 |
 |  Method resolution order:
 |      PRESSURESCALE
 |      _generic_imd_block
 |      pygromos.files.blocks._general_blocks._generic_gromos_block
 |      builtins.object
 |
 |  Methods defined here:
 |
 |  __init__(self, COUPLE: int = 0, SCALE: int = 0, COMP: float = 0, TAUP: float = 0, VIRIAL: int = 0, SEMIANISOTROPIC: List[int] = [], PRES0: List[List[float]] = [[0, 0, 0], [0, 0, 0], [0, 0, 0]], content: List[str] = None)
 |      Initialize self.  See help(type(self)) for accurate signature.
 |
 |  read_content_from_str(self, content: List[str])
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes defined here:
 |
 |  __annotations__ = {&#39;COMP&#39;: &lt;class &#39;float&#39;&gt;, &#39;COUPLE&#39;: &lt;class &#39;int&#39;&gt;, &#39;...
 |
 |  name = &#39;PRESSURESCALE&#39;
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from _generic_imd_block:
 |
 |  block_to_string(self) -&gt; str
 |
 |  ----------------------------------------------------------------------
 |  Methods inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __copy__(self)
 |
 |  __deepcopy__(self, memo)
 |
 |  __eq__(self, _generic_gromos_block__o: object) -&gt; bool
 |      Return self==value.
 |
 |  __iter__(self)
 |
 |  __repr__(self)
 |      Return repr(self).
 |
 |  __str__(self)
 |      Return str(self).
 |
 |  get_name(self)
 |
 |  ----------------------------------------------------------------------
 |  Data descriptors inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __dict__
 |      dictionary for instance variables (if defined)
 |
 |  __weakref__
 |      list of weak references to the object (if defined)
 |
 |  ----------------------------------------------------------------------
 |  Data and other attributes inherited from pygromos.files.blocks._general_blocks._generic_gromos_block:
 |
 |  __hash__ = None

</pre></div></div>
</div>
<p>In the PRESSURESCALE block we tell Gromos to calculate and scale the pressure by setting <em>COUPLE</em> to <span class="math notranslate nohighlight">\(2\)</span>. As the box should be isotropically scaled we set <em>SCALE</em> equal to <span class="math notranslate nohighlight">\(1\)</span>. The weak-coupling method (Sec. 2-12.2.2) uses two additional parameters: <em>COMP</em> is the isothermal compressibility and <em>TAUP</em> is the coupling time. We are calculating the molecular virial (<em>VIRIAL</em> is equal to <span class="math notranslate nohighlight">\(2\)</span>), so intramolecular forces don’t contribute to the pressure. The next line is only used for
semi-anisotropic pressure coupling and can be ignored in our case. Finally, we have to specify the reference pressure in a tensor form</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[81]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">imd_file</span><span class="o">.</span><span class="n">PRESSURESCALE</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[81]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
PRESSURESCALE
#        COUPLE          SCALE   COMP    TAUP    VIRIAL
         2       1       0.000458        0.500000        2
#        SEMIANISOTROPIC COUPLINGS(X, Y, Z)
         1       1       1
#        PRES0(1...3,1...3)
         0.06102         0.0     0.0
         0.0     0.06102         0.0
         0.0     0.0     0.06102
END
</pre></div></div>
</div>
</section>
<section id="Perform-MD---Production">
<h3>Perform MD - Production<a class="headerlink" href="#Perform-MD---Production" title="Permalink to this headline">¶</a></h3>
<p>In the other blocks only minor things have changed: the temperature was set to 300K and the trajectories are written out less often (every 250th step only). Next we are going to set up the Gromos System. We will remove the position restraints and the old trajectories.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[82]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">md</span>

<span class="n">in_md_system</span> <span class="o">=</span> <span class="n">out_eq_system</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>

<span class="n">in_md_system</span><span class="o">.</span><span class="n">work_folder</span> <span class="o">=</span> <span class="n">project_dir</span>
<span class="n">in_md_system</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="s2">&quot;md&quot;</span>

<span class="c1">### Check simulation params</span>
<span class="n">in_md_system</span><span class="o">.</span><span class="n">imd</span> <span class="o">=</span> <span class="n">imd_file</span>

<span class="c1">## imd parameters</span>
<span class="n">in_md_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">STEP</span><span class="o">.</span><span class="n">NSTLIM</span> <span class="o">=</span> <span class="mi">1000</span>
<span class="n">in_md_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">WRITETRAJ</span><span class="o">.</span><span class="n">NTWX</span> <span class="o">=</span> <span class="mi">10</span>
<span class="n">in_md_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">WRITETRAJ</span><span class="o">.</span><span class="n">NTWE</span> <span class="o">=</span> <span class="mi">10</span>
<span class="n">in_md_system</span><span class="o">.</span><span class="n">imd</span><span class="o">.</span><span class="n">INITIALISE</span><span class="o">.</span><span class="n">NTIVEL</span> <span class="o">=</span> <span class="mi">0</span>

<span class="n">in_md_system</span><span class="o">.</span><span class="n">prepare_for_simulation</span><span class="p">()</span>

<span class="n">in_md_system</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3190364862.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-green-fg">from</span> pygromos<span class="ansi-blue-fg">.</span>simulations<span class="ansi-blue-fg">.</span>modules<span class="ansi-blue-fg">.</span>preset_simulation_modules <span class="ansi-green-fg">import</span> md
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-fg">----&gt; 3</span><span class="ansi-red-fg"> </span>in_md_system <span class="ansi-blue-fg">=</span> out_eq_system<span class="ansi-blue-fg">.</span>copy<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> in_md_system<span class="ansi-blue-fg">.</span>work_folder <span class="ansi-blue-fg">=</span> project_dir

<span class="ansi-red-fg">NameError</span>: name &#39;out_eq_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[83]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_md_system</span> <span class="o">=</span> <span class="n">md</span><span class="p">(</span><span class="n">in_gromos_system</span><span class="o">=</span><span class="n">in_md_system</span><span class="p">,</span>
                   <span class="n">step_name</span><span class="o">=</span><span class="s2">&quot;g_&quot;</span><span class="o">+</span><span class="n">in_md_system</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
                   <span class="n">simulation_runs</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
<span class="n">out_md_system</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/432426042.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> out_md_system = md(in_gromos_system=in_md_system,
</span><span class="ansi-green-intense-fg ansi-bold">      2</span>                    step_name<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;g_&#34;</span><span class="ansi-blue-fg">+</span>in_md_system<span class="ansi-blue-fg">.</span>name<span class="ansi-blue-fg">,</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span>                    simulation_runs=3)
<span class="ansi-green-intense-fg ansi-bold">      4</span> out_md_system

<span class="ansi-red-fg">NameError</span>: name &#39;in_md_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[84]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_md_system_path</span> <span class="o">=</span> <span class="n">out_md_system</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="n">out_md_system</span><span class="o">.</span><span class="n">work_folder</span><span class="o">+</span><span class="s2">&quot;/md_result.obj&quot;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2064924672.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>out_md_system_path <span class="ansi-blue-fg">=</span> out_md_system<span class="ansi-blue-fg">.</span>save<span class="ansi-blue-fg">(</span>out_md_system<span class="ansi-blue-fg">.</span>work_folder<span class="ansi-blue-fg">+</span><span class="ansi-blue-fg">&#34;/md_result.obj&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;out_md_system&#39; is not defined
</pre></div></div>
</div>
<p>After all the jobs are finished, you should start to analyse the trajectories.</p>
</section>
<section id="id5">
<h3>Analysis<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<section id="First-Look-Analysis">
<h4>First Look Analysis<a class="headerlink" href="#First-Look-Analysis" title="Permalink to this headline">¶</a></h4>
<p>Let’s first have a look on the final coordinates.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[85]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">out_md_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">shift_periodic_boundary</span><span class="p">()</span>
<span class="n">out_md_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">recenter_pbc</span><span class="p">()</span>
<span class="n">out_md_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">recreate_view</span><span class="p">()</span>
<span class="n">out_md_system</span><span class="o">.</span><span class="n">cnf</span><span class="o">.</span><span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2399734183.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>out_md_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>shift_periodic_boundary<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span> out_md_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>recenter_pbc<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> out_md_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>recreate_view<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> out_md_system<span class="ansi-blue-fg">.</span>cnf<span class="ansi-blue-fg">.</span>view

<span class="ansi-red-fg">NameError</span>: name &#39;out_md_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[86]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">coordinate_traj</span> <span class="o">=</span> <span class="n">out_md_system</span><span class="o">.</span><span class="n">trc</span>

<span class="n">coordinate_traj</span><span class="o">.</span><span class="n">image_molecules</span><span class="p">()</span>
<span class="n">coordinate_traj</span><span class="o">.</span><span class="n">recreate_view</span><span class="p">()</span>
<span class="n">view</span> <span class="o">=</span> <span class="n">coordinate_traj</span><span class="o">.</span><span class="n">view</span>
<span class="n">view</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3818948884.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>coordinate_traj <span class="ansi-blue-fg">=</span> out_md_system<span class="ansi-blue-fg">.</span>trc
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> coordinate_traj<span class="ansi-blue-fg">.</span>image_molecules<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> coordinate_traj<span class="ansi-blue-fg">.</span>recreate_view<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> view <span class="ansi-blue-fg">=</span> coordinate_traj<span class="ansi-blue-fg">.</span>view

<span class="ansi-red-fg">NameError</span>: name &#39;out_md_system&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Energy-Analysis">
<h4>Energy Analysis<a class="headerlink" href="#Energy-Analysis" title="Permalink to this headline">¶</a></h4>
<p>The program calculates the average of the specified properties as well as the root-mean-square deviations (rmsd) and a statistical error estimate (error est.). The error estimate is calculated from block averages of growing sizes extrapolating to infinite block size1</p>
<hr class="docutils" />
<p><strong>Warning:</strong> Sometimes the error estimates are NaN (not a number), which is due to the fact that we do not have enough values to calculate a meaningful error estimate. ***</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[87]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">energy_traj</span> <span class="o">=</span> <span class="n">out_md_system</span><span class="o">.</span><span class="n">tre</span>
<span class="n">total_energies</span> <span class="o">=</span> <span class="n">energy_traj</span><span class="o">.</span><span class="n">get_totals</span><span class="p">()</span>
<span class="n">total_energies</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/782681520.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>energy_traj <span class="ansi-blue-fg">=</span> out_md_system<span class="ansi-blue-fg">.</span>tre
<span class="ansi-green-intense-fg ansi-bold">      2</span> total_energies <span class="ansi-blue-fg">=</span> energy_traj<span class="ansi-blue-fg">.</span>get_totals<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> total_energies

<span class="ansi-red-fg">NameError</span>: name &#39;out_md_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[88]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Next we plot some data of the energies to se their development.</span>

<span class="n">time_axis</span> <span class="o">=</span> <span class="n">energy_traj</span><span class="o">.</span><span class="n">time</span>
<span class="n">total_energies</span><span class="o">.</span><span class="n">totpot</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">total_energies</span><span class="o">.</span><span class="n">totkin</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">total_energies</span><span class="o">.</span><span class="n">totene</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;t [ps]&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;V [kJ/mol]&quot;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Simulation Data&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1783350399.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#Next we plot some data of the energies to se their development.</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span>
<span class="ansi-green-fg">----&gt; 3</span><span class="ansi-red-fg"> </span>time_axis <span class="ansi-blue-fg">=</span> energy_traj<span class="ansi-blue-fg">.</span>time
<span class="ansi-green-intense-fg ansi-bold">      4</span> total_energies<span class="ansi-blue-fg">.</span>totpot<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> total_energies<span class="ansi-blue-fg">.</span>totkin<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;energy_traj&#39; is not defined
</pre></div></div>
</div>
<p>In the Following we want to see the different contributions of the molecules with their VdW and Culomb interaxtions. we add up the van der Waals and Coulomb energies of the peptide-peptide interactions.</p>
<p>Now we want to look at two variables: * peptide_water_nonbonded * peptide_Cl_nonbonded</p>
<p>The information about the nonbondeds can be retrieved from the ForceGroup Nonbonded contributions. If you check the used imd file, you reckognize three Force groups in the FORCEGROUP block. These are limited by the last atom of the group and you can check the group atoms in the topology or coordinate file. They should correspond here to: 1. Peptide 2. 2 Cl- ions 3. Water molecules</p>
<p>The interactions of these forcegroups are stored in the following pattern, which can be adapted to any force group number (PseudoCode: for i in range(1+nFroceGroups) for j in range(i, nFroceGroups)):</p>
<ol class="arabic simple">
<li><p>ForceGroup - 1. ForceGroup: intra force group nonbondeds (peptide - peptide)</p></li>
<li><p>ForceGroup - 2. ForceGroup: inter force group nonbondeds between force group 1 and 2 (peptide - Cl-)</p></li>
<li><p>ForceGroup - 3. ForceGroup: inter force group nonbondeds between force group 1 and 3 (peptide - Water)</p></li>
<li><p>ForceGroup - 2. ForceGroup: intra force group nonbondeds (Cl- - Cl-)</p></li>
<li><p>ForceGroup - 3. ForceGroup: inter force group nonbondeds between force group 2 and 3 (Cl- - Water)</p></li>
<li><p>ForceGroup - 3. ForceGroup: intra force group nonbondeds (Water - Water)</p></li>
</ol>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[89]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#First lets get the Force Group Energy contributions:</span>
<span class="n">forceGroupNonbondedContributions</span> <span class="o">=</span> <span class="n">energy_traj</span><span class="o">.</span><span class="n">get_nonbondedContributions</span><span class="p">()</span>

<span class="c1">#Here we give each Force Group contribution category nice names:</span>
<span class="n">peptide_peptide_nonbonded</span> <span class="o">=</span> <span class="n">forceGroupNonbondedContributions</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span>
<span class="n">peptide_Cl_nonbonded</span> <span class="o">=</span> <span class="n">forceGroupNonbondedContributions</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">2</span><span class="p">]</span>
<span class="n">peptide_water_nonbonded</span> <span class="o">=</span> <span class="n">forceGroupNonbondedContributions</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">3</span><span class="p">]</span>

<span class="n">Cl_Cl_nonbonded</span> <span class="o">=</span> <span class="n">forceGroupNonbondedContributions</span><span class="p">[</span><span class="mi">2</span><span class="p">][</span><span class="mi">2</span><span class="p">]</span>
<span class="n">Cl_water_nonbonded</span> <span class="o">=</span> <span class="n">forceGroupNonbondedContributions</span><span class="p">[</span><span class="mi">2</span><span class="p">][</span><span class="mi">3</span><span class="p">]</span>

<span class="n">water_water_nonbonded</span> <span class="o">=</span> <span class="n">forceGroupNonbondedContributions</span><span class="p">[</span><span class="mi">3</span><span class="p">][</span><span class="mi">3</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2615102029.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#First lets get the Force Group Energy contributions:</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> </span>forceGroupNonbondedContributions <span class="ansi-blue-fg">=</span> energy_traj<span class="ansi-blue-fg">.</span>get_nonbondedContributions<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> <span class="ansi-red-fg">#Here we give each Force Group contribution category nice names:</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> peptide_peptide_nonbonded <span class="ansi-blue-fg">=</span> forceGroupNonbondedContributions<span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">]</span>

<span class="ansi-red-fg">NameError</span>: name &#39;energy_traj&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[90]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Here we look at the different Nonbonded contributions of each</span>
<span class="n">peptide_peptide_nonbonded</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Peptide_intra_Nonbondeds&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;V [kJ/mol]&quot;</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;writeOuts&quot;</span><span class="p">,</span> <span class="n">subplots</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">Cl_Cl_nonbonded</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;CL_intra_Nonbondeds&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;V [kJ/mol]&quot;</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;writeOuts&quot;</span><span class="p">,</span> <span class="n">subplots</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">water_water_nonbonded</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Water_intra_Nonbondeds&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;V [kJ/mol]&quot;</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;writeOuts&quot;</span><span class="p">,</span> <span class="n">subplots</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2680003299.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#Here we look at the different Nonbonded contributions of each</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> </span>peptide_peptide_nonbonded<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>title<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;Peptide_intra_Nonbondeds&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;V [kJ/mol]&#34;</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;writeOuts&#34;</span><span class="ansi-blue-fg">,</span> subplots<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> Cl_Cl_nonbonded<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>title<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;CL_intra_Nonbondeds&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;V [kJ/mol]&#34;</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;writeOuts&#34;</span><span class="ansi-blue-fg">,</span> subplots<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> water_water_nonbonded<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>title<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;Water_intra_Nonbondeds&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;V [kJ/mol]&#34;</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;writeOuts&#34;</span><span class="ansi-blue-fg">,</span> subplots<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;peptide_peptide_nonbonded&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[91]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#Next we calculate the sum of all contributions</span>
<span class="c1">#The sum over the axis 1 sums up all contribution types for a timestep t.</span>
<span class="n">peptide_peptide_nonbonded</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s2">&quot;peptide - peptide&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">peptide_Cl_nonbonded</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s2">&quot;peptide - Cl-&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">peptide_water_nonbonded</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s2">&quot;peptide - Water&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;V [kJ/mol]&quot;</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;writeOuts&quot;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Peptide Nonbondeds&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3181684095.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg">#Next we calculate the sum of all contributions</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span> <span class="ansi-red-fg">#The sum over the axis 1 sums up all contribution types for a timestep t.</span>
<span class="ansi-green-fg">----&gt; 3</span><span class="ansi-red-fg"> </span>peptide_peptide_nonbonded<span class="ansi-blue-fg">.</span>sum<span class="ansi-blue-fg">(</span>axis<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;peptide - peptide&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> peptide_Cl_nonbonded<span class="ansi-blue-fg">.</span>sum<span class="ansi-blue-fg">(</span>axis<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;peptide - Cl-&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      5</span> peptide_water_nonbonded<span class="ansi-blue-fg">.</span>sum<span class="ansi-blue-fg">(</span>axis<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;peptide - Water&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;V [kJ/mol]&#34;</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;writeOuts&#34;</span><span class="ansi-blue-fg">,</span> title<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;Peptide Nonbondeds&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;peptide_peptide_nonbonded&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[92]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Here we look at the total nonbonded Contributions of the peptide</span>
<span class="n">peptide_nonbonded</span> <span class="o">=</span> <span class="n">peptide_peptide_nonbonded</span><span class="o">+</span><span class="n">peptide_Cl_nonbonded</span><span class="o">+</span><span class="n">peptide_water_nonbonded</span>
<span class="c1">#The sum over the axis 1 sums up all contribution types for a timestep t.</span>
<span class="n">peptide_nonbonded</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;peptide&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;V [kJ/mol]&quot;</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;steps&quot;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Total Nonbonded Contributions of peptide&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3405475826.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-intense-fg ansi-bold">      1</span> <span class="ansi-red-fg"># Here we look at the total nonbonded Contributions of the peptide</span>
<span class="ansi-green-fg">----&gt; 2</span><span class="ansi-red-fg"> </span>peptide_nonbonded <span class="ansi-blue-fg">=</span> peptide_peptide_nonbonded<span class="ansi-blue-fg">+</span>peptide_Cl_nonbonded<span class="ansi-blue-fg">+</span>peptide_water_nonbonded
<span class="ansi-green-intense-fg ansi-bold">      3</span> <span class="ansi-red-fg">#The sum over the axis 1 sums up all contribution types for a timestep t.</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> peptide_nonbonded<span class="ansi-blue-fg">.</span>sum<span class="ansi-blue-fg">(</span>axis<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span> label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;peptide&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;V [kJ/mol]&#34;</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;steps&#34;</span><span class="ansi-blue-fg">,</span> title<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;Total Nonbonded Contributions of peptide&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;peptide_peptide_nonbonded&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Coordinate-Analysis">
<h4>Coordinate Analysis<a class="headerlink" href="#Coordinate-Analysis" title="Permalink to this headline">¶</a></h4>
<p>In the next step, we want to have a look at the coordinates of our simulation. first we will dcalculate the rmsd, second we will analyse the distance of the CL- to the positive charged atoms in the peptide.</p>
<hr class="docutils" />
<p><strong>Warning</strong>: Under Development ;) ***</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[93]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">final_coordinate</span> <span class="o">=</span> <span class="n">out_md_system</span><span class="o">.</span><span class="n">cnf</span>
<span class="n">coordinate_traj</span> <span class="o">=</span> <span class="n">out_md_system</span><span class="o">.</span><span class="n">trc</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3848250193.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>final_coordinate <span class="ansi-blue-fg">=</span> out_md_system<span class="ansi-blue-fg">.</span>cnf
<span class="ansi-green-intense-fg ansi-bold">      2</span> coordinate_traj <span class="ansi-blue-fg">=</span> out_md_system<span class="ansi-blue-fg">.</span>trc

<span class="ansi-red-fg">NameError</span>: name &#39;out_md_system&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[94]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">coordinate_traj</span><span class="o">.</span><span class="n">rmsd</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;RMSD&quot;</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;t [ps]&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1569724467.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>coordinate_traj<span class="ansi-blue-fg">.</span>rmsd<span class="ansi-blue-fg">(</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>sum<span class="ansi-blue-fg">(</span>axis<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;RMSD&#34;</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;t [ps]&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;coordinate_traj&#39; is not defined
</pre></div></div>
</div>
</section>
<section id="Distance-CL--to-the-peptide">
<h4>Distance CL- to the peptide<a class="headerlink" href="#Distance-CL--to-the-peptide" title="Permalink to this headline">¶</a></h4>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[95]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;CL- atoms:&quot;</span><span class="p">,</span> <span class="p">[(</span><span class="n">ap</span><span class="o">.</span><span class="n">atomType</span><span class="p">,</span> <span class="n">ap</span><span class="o">.</span><span class="n">atomID</span><span class="p">)</span> <span class="k">for</span> <span class="n">ap</span> <span class="ow">in</span> <span class="n">final_coordinate</span><span class="o">.</span><span class="n">POSITION</span> <span class="k">if</span><span class="p">(</span><span class="n">ap</span><span class="o">.</span><span class="n">resName</span> <span class="o">==</span> <span class="s2">&quot;CL-&quot;</span><span class="p">)])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;ARG atoms:&quot;</span><span class="p">,</span> <span class="p">[(</span><span class="n">ap</span><span class="o">.</span><span class="n">atomType</span><span class="p">,</span> <span class="n">ap</span><span class="o">.</span><span class="n">atomID</span><span class="p">)</span> <span class="k">for</span> <span class="n">ap</span> <span class="ow">in</span> <span class="n">final_coordinate</span><span class="o">.</span><span class="n">POSITION</span> <span class="k">if</span><span class="p">(</span><span class="n">ap</span><span class="o">.</span><span class="n">resName</span> <span class="o">==</span> <span class="s2">&quot;ARG&quot;</span><span class="p">)])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;LYSH atoms:&quot;</span><span class="p">,</span> <span class="p">[(</span><span class="n">ap</span><span class="o">.</span><span class="n">atomType</span><span class="p">,</span> <span class="n">ap</span><span class="o">.</span><span class="n">atomID</span><span class="p">)</span> <span class="k">for</span> <span class="n">ap</span> <span class="ow">in</span> <span class="n">final_coordinate</span><span class="o">.</span><span class="n">POSITION</span> <span class="k">if</span><span class="p">(</span><span class="n">ap</span><span class="o">.</span><span class="n">resName</span> <span class="o">==</span> <span class="s2">&quot;LYSH&quot;</span><span class="p">)])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;NTERM atoms:&quot;</span><span class="p">,</span> <span class="p">[(</span><span class="n">ap</span><span class="o">.</span><span class="n">atomType</span><span class="p">,</span> <span class="n">ap</span><span class="o">.</span><span class="n">atomID</span><span class="p">)</span> <span class="k">for</span> <span class="n">ap</span> <span class="ow">in</span> <span class="n">final_coordinate</span><span class="o">.</span><span class="n">POSITION</span> <span class="k">if</span><span class="p">(</span><span class="n">ap</span><span class="o">.</span><span class="n">resName</span> <span class="o">==</span> <span class="s2">&quot;VAL&quot;</span><span class="p">)])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/1045973654.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>print<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#34;CL- atoms:&#34;</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>atomType<span class="ansi-blue-fg">,</span> ap<span class="ansi-blue-fg">.</span>atomID<span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">for</span> ap <span class="ansi-green-fg">in</span> final_coordinate<span class="ansi-blue-fg">.</span>POSITION <span class="ansi-green-fg">if</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>resName <span class="ansi-blue-fg">==</span> <span class="ansi-blue-fg">&#34;CL-&#34;</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      2</span> print<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#34;ARG atoms:&#34;</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>atomType<span class="ansi-blue-fg">,</span> ap<span class="ansi-blue-fg">.</span>atomID<span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">for</span> ap <span class="ansi-green-fg">in</span> final_coordinate<span class="ansi-blue-fg">.</span>POSITION <span class="ansi-green-fg">if</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>resName <span class="ansi-blue-fg">==</span> <span class="ansi-blue-fg">&#34;ARG&#34;</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      3</span> print<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#34;LYSH atoms:&#34;</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>atomType<span class="ansi-blue-fg">,</span> ap<span class="ansi-blue-fg">.</span>atomID<span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">for</span> ap <span class="ansi-green-fg">in</span> final_coordinate<span class="ansi-blue-fg">.</span>POSITION <span class="ansi-green-fg">if</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>resName <span class="ansi-blue-fg">==</span> <span class="ansi-blue-fg">&#34;LYSH&#34;</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">      4</span> print<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#34;NTERM atoms:&#34;</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>atomType<span class="ansi-blue-fg">,</span> ap<span class="ansi-blue-fg">.</span>atomID<span class="ansi-blue-fg">)</span> <span class="ansi-green-fg">for</span> ap <span class="ansi-green-fg">in</span> final_coordinate<span class="ansi-blue-fg">.</span>POSITION <span class="ansi-green-fg">if</span><span class="ansi-blue-fg">(</span>ap<span class="ansi-blue-fg">.</span>resName <span class="ansi-blue-fg">==</span> <span class="ansi-blue-fg">&#34;VAL&#34;</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;final_coordinate&#39; is not defined
</pre></div></div>
</div>
<p>Calculate the distances now to the different CL- ions.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[96]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">coordinate_traj</span><span class="o">.</span><span class="n">distances</span><span class="p">(</span><span class="n">atom_pairs</span><span class="o">=</span><span class="p">[[</span><span class="mi">72</span><span class="p">,</span> <span class="mi">37</span><span class="p">],</span> <span class="p">[</span><span class="mi">73</span><span class="p">,</span> <span class="mi">37</span><span class="p">]])</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s2">&quot;CL-1/2 - ARG-CZ&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;t [ps]&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;r [nm]&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/2849644629.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>coordinate_traj<span class="ansi-blue-fg">.</span>distances<span class="ansi-blue-fg">(</span>atom_pairs<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">72</span><span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">37</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">73</span><span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">37</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;CL-1/2 - ARG-CZ&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;t [ps]&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;r [nm]&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;coordinate_traj&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[97]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">coordinate_traj</span><span class="o">.</span><span class="n">distances</span><span class="p">(</span><span class="n">atom_pairs</span><span class="o">=</span><span class="p">[[</span><span class="mi">72</span><span class="p">,</span> <span class="mi">53</span><span class="p">],</span> <span class="p">[</span><span class="mi">73</span><span class="p">,</span> <span class="mi">53</span><span class="p">]])</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s2">&quot;CL-1/2 - LYSH-NZ&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;t [ps]&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;r [nm]&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/3432102820.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>coordinate_traj<span class="ansi-blue-fg">.</span>distances<span class="ansi-blue-fg">(</span>atom_pairs<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">72</span><span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">53</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">73</span><span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">53</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;CL-1/2 - LYSH-NZ&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;t [ps]&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;r [nm]&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;coordinate_traj&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[98]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">coordinate_traj</span><span class="o">.</span><span class="n">distances</span><span class="p">(</span><span class="n">atom_pairs</span><span class="o">=</span><span class="p">[[</span><span class="mi">72</span><span class="p">,</span> <span class="mi">3</span><span class="p">],</span> <span class="p">[</span><span class="mi">73</span><span class="p">,</span> <span class="mi">3</span><span class="p">]])</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="s2">&quot;CL-1/2 - N-TERM&quot;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s2">&quot;t [ps]&quot;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;r [nm]&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
<span class="ansi-red-fg">---------------------------------------------------------------------------</span>
<span class="ansi-red-fg">NameError</span>                                 Traceback (most recent call last)
<span class="ansi-green-fg">/tmp/ipykernel_32503/5725001.py</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span>
<span class="ansi-green-fg">----&gt; 1</span><span class="ansi-red-fg"> </span>coordinate_traj<span class="ansi-blue-fg">.</span>distances<span class="ansi-blue-fg">(</span>atom_pairs<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">72</span><span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">3</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">,</span> <span class="ansi-blue-fg">[</span><span class="ansi-cyan-fg">73</span><span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">3</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;CL-1/2 - N-TERM&#34;</span><span class="ansi-blue-fg">,</span> legend<span class="ansi-blue-fg">=</span><span class="ansi-green-fg">True</span><span class="ansi-blue-fg">,</span> xlabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;t [ps]&#34;</span><span class="ansi-blue-fg">,</span> ylabel<span class="ansi-blue-fg">=</span><span class="ansi-blue-fg">&#34;r [nm]&#34;</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">NameError</span>: name &#39;coordinate_traj&#39; is not defined
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
<script type="application/vnd.jupyter.widget-state+json">
{"state": {"ef5b63ac2221400eb26f5a757ddeb918": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "0510be9c94334d1db60232e2545d4e9a": {"model_name": "ColormakerRegistryModel", "model_module": "nglview-js-widgets", "model_module_version": "3.0.1", "state": {"_dom_classes": [], "_model_module": "nglview-js-widgets", "_model_module_version": "3.0.1", "_model_name": "ColormakerRegistryModel", "_msg_ar": [], "_msg_q": [], "_ready": false, "_view_count": null, "_view_module": "nglview-js-widgets", "_view_module_version": "3.0.1", "_view_name": "ColormakerRegistryView", "layout": "IPY_MODEL_ef5b63ac2221400eb26f5a757ddeb918"}}, "e6bd327f1914475fa07705522900407c": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "ddfa8573606e4e0e9b14d29ea3130984": {"model_name": "NGLModel", "model_module": "nglview-js-widgets", "model_module_version": "3.0.1", "state": {"_camera_orientation": [], "_camera_str": "orthographic", "_dom_classes": [], "_gui_theme": null, "_ibtn_fullscreen": "IPY_MODEL_6fa1c8bdc03a4781b663b89d99028431", "_igui": null, "_iplayer": "IPY_MODEL_ab38310ee0a2418b940080d9fd70f7ba", "_model_module": "nglview-js-widgets", "_model_module_version": "3.0.1", "_model_name": "NGLModel", "_ngl_color_dict": {}, "_ngl_coordinate_resource": {}, "_ngl_full_stage_parameters": {}, "_ngl_msg_archive": [{"target": "Stage", "type": "call_method", "methodName": "loadFile", "reconstruc_color_scheme": false, "args": [{"type": "blob", "data": "REMARK   1 CREATED WITH MDTraj 1.9.7, 2022-03-31\nCRYST1    0.000    0.000    0.000   0.00   0.00   0.00 P 1           1 \nMODEL        0\nATOM      1  H   VAL A   1       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM      2  H2  VAL A   1       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM      3  N   VAL A   1      11.962  14.133  15.298  1.00  0.00      0    N  \nATOM      4  H3  VAL A   1       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM      5  CA  VAL A   1      12.373  13.224  14.219  1.00  0.00      0    C  \nATOM      6  CB  VAL A   1      11.161  12.687  13.454  1.00  0.00      0    C  \nATOM      7  CG1 VAL A   1      10.417  13.695  12.576  1.00  0.00      0    C  \nATOM      8  CG2 VAL A   1      11.578  11.481  12.611  1.00  0.00      0    C  \nATOM      9  C   VAL A   1      13.357  13.933  13.286  1.00  0.00      0    C  \nATOM     10  O   VAL A   1      14.478  13.442  13.166  1.00  0.00      0    O  \nATOM     11  N   TYR A   2      12.978  15.143  12.885  1.00  0.00      0    N  \nATOM     12  H   TYR A   2       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     13  CA  TYR A   2      13.818  15.944  11.983  1.00  0.00      0    C  \nATOM     14  CB  TYR A   2      13.208  17.313  11.677  1.00  0.00      0    C  \nATOM     15  CG  TYR A   2      13.167  18.287  12.855  1.00  0.00      0    C  \nATOM     16  CD1 TYR A   2      12.093  18.237  13.736  1.00  0.00      0    C  \nATOM     17  HD1 TYR A   2       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     18  CD2 TYR A   2      14.204  19.196  13.030  1.00  0.00      0    C  \nATOM     19  HD2 TYR A   2       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     20  CE1 TYR A   2      12.041  19.143  14.789  1.00  0.00      0    C  \nATOM     21  HE1 TYR A   2       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     22  CE2 TYR A   2      14.157  20.099  14.086  1.00  0.00      0    C  \nATOM     23  HE2 TYR A   2       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     24  CZ  TYR A   2      13.065  20.069  14.946  1.00  0.00      0    C  \nATOM     25  OH  TYR A   2      13.030  21.037  15.900  1.00  0.00      0    O  \nATOM     26  HH  TYR A   2       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     27  C   TYR A   2      15.179  16.177  12.642  1.00  0.00      0    C  \nATOM     28  O   TYR A   2      16.249  15.861  12.124  1.00  0.00      0    O  \nATOM     29  N   ARG A   3      15.144  16.571  13.911  1.00  0.00      0    N  \nATOM     30  H   ARG A   3       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     31  CA  ARG A   3      16.424  16.800  14.597  1.00  0.00      0    C  \nATOM     32  CB  ARG A   3      16.170  17.211  16.048  1.00  0.00      0    C  \nATOM     33  CG  ARG A   3      15.399  16.173  16.865  1.00  0.00      0    C  \nATOM     34  CD  ARG A   3      16.089  16.060  18.225  1.00  0.00      0    C  \nATOM     35  NE  ARG A   3      15.274  15.255  19.147  1.00  0.00      0    N  \nATOM     36  HE  ARG A   3       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     37  CZ  ARG A   3      15.653  14.920  20.388  1.00  0.00      0    C  \nATOM     38  NH1 ARG A   3      16.864  15.257  20.851  1.00  0.00      0    N  \nATOM     39 HH11 ARG A   3       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     40 HH12 ARG A   3       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     41  NH2 ARG A   3      14.852  14.186  21.172  1.00  0.00      0    N  \nATOM     42 HH21 ARG A   3       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     43 HH22 ARG A   3       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     44  C   ARG A   3      17.299  15.545  14.571  1.00  0.00      0    C  \nATOM     45  O   ARG A   3      18.454  15.595  14.150  1.00  0.00      0    O  \nATOM     46  N   LYS A   4      16.682  14.436  14.970  1.00  0.00      0    N  \nATOM     47  H   LYS A   4       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     48  CA  LYS A   4      17.388  13.148  15.027  1.00  0.00      0    C  \nATOM     49  CB  LYS A   4      16.465  11.968  15.338  1.00  0.00      0    C  \nATOM     50  CG  LYS A   4      15.723  12.130  16.667  1.00  0.00      0    C  \nATOM     51  CD  LYS A   4      14.971  10.821  16.914  1.00  0.00      0    C  \nATOM     52  CE  LYS A   4      13.965  10.972  18.056  1.00  0.00      0    C  \nATOM     53  NZ  LYS A   4      13.210   9.745  18.347  1.00  0.00      0    N  \nATOM     54  HZ1 LYS A   4       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     55  HZ2 LYS A   4       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     56  HZ3 LYS A   4       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     57  C   LYS A   4      17.983  12.876  13.644  1.00  0.00      0    C  \nATOM     58  O   LYS A   4      19.175  12.581  13.590  1.00  0.00      0    O  \nATOM     59  N   GLN A   5      17.161  12.936  12.600  1.00  0.00      0    N  \nATOM     60  H   GLN A   5       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     61  CA  GLN A   5      17.694  12.751  11.243  1.00  0.00      0    C  \nATOM     62  CB  GLN A   5      16.567  13.002  10.238  1.00  0.00      0    C  \nATOM     63  CG  GLN A   5      16.270  14.468   9.917  1.00  0.00      0    C  \nATOM     64  CD  GLN A   5      15.231  14.735   8.827  1.00  0.00      0    C  \nATOM     65  OE1 GLN A   5      15.015  13.909   7.942  1.00  0.00      0    O  \nATOM     66  NE2 GLN A   5      14.603  15.905   8.912  1.00  0.00      0    N  \nATOM     67 HE21 GLN A   5       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     68 HE22 GLN A   5       0.000   0.000   0.000  1.00  0.00      0    H  \nATOM     69  C   GLN A   5      18.929  13.599  10.933  1.00  0.00      0    C  \nATOM     70  O   GLN A   5      19.919  12.968  10.503  1.00  0.00      0    O  \nATOM     71  OXT GLN A   5      18.851  14.826  11.160  1.00  0.00      0    O  \nTER      72      GLN A   5\nENDMDL\nCONECT   57   59\nCONECT   59   57\nEND\n", "binary": false}], "kwargs": {"name": "nglview.adaptor.MDTrajTrajectory", "defaultRepresentation": false, "ext": "pdb"}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "removeAllRepresentations", "reconstruc_color_scheme": false, "args": [], "kwargs": {}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "addRepresentation", "reconstruc_color_scheme": false, "args": ["cartoon"], "kwargs": {"sele": "1 2 3 4 5", "color": "residueindex"}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "addRepresentation", "reconstruc_color_scheme": false, "args": ["cartoon"], "kwargs": {"sele": "ARG GLN VAL TYR LYS", "color": "residueindex"}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "addRepresentation", "reconstruc_color_scheme": false, "args": ["hyperball"], "kwargs": {"sele": "ARG GLN VAL TYR LYS"}}], "_ngl_original_stage_parameters": {}, "_ngl_repr_dict": {}, "_ngl_serialize": false, "_ngl_version": "", "_ngl_view_id": [], "_player_dict": {}, "_scene_position": {}, "_scene_rotation": {}, "_synced_model_ids": [], "_synced_repr_model_ids": [], "_view_count": null, "_view_height": "", "_view_module": "nglview-js-widgets", "_view_module_version": "3.0.1", "_view_name": "NGLView", "_view_width": "", "background": "white", "frame": 0, "gui_style": null, "layout": "IPY_MODEL_e6bd327f1914475fa07705522900407c", "max_frame": 0, "n_components": 0, "picked": {}}}, "97369831b3374f5c8149583ed02e0916": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "9576726f4016427d89c0ae3cdeac10be": {"model_name": "ImageModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "ImageModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "ImageView", "format": "png", "height": "", "layout": "IPY_MODEL_97369831b3374f5c8149583ed02e0916", "width": "900.0"}, "buffers": [{"data": "", "encoding": "base64", "path": ["value"]}]}, "54ab859052254339bae2d50b2e387ff1": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "b14adc6584b14074beac9e59167289f8": {"model_name": "DescriptionStyleModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "DescriptionStyleModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "StyleView", "description_width": ""}}, "7206372c10e442238d8dfa7d2ebc9486": {"model_name": "PlayModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "PlayModel", "_playing": false, "_repeat": false, "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "PlayView", "description": "", "description_tooltip": null, "disabled": false, "interval": 100, "layout": "IPY_MODEL_54ab859052254339bae2d50b2e387ff1", "max": 0, "min": 0, "show_repeat": true, "step": 1, "style": "IPY_MODEL_b14adc6584b14074beac9e59167289f8", "value": 0}}, "e7bf6e20fddd4c98a27460cd40a3ce40": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "23c5a1d524834805b0da3cf07f944f1f": {"model_name": "SliderStyleModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "SliderStyleModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "StyleView", "description_width": "", "handle_color": null}}, "80c71eef5ff14c4a947ff225049a7ed1": {"model_name": "IntSliderModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "IntSliderModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "IntSliderView", "continuous_update": true, "description": "", "description_tooltip": null, "disabled": false, "layout": "IPY_MODEL_e7bf6e20fddd4c98a27460cd40a3ce40", "max": 0, "min": 0, "orientation": "horizontal", "readout": true, "readout_format": "d", "step": 1, "style": "IPY_MODEL_23c5a1d524834805b0da3cf07f944f1f", "value": 0}}, "054e6cc74f4d4447bd07340495460839": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "ab38310ee0a2418b940080d9fd70f7ba": {"model_name": "HBoxModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "HBoxModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "HBoxView", "box_style": "", "children": ["IPY_MODEL_7206372c10e442238d8dfa7d2ebc9486", "IPY_MODEL_80c71eef5ff14c4a947ff225049a7ed1"], "layout": "IPY_MODEL_054e6cc74f4d4447bd07340495460839"}}, "8c2472f04f074010a332b298c65b8187": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_7206372c10e442238d8dfa7d2ebc9486", "value"], "target": ["IPY_MODEL_80c71eef5ff14c4a947ff225049a7ed1", "value"]}}, "4468d158ec8a4301b367447af7c99c39": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_7206372c10e442238d8dfa7d2ebc9486", "value"], "target": ["IPY_MODEL_ddfa8573606e4e0e9b14d29ea3130984", "frame"]}}, "ed65091ade3d480382ca87827eaeb65b": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_7206372c10e442238d8dfa7d2ebc9486", "max"], "target": ["IPY_MODEL_ddfa8573606e4e0e9b14d29ea3130984", "max_frame"]}}, "a98c0b92870548d1acbf28400fb92495": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_80c71eef5ff14c4a947ff225049a7ed1", "max"], "target": ["IPY_MODEL_ddfa8573606e4e0e9b14d29ea3130984", "max_frame"]}}, "c90614f9ade040328c395095593911d7": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": "34px"}}, "da295905be4f43919c4ccfb08debee36": {"model_name": "ButtonStyleModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "ButtonStyleModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "StyleView", "button_color": null, "font_weight": ""}}, "6fa1c8bdc03a4781b663b89d99028431": {"model_name": "ButtonModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "ButtonModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "ButtonView", "button_style": "", "description": "", "disabled": false, "icon": "compress", "layout": "IPY_MODEL_c90614f9ade040328c395095593911d7", "style": "IPY_MODEL_da295905be4f43919c4ccfb08debee36", "tooltip": ""}}, "15247cab89fd4d8897986dd47440a3d0": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "2eb3167caf424bfa8a829e101be1a6e3": {"model_name": "NGLModel", "model_module": "nglview-js-widgets", "model_module_version": "3.0.1", "state": {"_camera_orientation": [], "_camera_str": "orthographic", "_dom_classes": [], "_gui_theme": null, "_ibtn_fullscreen": "IPY_MODEL_144b766967f14c2fa514a09e6658089e", "_igui": null, "_iplayer": "IPY_MODEL_547091ec4f8f4d4fa896aa1e768de37d", "_model_module": "nglview-js-widgets", "_model_module_version": "3.0.1", "_model_name": "NGLModel", "_ngl_color_dict": {}, "_ngl_coordinate_resource": {}, "_ngl_full_stage_parameters": {}, "_ngl_msg_archive": [{"target": "Stage", "type": "call_method", "methodName": "loadFile", "reconstruc_color_scheme": false, "args": [{"type": "blob", "data": "REMARK   1 CREATED WITH MDTraj 1.9.7, 2022-03-31\nCRYST1    0.000    0.000    0.000   0.00   0.00   0.00 P 1           1 \nMODEL        0\nATOM      1  H   VAL A   1      12.418  15.016  15.183  1.00  0.00      0    H  \nATOM      2  H2  VAL A   1      12.216  13.739  16.181  1.00  0.00      0    H  \nATOM      3  N   VAL A   1      11.962  14.133  15.298  1.00  0.00      0    N  \nATOM      4  H3  VAL A   1      10.971  14.262  15.266  1.00  0.00      0    H  \nATOM      5  CA  VAL A   1      12.373  13.224  14.219  1.00  0.00      0    C  \nATOM      6  CB  VAL A   1      11.161  12.687  13.454  1.00  0.00      0    C  \nATOM      7  CG1 VAL A   1      10.417  13.695  12.576  1.00  0.00      0    C  \nATOM      8  CG2 VAL A   1      11.578  11.481  12.611  1.00  0.00      0    C  \nATOM      9  C   VAL A   1      13.357  13.933  13.286  1.00  0.00      0    C  \nATOM     10  O   VAL A   1      14.478  13.442  13.166  1.00  0.00      0    O  \nATOM     11  N   TYR A   2      12.978  15.143  12.885  1.00  0.00      0    N  \nATOM     12  H   TYR A   2      12.106  15.515  13.203  1.00  0.00      0    H  \nATOM     13  CA  TYR A   2      13.818  15.944  11.983  1.00  0.00      0    C  \nATOM     14  CB  TYR A   2      13.208  17.313  11.677  1.00  0.00      0    C  \nATOM     15  CG  TYR A   2      13.167  18.287  12.855  1.00  0.00      0    C  \nATOM     16  CD1 TYR A   2      12.093  18.237  13.736  1.00  0.00      0    C  \nATOM     17  HD1 TYR A   2      11.303  17.498  13.603  1.00  0.00      0    H  \nATOM     18  CD2 TYR A   2      14.204  19.196  13.030  1.00  0.00      0    C  \nATOM     19  HD2 TYR A   2      15.050  19.201  12.343  1.00  0.00      0    H  \nATOM     20  CE1 TYR A   2      12.041  19.143  14.789  1.00  0.00      0    C  \nATOM     21  HE1 TYR A   2      11.203  19.127  15.486  1.00  0.00      0    H  \nATOM     22  CE2 TYR A   2      14.157  20.099  14.086  1.00  0.00      0    C  \nATOM     23  HE2 TYR A   2      14.962  20.818  14.237  1.00  0.00      0    H  \nATOM     24  CZ  TYR A   2      13.065  20.069  14.946  1.00  0.00      0    C  \nATOM     25  OH  TYR A   2      13.030  21.037  15.900  1.00  0.00      0    O  \nATOM     26  HH  TYR A   2      13.859  21.593  15.842  1.00  0.00      0    H  \nATOM     27  C   TYR A   2      15.179  16.177  12.642  1.00  0.00      0    C  \nATOM     28  O   TYR A   2      16.249  15.861  12.124  1.00  0.00      0    O  \nATOM     29  N   ARG A   3      15.144  16.571  13.911  1.00  0.00      0    N  \nATOM     30  H   ARG A   3      14.274  16.707  14.384  1.00  0.00      0    H  \nATOM     31  CA  ARG A   3      16.424  16.800  14.597  1.00  0.00      0    C  \nATOM     32  CB  ARG A   3      16.170  17.211  16.048  1.00  0.00      0    C  \nATOM     33  CG  ARG A   3      15.399  16.173  16.865  1.00  0.00      0    C  \nATOM     34  CD  ARG A   3      16.089  16.060  18.225  1.00  0.00      0    C  \nATOM     35  NE  ARG A   3      15.274  15.255  19.147  1.00  0.00      0    N  \nATOM     36  HE  ARG A   3      14.381  14.938  18.828  1.00  0.00      0    H  \nATOM     37  CZ  ARG A   3      15.653  14.920  20.388  1.00  0.00      0    C  \nATOM     38  NH1 ARG A   3      16.864  15.257  20.851  1.00  0.00      0    N  \nATOM     39 HH11 ARG A   3      17.134  15.000  21.779  1.00  0.00      0    H  \nATOM     40 HH12 ARG A   3      17.498  15.765  20.268  1.00  0.00      0    H  \nATOM     41  NH2 ARG A   3      14.852  14.186  21.172  1.00  0.00      0    N  \nATOM     42 HH21 ARG A   3      15.145  13.942  22.096  1.00  0.00      0    H  \nATOM     43 HH22 ARG A   3      13.962  13.883  20.833  1.00  0.00      0    H  \nATOM     44  C   ARG A   3      17.299  15.545  14.571  1.00  0.00      0    C  \nATOM     45  O   ARG A   3      18.454  15.595  14.150  1.00  0.00      0    O  \nATOM     46  N   LYS A   4      16.682  14.436  14.970  1.00  0.00      0    N  \nATOM     47  H   LYS A   4      15.719  14.479  15.236  1.00  0.00      0    H  \nATOM     48  CA  LYS A   4      17.388  13.148  15.027  1.00  0.00      0    C  \nATOM     49  CB  LYS A   4      16.465  11.968  15.338  1.00  0.00      0    C  \nATOM     50  CG  LYS A   4      15.723  12.130  16.667  1.00  0.00      0    C  \nATOM     51  CD  LYS A   4      14.971  10.821  16.914  1.00  0.00      0    C  \nATOM     52  CE  LYS A   4      13.965  10.972  18.056  1.00  0.00      0    C  \nATOM     53  NZ  LYS A   4      13.210   9.745  18.347  1.00  0.00      0    N  \nATOM     54  HZ1 LYS A   4      13.503   9.018  17.726  1.00  0.00      0    H  \nATOM     55  HZ2 LYS A   4      12.233   9.917  18.223  1.00  0.00      0    H  \nATOM     56  HZ3 LYS A   4      13.379   9.465  19.292  1.00  0.00      0    H  \nATOM     57  C   LYS A   4      17.983  12.876  13.644  1.00  0.00      0    C  \nATOM     58  O   LYS A   4      19.175  12.581  13.590  1.00  0.00      0    O  \nATOM     59  N   GLN A   5      17.161  12.936  12.600  1.00  0.00      0    N  \nATOM     60  H   GLN A   5      16.185  13.106  12.738  1.00  0.00      0    H  \nATOM     61  CA  GLN A   5      17.694  12.751  11.243  1.00  0.00      0    C  \nATOM     62  CB  GLN A   5      16.567  13.002  10.238  1.00  0.00      0    C  \nATOM     63  CG  GLN A   5      16.270  14.468   9.917  1.00  0.00      0    C  \nATOM     64  CD  GLN A   5      15.231  14.735   8.827  1.00  0.00      0    C  \nATOM     65  OE1 GLN A   5      15.015  13.909   7.942  1.00  0.00      0    O  \nATOM     66  NE2 GLN A   5      14.603  15.905   8.912  1.00  0.00      0    N  \nATOM     67 HE21 GLN A   5      13.910  16.151   8.235  1.00  0.00      0    H  \nATOM     68 HE22 GLN A   5      14.824  16.539   9.653  1.00  0.00      0    H  \nATOM     69  C   GLN A   5      18.929  13.599  10.933  1.00  0.00      0    C  \nATOM     70  O   GLN A   5      19.919  12.968  10.503  1.00  0.00      0    O  \nATOM     71  OXT GLN A   5      18.851  14.826  11.160  1.00  0.00      0    O  \nTER      72      GLN A   5\nENDMDL\nCONECT   57   59\nCONECT   59   57\nEND\n", "binary": false}], "kwargs": {"name": "nglview.adaptor.MDTrajTrajectory", "defaultRepresentation": false, "ext": "pdb"}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "removeAllRepresentations", "reconstruc_color_scheme": false, "args": [], "kwargs": {}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "addRepresentation", "reconstruc_color_scheme": false, "args": ["cartoon"], "kwargs": {"sele": "1 2 3 4 5", "color": "residueindex"}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "addRepresentation", "reconstruc_color_scheme": false, "args": ["cartoon"], "kwargs": {"sele": "ARG GLN VAL TYR LYS", "color": "residueindex"}}, {"component_index": 0, "target": "compList", "type": "call_method", "methodName": "addRepresentation", "reconstruc_color_scheme": false, "args": ["hyperball"], "kwargs": {"sele": "ARG GLN VAL TYR LYS"}}], "_ngl_original_stage_parameters": {}, "_ngl_repr_dict": {}, "_ngl_serialize": false, "_ngl_version": "", "_ngl_view_id": [], "_player_dict": {}, "_scene_position": {}, "_scene_rotation": {}, "_synced_model_ids": [], "_synced_repr_model_ids": [], "_view_count": null, "_view_height": "", "_view_module": "nglview-js-widgets", "_view_module_version": "3.0.1", "_view_name": "NGLView", "_view_width": "", "background": "white", "frame": 0, "gui_style": null, "layout": "IPY_MODEL_15247cab89fd4d8897986dd47440a3d0", "max_frame": 0, "n_components": 0, "picked": {}}}, "3f309ae1f7374cd3b0ad61281ec45550": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "e204a385cd4e49b5af84af4112792f62": {"model_name": "ImageModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "ImageModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "ImageView", "format": "png", "height": "", "layout": "IPY_MODEL_3f309ae1f7374cd3b0ad61281ec45550", "width": "900.0"}, "buffers": [{"data": "", "encoding": "base64", "path": ["value"]}]}, "e16efc55438142cb9802b35b7348f1e5": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "094574451ab74a0fb038b53384a80aaf": {"model_name": "DescriptionStyleModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "DescriptionStyleModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "StyleView", "description_width": ""}}, "daad4855d90f41ea8d7076518665e724": {"model_name": "PlayModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "PlayModel", "_playing": false, "_repeat": false, "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "PlayView", "description": "", "description_tooltip": null, "disabled": false, "interval": 100, "layout": "IPY_MODEL_e16efc55438142cb9802b35b7348f1e5", "max": 0, "min": 0, "show_repeat": true, "step": 1, "style": "IPY_MODEL_094574451ab74a0fb038b53384a80aaf", "value": 0}}, "9ee67891b09f4664a3dae2cac00de2e8": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "b5463a8c697042b986483dc7865c9573": {"model_name": "SliderStyleModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "SliderStyleModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "StyleView", "description_width": "", "handle_color": null}}, "94eb1535b76e44d68af83bc654b9edf6": {"model_name": "IntSliderModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "IntSliderModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "IntSliderView", "continuous_update": true, "description": "", "description_tooltip": null, "disabled": false, "layout": "IPY_MODEL_9ee67891b09f4664a3dae2cac00de2e8", "max": 0, "min": 0, "orientation": "horizontal", "readout": true, "readout_format": "d", "step": 1, "style": "IPY_MODEL_b5463a8c697042b986483dc7865c9573", "value": 0}}, "d3a193bbf5e84947bd2899dc1ee994b0": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": null}}, "547091ec4f8f4d4fa896aa1e768de37d": {"model_name": "HBoxModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "HBoxModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "HBoxView", "box_style": "", "children": ["IPY_MODEL_daad4855d90f41ea8d7076518665e724", "IPY_MODEL_94eb1535b76e44d68af83bc654b9edf6"], "layout": "IPY_MODEL_d3a193bbf5e84947bd2899dc1ee994b0"}}, "e32e17de44044227bdeb01d2c00a4c05": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_daad4855d90f41ea8d7076518665e724", "value"], "target": ["IPY_MODEL_94eb1535b76e44d68af83bc654b9edf6", "value"]}}, "ee5827c81d084d65b1fb668cf06b574b": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_daad4855d90f41ea8d7076518665e724", "value"], "target": ["IPY_MODEL_2eb3167caf424bfa8a829e101be1a6e3", "frame"]}}, "283e425ddcd84d698cc5c6aa6632c659": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_daad4855d90f41ea8d7076518665e724", "max"], "target": ["IPY_MODEL_2eb3167caf424bfa8a829e101be1a6e3", "max_frame"]}}, "5244dbb6ce4d4a03ba283134aa2d65bc": {"model_name": "LinkModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "LinkModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": null, "source": ["IPY_MODEL_94eb1535b76e44d68af83bc654b9edf6", "max"], "target": ["IPY_MODEL_2eb3167caf424bfa8a829e101be1a6e3", "max_frame"]}}, "260adfee5aa74267982a56c3b66259f8": {"model_name": "LayoutModel", "model_module": "@jupyter-widgets/base", "model_module_version": "1.2.0", "state": {"_model_module": "@jupyter-widgets/base", "_model_module_version": "1.2.0", "_model_name": "LayoutModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "LayoutView", "align_content": null, "align_items": null, "align_self": null, "border": null, "bottom": null, "display": null, "flex": null, "flex_flow": null, "grid_area": null, "grid_auto_columns": null, "grid_auto_flow": null, "grid_auto_rows": null, "grid_column": null, "grid_gap": null, "grid_row": null, "grid_template_areas": null, "grid_template_columns": null, "grid_template_rows": null, "height": null, "justify_content": null, "justify_items": null, "left": null, "margin": null, "max_height": null, "max_width": null, "min_height": null, "min_width": null, "object_fit": null, "object_position": null, "order": null, "overflow": null, "overflow_x": null, "overflow_y": null, "padding": null, "right": null, "top": null, "visibility": null, "width": "34px"}}, "298b107fb2264149846066934a0ad218": {"model_name": "ButtonStyleModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "ButtonStyleModel", "_view_count": null, "_view_module": "@jupyter-widgets/base", "_view_module_version": "1.2.0", "_view_name": "StyleView", "button_color": null, "font_weight": ""}}, "144b766967f14c2fa514a09e6658089e": {"model_name": "ButtonModel", "model_module": "@jupyter-widgets/controls", "model_module_version": "1.5.0", "state": {"_dom_classes": [], "_model_module": "@jupyter-widgets/controls", "_model_module_version": "1.5.0", "_model_name": "ButtonModel", "_view_count": null, "_view_module": "@jupyter-widgets/controls", "_view_module_version": "1.5.0", "_view_name": "ButtonView", "button_style": "", "description": "", "disabled": false, "icon": "compress", "layout": "IPY_MODEL_260adfee5aa74267982a56c3b66259f8", "style": "IPY_MODEL_298b107fb2264149846066934a0ad218", "tooltip": ""}}}, "version_major": 2, "version_minor": 0}
</script></section>
</section>
</section>
</section>


           </div>

          </div>
          <footer>

    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">

        <a href="../_source/modules.html" class="btn btn-neutral float-right" title="Documentation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>


        <a href="index.html" class="btn btn-neutral float-left" title="Tutorials" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>

    </div>


  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Benjamin Ries, Marc Lehner, Salome Rieder. Project structure based on the Computational Molecular Science Python Cookiecutter version 1.3

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.

</footer>

        </div>
      </div>

    </section>

  </div>



  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>






</body>
</html>
