<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to PyGromosTools &mdash; PyGromosTools  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script src="https://unpkg.com/@jupyter-widgets/html-manager@^0.20.1/dist/embed-amd.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Welcome to PyGromosTools" href="introduction.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="#" class="icon icon-home"> PyGromosTools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Welcome to PyGromosTools</a></li>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="Examples/index.html">Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="_source/modules.html">Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="styleguide.html">Style Guide for PyGromosTools</a></li>
<li class="toctree-l1"><a class="reference internal" href="code_of_conduct.html">Contributor Covenant Code of Conduct</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">PyGromosTools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
      <li>Welcome to PyGromosTools</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<a class="reference external image-reference" href=".img/PyGromosToolsLogo_smallerWithBorders.png#gh-light-mode-only"><img alt="PyGromosBanLight" src=".img/PyGromosToolsLogo_smallerWithBorders.png#gh-light-mode-only" /></a>
<section id="welcome-to-pygromostools">
<h1>Welcome to PyGromosTools<a class="headerlink" href="#welcome-to-pygromostools" title="Permalink to this headline"></a></h1>
<a class="reference external image-reference" href="https://github.com/rinikerlab/PyGromosTools/actions/workflows/CI.yaml"><img alt="CI" src="https://github.com/rinikerlab/PyGromosTools/actions/workflows/CI.yaml/badge.svg" /></a>
<a class="reference external image-reference" href="https://github.com/rinikerlab/PyGromosTools/actions/workflows/pre-commit.yml"><img alt="pre-commit" src="https://github.com/rinikerlab/PyGromosTools/actions/workflows/pre-commit.yml/badge.svg?branch=release3" /></a>
<a class="reference external image-reference" href="https://codecov.io/gh/rinikerlab/PyGromosTools"><img alt="codecov" src="https://codecov.io/gh/rinikerlab/PyGromosTools/branch/main/graph/badge.svg?token=R36KJCEKEC" /></a>
<a class="reference external image-reference" href="https://lgtm.com/projects/g/rinikerlab/PyGromosTools/context:python"><img alt="Language grade: Python" src="https://img.shields.io/lgtm/grade/python/g/rinikerlab/PyGromosTools.svg?logo=lgtm&amp;logoWidth=18" /></a>
<a class="reference external image-reference" href="https://zenodo.org/badge/latestdoi/323972251"><img alt="DOI" src="https://zenodo.org/badge/323972251.svg" /></a>
<a class="reference external image-reference" href="https://rinikerlab.github.io/PyGromosTools/"><img alt="Documentation" src="https://img.shields.io/badge/Documentation-here-white.svg" /></a>
<section id="description">
<h2>Description<a class="headerlink" href="#description" title="Permalink to this headline"></a></h2>
<blockquote>
<div><p>The aim of the module is to bring GROMOS to the Python3 World!
This repository should make it easier to work with GROMOS in Python and should enable the user to write cleaner, more reliable and adaptable code.</p>
<p>General informations about functions can be found in our wiki and usage example for many general functions and theire relations are shown in jupyter notebooks in the examples in the example folder.</p>
</div></blockquote>
</section>
<section id="content">
<h2>Content<a class="headerlink" href="#content" title="Permalink to this headline"></a></h2>
<ul>
<li><p>GROMOS wrappers</p>
<ul class="simple">
<li><p>GromosXX wrapper: for simulation execution</p></li>
<li><p>GromosPP wrapper: for GROMOS++ program usage</p></li>
</ul>
</li>
<li><p>File handling of all GROMOS file types for automated creation/modification/analysis :</p>
<ul>
<li><p>coordinate files CNF:</p>
<ul class="simple">
<li><p>read and analyse CNF files</p></li>
<li><p>generate CNF files from RDKit</p></li>
<li><p>generate CNF files from SDF</p></li>
</ul>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">cnf</span> <span class="o">=</span> <span class="n">Cnf</span><span class="p">(</span><span class="n">input_value</span><span class="o">=</span><span class="s2">&quot;file_name&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">cnf</span><span class="o">.</span><span class="n">GENBOX</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>topology files:</p>
<ul class="simple">
<li><p>create topologies from a forcefield</p>
<ul>
<li><p>GROMOS 2016H66 / 54A7</p></li>
<li><p>OpenForceField</p></li>
<li><p>SerenityForceField</p></li>
</ul>
</li>
<li><p>modify topologies</p>
<ul>
<li><p>add new atoms</p></li>
<li><p>modify force parameters</p></li>
</ul>
</li>
</ul>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">top</span> <span class="o">=</span> <span class="n">Top</span><span class="p">(</span><span class="n">input_value</span><span class="o">=</span><span class="s2">&quot;file_path&quot;</span><span class="p">)</span>
<span class="n">top</span><span class="o">.</span><span class="n">add_new_SOLUTEATOM</span><span class="p">(</span><span class="n">ATNM</span><span class="o">=</span><span class="mi">42</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">top</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>simulation parameter files IMD</p>
<ul class="simple">
<li><p>a wide option of templates provided</p></li>
<li><p>modify IMD files to fit your simulation</p></li>
</ul>
<div class="highlight-pythons notranslate"><div class="highlight"><pre><span></span>imd = Imd(input_value=&quot;file_path&quot;)
imd.INITIALISE.TEMPI = 137
print(imd)
</pre></div>
</div>
</li>
<li><p>trajectories (tre, trc, trg, …)</p>
<ul class="simple">
<li><p>analyse trajectories with Pandas data frames</p></li>
<li><p>standard analysis like RSMD, RDF, … for trc</p></li>
<li><p>auto saving of results for later use as hdf5</p></li>
<li><p>ene_ana like tools for tre</p></li>
<li><p>easy to add costume analysis tools</p></li>
</ul>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">trc</span> <span class="o">=</span> <span class="n">Trc</span><span class="p">(</span><span class="n">input_value</span><span class="o">=</span><span class="s2">&quot;file_path&quot;</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">trc</span><span class="o">.</span><span class="n">rmsd</span><span class="p">()</span><span class="o">.</span><span class="n">mean</span><span class="p">())</span>
</pre></div>
</div>
</li>
<li><p>replica exchange files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">repdat</span><span class="o">.</span><span class="n">dat</span>
</pre></div>
</div>
</li>
<li><p>classes for single blocks of each of these files.</p></li>
</ul>
</li>
<li><p>Automation and file management system <code class="docutils literal notranslate"><span class="pre">gromos_system</span></code></p>
<ul class="simple">
<li><p>offers clean file management for simulations</p></li>
<li><p>offers a high level of automation</p></li>
<li><p>equiped with simulation queuing system</p></li>
<li><p>includes many force fields</p></li>
</ul>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">ff</span><span class="o">=</span><span class="n">forcefield_system</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s2">&quot;openforcefield&quot;</span><span class="p">)</span>
<span class="n">gsys</span> <span class="o">=</span> <span class="n">Gromos_System</span><span class="p">(</span><span class="n">work_folder</span><span class="o">=</span><span class="s2">&quot;dir&quot;</span><span class="p">,</span> <span class="n">in_smiles</span><span class="o">=</span><span class="s2">&quot;C1CCCCC1&quot;</span><span class="p">,</span> <span class="n">auto_convert</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">Forcefield</span><span class="o">=</span><span class="n">ff</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">gsys</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Simulation Submission and Execution :</p>
<ul class="simple">
<li><p>Different Types of Simulation modules, like MD, SD or Emin.</p></li>
<li><p>Can be executed locally or on a cluster</p></li>
<li><p>easy to automatize and combine with analysis routines</p></li>
</ul>
<p>Run on a local machine:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.files.gromos_system</span> <span class="kn">import</span> <span class="n">Gromos_System</span>
<span class="kn">from</span> <span class="nn">pygromos.simulations.hpc_queuing.submission_systems.local</span> <span class="kn">import</span> <span class="n">LOCAL</span> <span class="k">as</span> <span class="n">subSystem</span>
<span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">emin</span>

<span class="c1"># define file paths</span>
<span class="n">root_dir</span> <span class="o">=</span> <span class="s2">&quot;./example_files/SD_Simulation&quot;</span>
<span class="n">root_in_dir</span> <span class="o">=</span> <span class="n">root_dir</span><span class="o">+</span><span class="s2">&quot;/SD_input&quot;</span>
<span class="n">cnf_path</span> <span class="o">=</span> <span class="n">root_in_dir</span><span class="o">+</span><span class="s2">&quot;/6J29_unitedatom_optimised_geometry.cnf&quot;</span>
<span class="n">top_path</span> <span class="o">=</span> <span class="n">root_in_dir</span> <span class="o">+</span> <span class="s2">&quot;/6J29.top&quot;</span>
<span class="n">sys_name</span> <span class="o">=</span> <span class="s2">&quot;6J29&quot;</span>

<span class="c1"># Build gromos System</span>
<span class="n">grom_system</span> <span class="o">=</span> <span class="n">Gromos_System</span><span class="p">(</span><span class="n">in_cnf_path</span><span class="o">=</span><span class="n">cnf_path</span><span class="p">,</span> <span class="n">in_top_path</span><span class="o">=</span><span class="n">top_path</span><span class="p">,</span>
                            <span class="n">system_name</span><span class="o">=</span><span class="n">sys_name</span><span class="p">,</span> <span class="n">work_folder</span><span class="o">=</span><span class="n">root_in_dir</span><span class="p">)</span>
<span class="c1"># Run Emin</span>
<span class="n">emin_gromos_system</span><span class="p">,</span> <span class="n">jobID</span> <span class="o">=</span> <span class="n">emin</span><span class="p">(</span><span class="n">in_gromos_system</span><span class="o">=</span><span class="n">grom_system</span><span class="p">,</span> <span class="n">project_dir</span><span class="o">=</span><span class="n">root_dir</span><span class="p">,</span>
                        <span class="n">step_name</span><span class="o">=</span><span class="n">step_name</span><span class="p">,</span> <span class="n">submission_system</span><span class="o">=</span><span class="n">subSystem</span><span class="p">())</span>
</pre></div>
</div>
<p>Run on LSF-Cluster:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pygromos.files.gromos_system</span> <span class="kn">import</span> <span class="n">Gromos_System</span>
<span class="kn">from</span> <span class="nn">pygromos.simulations.hpc_queuing.submission_systems.lsf</span> <span class="kn">import</span> <span class="n">LSF</span> <span class="k">as</span> <span class="n">subSystem</span>
<span class="kn">from</span> <span class="nn">pygromos.simulations.modules.preset_simulation_modules</span> <span class="kn">import</span> <span class="n">emin</span>

<span class="c1"># define file paths</span>
<span class="n">root_dir</span> <span class="o">=</span> <span class="s2">&quot;./example_files/SD_Simulation&quot;</span>
<span class="n">root_in_dir</span> <span class="o">=</span> <span class="n">root_dir</span><span class="o">+</span><span class="s2">&quot;/SD_input&quot;</span>
<span class="n">cnf_path</span> <span class="o">=</span> <span class="n">root_in_dir</span><span class="o">+</span><span class="s2">&quot;/6J29_unitedatom_optimised_geometry.cnf&quot;</span>
<span class="n">top_path</span> <span class="o">=</span> <span class="n">root_in_dir</span> <span class="o">+</span> <span class="s2">&quot;/6J29.top&quot;</span>
<span class="n">sys_name</span> <span class="o">=</span> <span class="s2">&quot;6J29&quot;</span>

<span class="c1"># Build gromos System:</span>
<span class="n">grom_system</span> <span class="o">=</span> <span class="n">Gromos_System</span><span class="p">(</span><span class="n">in_cnf_path</span><span class="o">=</span><span class="n">cnf_path</span><span class="p">,</span> <span class="n">in_top_path</span><span class="o">=</span><span class="n">top_path</span><span class="p">,</span>
                          <span class="n">system_name</span><span class="o">=</span><span class="n">sys_name</span><span class="p">,</span> <span class="n">work_folder</span><span class="o">=</span><span class="n">root_in_dir</span><span class="p">)</span>

<span class="c1"># Run Emin</span>
<span class="n">sub_system</span> <span class="o">=</span> <span class="n">subSystem</span><span class="p">(</span><span class="n">nmpi</span><span class="o">=</span><span class="mi">4</span><span class="p">)</span> <span class="c1"># allows parallelization</span>
<span class="n">emin_gromos_system</span><span class="p">,</span> <span class="n">jobID</span> <span class="o">=</span> <span class="n">emin</span><span class="p">(</span><span class="n">in_gromos_system</span><span class="o">=</span><span class="n">grom_system</span><span class="p">,</span> <span class="n">project_dir</span><span class="o">=</span><span class="n">root_dir</span><span class="p">,</span>
                        <span class="n">step_name</span><span class="o">=</span><span class="n">step_name</span><span class="p">,</span> <span class="n">submission_system</span><span class="o">=</span><span class="n">sub_system</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Other utilities:</p>
<ul class="simple">
<li><p>Bash wrappers for GROMOS</p></li>
<li><p>Amino acid library</p></li>
</ul>
</li>
</ul>
</section>
<section id="quick-setup">
<h2>Quick Setup<a class="headerlink" href="#quick-setup" title="Permalink to this headline"></a></h2>
<p>Quick Start - move to the root folder of this repository:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># build environment</span>
conda env create -f conda_env.yml
conda develop -n pygromos <span class="si">${</span><span class="nv">PWD</span><span class="si">}</span>

<span class="c1"># activate environment</span>
conda activate pygromos
</pre></div>
</div>
<p>Please make sure, that you have GROMOS (www.gromos.net) binaries around, if you want to use the MD-Package. We sadly can not provide the source code for this package, as it is currently not open-source.</p>
<p>If you find a bug or have an feature request, please raise an Issue on GitHub.</p>
<p>For more information, see INSTALL.md file for more informations.</p>
</section>
<section id="contributions">
<h2>Contributions<a class="headerlink" href="#contributions" title="Permalink to this headline"></a></h2>
<p>You want to contribute? Awesome! We are happy to support you in this process.
For any contribution, please check out the CODE_OF_CONDUCT.md file and the style guide in styleguide.md.
There will be a small code revision for code contributions, to verify that everything is in place.</p>
</section>
<section id="publications">
<h2>Publications<a class="headerlink" href="#publications" title="Permalink to this headline"></a></h2>
<p>Scientific Literature using PyGromosTools:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1007/s10822-022-00445-6">RestraintMaker: a graph-based approach to select distance restraints in free-energy calculations with dual topology; Benjamin Ries^, Salomé Rieder^, Clemens Rhiner, Philippe H. Hünenberger and Sereina Riniker (2022).</a></p></li>
<li><p><a class="reference external" href="https://link.springer.com/article/10.1007/s10822-021-00436-z">Relative Free-Energy Calculations for Scaffold Hopping-Type Transformations with an Automated RE-EDS Sampling Procedure; Benjamin Ries, Karl Normak, R.Gregor Weiß, Salomé Rieder, Emília P. Barros, Candide Champion, Gerhard König, Sereina Riniker (2022)</a></p></li>
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/33750117/">Modulation of the Passive Permeability of Semipeptidic Macrocycles: N- and C-Methylations Fine-Tune Conformation and Properties; Christian Comeau, Benjamin Ries, Thomas Stadelmann, et al. (2021)</a></p></li>
</ul>
<p>^ contributed equally</p>
</section>
<section id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permalink to this headline"></a></h2>
<p>Many thanks to Robin Wolf for the logo design!</p>
</section>
<section id="copyright">
<h2>Copyright<a class="headerlink" href="#copyright" title="Permalink to this headline"></a></h2>
<p>Copyright (c) 2020, Benjamin Ries, Marc Lehner, Salome Rieder, Felix Pultar, Paul Katzberger, Candide Champion</p>
<p>Project based on the
<a class="reference external" href="https://github.com/molssi/cookiecutter-cms">Computational Molecular Science Python Cookiecutter</a> version 1.3.</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Welcome to PyGromosTools</a><ul>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#description">Description</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#content">Content</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#quick-setup">Quick Setup</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#contributions">Contributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#publications">Publications</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#acknowledgements">Acknowledgements</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#copyright">Copyright</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation guide</a><ul>
<li class="toctree-l2"><a class="reference internal" href="install.html#environment-setup">Environment Setup</a></li>
<li class="toctree-l2"><a class="reference internal" href="install.html#pygromostools-setup">PyGromosTools Setup</a><ul>
<li class="toctree-l3"><a class="reference internal" href="install.html#using-the-package">Using the Package</a></li>
<li class="toctree-l3"><a class="reference internal" href="install.html#developing-the-package">Developing the Package</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Tutorials/index.html">Tutorials</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Tutorials/tutorial_gromos_pipeline.html">Gromos Tutorial</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Tutorials/tutorial_gromos_pipeline.html#Build-the-input-files">Build the input files</a></li>
<li class="toctree-l3"><a class="reference internal" href="Tutorials/tutorial_gromos_pipeline.html#Solvatistation-and-Solvent-Energy-Minimization">Solvatistation and Solvent Energy Minimization</a></li>
<li class="toctree-l3"><a class="reference internal" href="Tutorials/tutorial_gromos_pipeline.html#Thermalisation-and-equilibration.">Thermalisation and equilibration.</a></li>
<li class="toctree-l3"><a class="reference internal" href="Tutorials/tutorial_gromos_pipeline.html#Molecular-dynamics-sampling-simulation.">Molecular dynamics sampling simulation.</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Examples/index.html">Examples</a><ul>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html">TI Calculation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#Imports">Imports</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#Input-files">Input files</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#Vacuum-Simulation">Vacuum Simulation</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#RUN-Test-SD-EQ">RUN Test SD EQ</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#Further-Analysis:">Further Analysis:</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#Lambda-Sampling">Lambda Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_TI_gromos_simulation.html#Submission">Submission</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_amber2gromos.html">Amber2Gromos Example</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_amber2gromos.html#create-top-+-cnf-for-aniline">create top + cnf for aniline</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_amber2gromos.html#create-top-+-cnf-for-aniline,-solvated-in-CHCl3">create top + cnf for aniline, solvated in CHCl3</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_amber2gromos.html#create-top-+-cnf-for-aniline,-solvated-in-TIP3P-water">create top + cnf for aniline, solvated in TIP3P water</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_gromos_on_HPC_Clusters.html">Usage of the PyGromosTools on HPC-cluster</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_on_HPC_Clusters.html#Initialization">Initialization</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_on_HPC_Clusters.html#Energy-minimization">Energy minimization</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_on_HPC_Clusters.html#Equilibration-followed-by-subsequent-production-runs">Equilibration followed by subsequent production runs</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_gromos_qmmm.html">QM/MM in Gromos</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_qmmm.html#Support-for-QMMM-functionality-in-GROMOS-input-files">Support for QMMM functionality in <code class="docutils literal notranslate"><span class="pre">GROMOS</span></code> input files</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_qmmm.html#Directly-manipulate-a-QMMM-specification-file">Directly manipulate a QMMM specification file</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_qmmm.html#Print-out-and-change-some-blocks-in-the-QMMM-specification-file">Print out and change some blocks in the QMMM specification file</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_gromos_qmmm.html#Run-QM/MM-Simulations">Run QM/MM Simulations</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_hvap.html">Calculation of free energy of evaporization</a></li>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_openff.html">OpenFF Automatic Parametrization</a></li>
<li class="toctree-l2"><a class="reference internal" href="Examples/example_solvation_free_energy.html">Calculation of Self Solvation Free Energy</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_solvation_free_energy.html#Do-Imports">Do Imports</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_solvation_free_energy.html#Choose-Molecule-to-run-calculation-for">Choose Molecule to run calculation for</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_solvation_free_energy.html#Create-The-Solvation-free-energy-calculation-system">Create The Solvation free energy calculation system</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_solvation_free_energy.html#Equilibrate-System">Equilibrate System</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_solvation_free_energy.html#Do-TI-calculation">Do TI calculation</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/example_solvation_free_energy.html#Read-out-results">Read out results</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Examples/developer_examples/index.html">Developer Examples</a><ul>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_PyGromosPP.html">Implemented pyGromosPP programs</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_gromos_files.html">PyGromos File Examples</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_gromos_simulation.html">SD Simulations with Submission System</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_gromos_trajectories.html">Gromos Trajectory evaluation with Pygromos and Pandas</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_gromos_tutorial_on_low_level.html">Gromos Tutorial Pipeline - OLD</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_submission_systems.html">Submission Systems</a></li>
<li class="toctree-l3"><a class="reference internal" href="Examples/developer_examples/dev_example_topology_creation.html">Topology creation and modifications with PyGromosTools</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="_source/modules.html">Documentation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="_source/pygromos.html">pygromos package</a><ul>
<li class="toctree-l3"><a class="reference internal" href="_source/pygromos.html#subpackages">Subpackages</a></li>
<li class="toctree-l3"><a class="reference internal" href="_source/pygromos.html#submodules">Submodules</a></li>
<li class="toctree-l3"><a class="reference internal" href="_source/pygromos.html#module-pygromos.pygromos">pygromos.pygromos module</a></li>
<li class="toctree-l3"><a class="reference internal" href="_source/pygromos.html#module-pygromos">Module contents</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="styleguide.html">Style Guide for PyGromosTools</a><ul>
<li class="toctree-l2"><a class="reference internal" href="styleguide.html#introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="styleguide.html#python-style-guide-in-short">Python style guide in short</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="code_of_conduct.html">Contributor Covenant Code of Conduct</a><ul>
<li class="toctree-l2"><a class="reference internal" href="code_of_conduct.html#our-pledge">Our Pledge</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_of_conduct.html#our-standards">Our Standards</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_of_conduct.html#our-responsibilities">Our Responsibilities</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_of_conduct.html#scope">Scope</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_of_conduct.html#enforcement">Enforcement</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_of_conduct.html#attribution">Attribution</a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="introduction.html" class="btn btn-neutral float-right" title="Welcome to PyGromosTools" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2020, Benjamin Ries, Marc Lehner, Salome Rieder. Project structure based on the Computational Molecular Science Python Cookiecutter version 1.3.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>
